---
title: "Multiple measures of depression to enhance validity of major depressive disorder in the UK Biobank"
author: "Kylie P. Glanville"
output:
  html_document:
    number_sections: yes
    toc: yes
    toc_depth: 4
    toc_float: yes
  word_document:
    toc: yes
    toc_depth: '4'
---

```{r load_libraries, include=FALSE, message=FALSE}

knitr::opts_chunk$set(echo = TRUE)

library(knitr)
library(kableExtra)
library(data.table)
library(VennDiagram)
library(eulerr)
library(scales)
library(gridExtra)
library(Hmisc)
library(ggplot2)
library(tibble)
library(tidyverse)
library(grid)
library(UpSetR)
library(formattable)
library(psychometric)
library(dplyr)
library(qqman)
library(magick)
library(cowplot)
library(corrplot)
library(psych)
library(pROC)
library(randomForest)
library(expss)
library(fmsb)

```

```{bash print_pheno_columns_rosalind_psychiatric, echo=FALSE, eval=FALSE}

# Print required phenotype columns

#Demographics
#   f.21022.0.0    Age.at.recruitment.0.0
#   f.31.0.0       Sex.0.0
#   f.189.0.0      Townsend.deprivation.index.at.recruitment.0.0
#   f.21001.0.0    Body.mass.index.BMI.0.0
#   f.21001.1.0    Body.mass.index.BMI.1.0
#   f.21001.2.0    Body.mass.index.BMI.2.0
#   f.20116.0.0    Smoking.status.0.0
#   f.20116.1.0    Smoking.status.1.0
#   f.20116.2.0    Smoking.status.2.0

awk '{print $1,$8329,$1655,$1943,$8317,$8318,$8319,$7671,$7672,$7673}' /file_path/phenotypes.txt > demographics

#Self-report
#   f.20002.0.0    Non.cancer.illness.code.self.reported.0.0 - 
#   f.20002.2.28   Non.cancer.illness.code.self.reported.2.28

paste <(awk '{print $1}' /file_path/phenotypes.txt) <(cut -f5793-5879 /file_path/phenotypes.txt) > self_report

#Treatment/medications
#   f.20003.0.0    Treatment.medication.code.0.0 - 
#   f.20003.2.47   Treatment.medication.code.2.47

paste <(awk '{print $1}' /file_path/phenotypes.txt) <(cut -f5880-6023 /file_path/phenotypes.txt) > medications

# Touchscreen questions
# 100025 Touchscreen
# ACE touchscreen question "Have you ever seen a psychiatrist for nerves, anxiety, tension or depression?"
# Coding	Meaning
# 1	Yes
# 0	No
# -1	Do not know
# -3	Prefer not to answer

#    f.2090.0.0     Seen.doctor.GP.for.nerves.anxiety.tension.or.depression.0.0
#    f.2090.1.0     Seen.doctor.GP.for.nerves.anxiety.tension.or.depression.1.0
#    f.2090.2.0     Seen.doctor.GP.for.nerves.anxiety.tension.or.depression.2.0
#    f.2100.0.0     Seen.a.psychiatrist.for.nerves.anxiety.tension.or.depression.0.0
#    f.2100.1.0     Seen.a.psychiatrist.for.nerves.anxiety.tension.or.depression.1.0
#    f.2100.2.0     Seen.a.psychiatrist.for.nerves.anxiety.tension.or.depression.2.0

paste <(awk '{print $1}' /file_path/phenotypes.txt) <(cut -f2497-2502 /file_path/phenotypes.txt) > touchscreen

# Print columns from MHQ screening: "Mental.health.problems.ever.diagnosed.by.a.professional". Use these to screen for psychosis in MHQ.

paste <(awk '{print $1}' /file_path/phenotypes.txt) <(cut -f13039-13054 /file_path/phenotypes.txt) > mhq_screening

# Print columns for 20446	"Ever had prolonged feelings of sadness or depression" to get number who completed the MHQ
      
awk '{print $1, $12963}'  /file_path/phenotypes.txt > mhq_participants

# D.Smith definition

#   f.20126.0.0    Bipolar.and.major.depression.status.0.0

# Coding	Meaning
# 0	No Bipolar or Depression
# 1	Bipolar I Disorder
# 2	Bipolar II Disorder
# 3	Probable Recurrent major depression (severe)
# 4	Probable Recurrent major depression (moderate)
# 5	Single Probable major depression episode

awk '{print $1, $7684}' /file_path/phenotypes.txt > smith

# Print columns for 4598	"Ever depressed for a whole week" to get number who completed the touchscreen psychosocial questions 
      
awk '{print $1, $3663, $3664, $3665}' /file_path/phenotypes.txt > smith_participants

```

```{bash recode_icd_self-report_medication_columns_rosalind_psychiatric, echo=FALSE, eval=FALSE}

# To recode self-report columns
# searched self-report codes for psychiatric disorders on UKB website and made "self_report_depression.txt", "self_report_scz.txt", "self_report_bipolar.txt"

# Recode self-report columns to 0/1 using CARS package in R. 

# Depression

library(CARS)
library(data.table)

# read in data
self_report <- read.table("/file_path/self_report", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
coding_self <- as.data.frame(fread("/file_path/self_report_depression.txt", header = F))

# make vector of codes
coding_self <- coding_self[,1]

# make a copy of df of UKB self report 
self_report_copy <- self_report

# make df filled with NA (number of columns = self_report - 1(no ID column))
temp_self_report_copy=as.data.frame(matrix(NA, ncol=87, nrow=nrow(self_report_copy)))

# populate empty df with 0/1 by recoding the self code in each cell, then sum along the rows to create new column for each diagnosis
for (i in coding_self) {
  temp_self_report_copy[,1:87] <- lapply(self_report_copy[,2:88], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  self_report_copy[[paste(i)]] <- rowSums(temp_self_report_copy[,1:87])
}

# print required columns
self_report_copy <- self_report_copy[,c(1,89:ncol(self_report_copy))]

# write table
write.table(self_report_copy, "/file_path/self_report_dep", row.names=F, col.name=T, quote=F)

# Schizophrenia

library(CARS)
library(data.table)

# read in data
self_report <- read.table("/file_path/self_report", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
coding_self <- as.data.frame(fread("/file_path/self_report_scz.txt", header = F))

# make vector of codes
coding_self <- coding_self[,1]

# make a copy of df of UKB self report 
self_report_copy <- self_report

# make df filled with NA (number of columns = self_report - 1(no ID column))
temp_self_report_copy=as.data.frame(matrix(NA, ncol=87, nrow=nrow(self_report_copy)))

# populate empty df with 0/1 by recoding the self code in each cell, then sum along the rows to create new column for each diagnosis
for (i in coding_self) {
  temp_self_report_copy[,1:87] <- lapply(self_report_copy[,2:88], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  self_report_copy[[paste(i)]] <- rowSums(temp_self_report_copy[,1:87])
}

# print required columns
self_report_copy <- self_report_copy[,c(1,89:ncol(self_report_copy))]

# write table
write.table(self_report_copy, "/file_path/self_report_schizophrenia", row.names=F, col.name=T, quote=F)

# Bipolar disorder

library(CARS)
library(data.table)

# read in data
self_report <- read.table("/file_path/self_report", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
coding_self <- as.data.frame(fread("/file_path/self_report_bipolar.txt", header = F))

# make vector of codes
coding_self <- coding_self[,1]

# make a copy of df of UKB self report 
self_report_copy <- self_report

# make df filled with NA (number of columns = self_report - 1(no ID column))
temp_self_report_copy=as.data.frame(matrix(NA, ncol=87, nrow=nrow(self_report_copy)))

# populate empty df with 0/1 by recoding the self code in each cell, then sum along the rows to create new column for each diagnosis
for (i in coding_self) {
  temp_self_report_copy[,1:87] <- lapply(self_report_copy[,2:88], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  self_report_copy[[paste(i)]] <- rowSums(temp_self_report_copy[,1:87])
}

# print required columns
self_report_copy <- self_report_copy[,c(1,89:ncol(self_report_copy))]

# write table
write.table(self_report_copy, "/file_path/self_report_bpd", row.names=F, col.name=T, quote=F)

# To recode mhq screening columns, got coding for MHQ screening from UKB website and saved as text file: mhq_screening_codes.txt

# Coding	Meaning
# 1	Social anxiety or social phobia
# 2	Schizophrenia
# 3	Any other type of psychosis or psychotic illness
# 4	A personality disorder
# 5	Any other phobia 
# 6	Panic attacks
# 7	Obsessive compulsive disorder 
# 10	Mania hypomania bipolar or manic-depression
# 11	Depression
# 12	Bulimia nervosa
# 13	Psychological over-eating or binge-eating
# 14	Autism, Aspergers or autistic spectrum disorder
# 15	Anxiety nerves or generalized anxiety disorder
# 16	Anorexia nervosa
# 17	Agoraphobia
# 18	Attention deficit or attention deficit and hyperactivity disorder

library(CARS)
library(data.table)

# read in data
self_report <- read.table("/file_path/mhq_screening", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
coding_self <- as.data.frame(fread("/file_path/mhq_screening_codes.txt", header = F))

# make vector of codes
coding_self <- coding_self[,1]

# make a copy of df of UKB self report 
self_report_copy <- self_report

# make df filled with NA (number of columns = self_report - 1(no ID column))
temp_self_report_copy=as.data.frame(matrix(NA, ncol=16, nrow=nrow(self_report_copy)))

# populate empty df with 0/1 by recoding the self code in each cell, then sum along the rows to create new column for each diagnosis
for (i in coding_self) {
  temp_self_report_copy[,1:16] <- lapply(self_report_copy[,2:17], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  self_report_copy[[paste(i)]] <- rowSums(temp_self_report_copy[,1:16])
}

# print required columns
self_report_copy <- self_report_copy[,c(1,18:ncol(self_report_copy))]

# write table
write.table(self_report_copy, "/file_path/mhq_screening_recoded", row.names=F, col.name=T, quote=F)

# To recode medication columns, used drug lists defined for depression and schizophrenia by KCL group: "antidepressants.txt" and "antipsychotics.txt". 

# Antidepressants

library(CARS)
library(data.table)

# read in data
medications <- read.table("/file_path/medications", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
antidepressant_codes <- as.data.frame(fread("/file_path/antidepressants.txt", header = F))

# make vector of codes
antidepressant_codes <- antidepressant_codes[,1]

# make a copy of df of UKB medications
medications_copy <- medications

# make df filled with NA (number of columns = medications - 1(no ID column))
temp_medications_copy=as.data.frame(matrix(NA, ncol=144, nrow=nrow(medications_copy)))

# populate empty df with 0/1 by recoding the med code in eachcell, then sum along the rows to create new column for each med
for (i in antidepressant_codes) {
  temp_medications_copy[,1:144] <- lapply(medications_copy[,2:145], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  medications_copy[[paste(i)]] <- rowSums(temp_medications_copy[,1:144])
}

# print required columns
medications_copy <- medications_copy[,c(1,146:ncol(medications_copy))]

# write tables
write.table(medications_copy, "/file_path/meds_antidepressants", row.names=F, col.name=T, quote=F)

# Antipsychotic

library(CARS)
library(data.table)

# read in data
medications <- read.table("/file_path/medications", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
antipsychotic_codes <- as.data.frame(fread("/file_path/antipsychotics.txt", header = F))

# make vector of codes
antipsychotic_codes <- antipsychotic_codes[,1]

# make a copy of df of UKB medications
medications_copy <- medications

# make df filled with NA (number of columns = medications - 1(no ID column))
temp_medications_copy=as.data.frame(matrix(NA, ncol=144, nrow=nrow(medications_copy)))

# populate empty df with 0/1 by recoding the med code in eachcell, then sum along the rows to create new column for each med
for (i in antipsychotic_codes) {
  temp_medications_copy[,1:144] <- lapply(medications_copy[,2:145], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  medications_copy[[paste(i)]] <- rowSums(temp_medications_copy[,1:144])
}

# print required columns
medications_copy <- medications_copy[,c(1,146:ncol(medications_copy))]

# write tables
write.table(medications_copy, "/file_path/meds_antipsychotics", row.names=F, col.name=T, quote=F)

```

```{bash extract_hes_data_portal_psychiatric, echo=FALSE, eval=FALSE}

# extracted ICD-10 codes relevant to depression and psychosis to build sql scripts to extract primary and secondary diagnoses.
# submitted SQL scripts to extract ICD data from HES portal

# Primary

SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'F20'
OR diag_icd10 = 'F200'
OR diag_icd10 = 'F201'
OR diag_icd10 = 'F202'
OR diag_icd10 = 'F203'
OR diag_icd10 = 'F204'
OR diag_icd10 = 'F205'
OR diag_icd10 = 'F206'
OR diag_icd10 = 'F208'
OR diag_icd10 = 'F209'
OR diag_icd10 = 'F21'
OR diag_icd10 = 'F22'
OR diag_icd10 = 'F220'
OR diag_icd10 = 'F228'
OR diag_icd10 = 'F229'
OR diag_icd10 = 'F23'
OR diag_icd10 = 'F230'
OR diag_icd10 = 'F231'
OR diag_icd10 = 'F232'
OR diag_icd10 = 'F233'
OR diag_icd10 = 'F238'
OR diag_icd10 = 'F239'
OR diag_icd10 = 'F24'
OR diag_icd10 = 'F25'
OR diag_icd10 = 'F250'
OR diag_icd10 = 'F251'
OR diag_icd10 = 'F252'
OR diag_icd10 = 'F258'
OR diag_icd10 = 'F259'
OR diag_icd10 = 'F28'
OR diag_icd10 = 'F29'
OR diag_icd10 = 'F30'
OR diag_icd10 = 'F300'
OR diag_icd10 = 'F301'
OR diag_icd10 = 'F302'
OR diag_icd10 = 'F308'
OR diag_icd10 = 'F309'
OR diag_icd10 = 'F31'
OR diag_icd10 = 'F310'
OR diag_icd10 = 'F311'
OR diag_icd10 = 'F312'
OR diag_icd10 = 'F313'
OR diag_icd10 = 'F314'
OR diag_icd10 = 'F315'
OR diag_icd10 = 'F316'
OR diag_icd10 = 'F317'
OR diag_icd10 = 'F318'
OR diag_icd10 = 'F319'
OR diag_icd10 = 'F32'
OR diag_icd10 = 'F320'
OR diag_icd10 = 'F321'
OR diag_icd10 = 'F322'
OR diag_icd10 = 'F323'
OR diag_icd10 = 'F328'
OR diag_icd10 = 'F329'
OR diag_icd10 = 'F33'
OR diag_icd10 = 'F330'
OR diag_icd10 = 'F331'
OR diag_icd10 = 'F332'
OR diag_icd10 = 'F333'
OR diag_icd10 = 'F334'
OR diag_icd10 = 'F338'
OR diag_icd10 = 'F339'
OR diag_icd10 = 'F34'
OR diag_icd10 = 'F340'
OR diag_icd10 = 'F341'
OR diag_icd10 = 'F348'
OR diag_icd10 = 'F349'
OR diag_icd10 = 'F38'
OR diag_icd10 = 'F380'
OR diag_icd10 = 'F381'
OR diag_icd10 = 'F388'
OR diag_icd10 = 'F39'
OR diag_icd10 = 'F40'
OR diag_icd10 = 'F400'
OR diag_icd10 = 'F401'
OR diag_icd10 = 'F402'
OR diag_icd10 = 'F408'
OR diag_icd10 = 'F409'
OR diag_icd10 = 'F41'
OR diag_icd10 = 'F410'
OR diag_icd10 = 'F411'
OR diag_icd10 = 'F412'
OR diag_icd10 = 'F413'
OR diag_icd10 = 'F418'
OR diag_icd10 = 'F419'
OR diag_icd10 = 'F42'
OR diag_icd10 = 'F420'
OR diag_icd10 = 'F421'
OR diag_icd10 = 'F422'
OR diag_icd10 = 'F428'
OR diag_icd10 = 'F429'
OR diag_icd10 = 'F43'
OR diag_icd10 = 'F430'
OR diag_icd10 = 'F431'
OR diag_icd10 = 'F432'
OR diag_icd10 = 'F438'
OR diag_icd10 = 'F439'
OR diag_icd10 = 'F44'
OR diag_icd10 = 'F440'
OR diag_icd10 = 'F441'
OR diag_icd10 = 'F442'
OR diag_icd10 = 'F443'
OR diag_icd10 = 'F444'
OR diag_icd10 = 'F445'
OR diag_icd10 = 'F446'
OR diag_icd10 = 'F447'
OR diag_icd10 = 'F448'
OR diag_icd10 = 'F449'
OR diag_icd10 = 'F45'
OR diag_icd10 = 'F450'
OR diag_icd10 = 'F451'
OR diag_icd10 = 'F452'
OR diag_icd10 = 'F453'
OR diag_icd10 = 'F454'
OR diag_icd10 = 'F458'
OR diag_icd10 = 'F459'
OR diag_icd10 = 'F48'
OR diag_icd10 = 'F480'
OR diag_icd10 = 'F481'
OR diag_icd10 = 'F488'
OR diag_icd10 = 'F489'

# Secondary

SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'F20'
OR diag_icd10 = 'F200'
OR diag_icd10 = 'F201'
OR diag_icd10 = 'F202'
OR diag_icd10 = 'F203'
OR diag_icd10 = 'F204'
OR diag_icd10 = 'F205'
OR diag_icd10 = 'F206'
OR diag_icd10 = 'F208'
OR diag_icd10 = 'F209'
OR diag_icd10 = 'F21'
OR diag_icd10 = 'F22'
OR diag_icd10 = 'F220'
OR diag_icd10 = 'F228'
OR diag_icd10 = 'F229'
OR diag_icd10 = 'F23'
OR diag_icd10 = 'F230'
OR diag_icd10 = 'F231'
OR diag_icd10 = 'F232'
OR diag_icd10 = 'F233'
OR diag_icd10 = 'F238'
OR diag_icd10 = 'F239'
OR diag_icd10 = 'F24'
OR diag_icd10 = 'F25'
OR diag_icd10 = 'F250'
OR diag_icd10 = 'F251'
OR diag_icd10 = 'F252'
OR diag_icd10 = 'F258'
OR diag_icd10 = 'F259'
OR diag_icd10 = 'F28'
OR diag_icd10 = 'F29'
OR diag_icd10 = 'F30'
OR diag_icd10 = 'F300'
OR diag_icd10 = 'F301'
OR diag_icd10 = 'F302'
OR diag_icd10 = 'F308'
OR diag_icd10 = 'F309'
OR diag_icd10 = 'F31'
OR diag_icd10 = 'F310'
OR diag_icd10 = 'F311'
OR diag_icd10 = 'F312'
OR diag_icd10 = 'F313'
OR diag_icd10 = 'F314'
OR diag_icd10 = 'F315'
OR diag_icd10 = 'F316'
OR diag_icd10 = 'F317'
OR diag_icd10 = 'F318'
OR diag_icd10 = 'F319'
OR diag_icd10 = 'F32'
OR diag_icd10 = 'F320'
OR diag_icd10 = 'F321'
OR diag_icd10 = 'F322'
OR diag_icd10 = 'F323'
OR diag_icd10 = 'F328'
OR diag_icd10 = 'F329'
OR diag_icd10 = 'F33'
OR diag_icd10 = 'F330'
OR diag_icd10 = 'F331'
OR diag_icd10 = 'F332'
OR diag_icd10 = 'F333'
OR diag_icd10 = 'F334'
OR diag_icd10 = 'F338'
OR diag_icd10 = 'F339'
OR diag_icd10 = 'F34'
OR diag_icd10 = 'F340'
OR diag_icd10 = 'F341'
OR diag_icd10 = 'F348'
OR diag_icd10 = 'F349'
OR diag_icd10 = 'F38'
OR diag_icd10 = 'F380'
OR diag_icd10 = 'F381'
OR diag_icd10 = 'F388'
OR diag_icd10 = 'F39'
OR diag_icd10 = 'F40'
OR diag_icd10 = 'F400'
OR diag_icd10 = 'F401'
OR diag_icd10 = 'F402'
OR diag_icd10 = 'F408'
OR diag_icd10 = 'F409'
OR diag_icd10 = 'F41'
OR diag_icd10 = 'F410'
OR diag_icd10 = 'F411'
OR diag_icd10 = 'F412'
OR diag_icd10 = 'F413'
OR diag_icd10 = 'F418'
OR diag_icd10 = 'F419'
OR diag_icd10 = 'F42'
OR diag_icd10 = 'F420'
OR diag_icd10 = 'F421'
OR diag_icd10 = 'F422'
OR diag_icd10 = 'F428'
OR diag_icd10 = 'F429'
OR diag_icd10 = 'F43'
OR diag_icd10 = 'F430'
OR diag_icd10 = 'F431'
OR diag_icd10 = 'F432'
OR diag_icd10 = 'F438'
OR diag_icd10 = 'F439'
OR diag_icd10 = 'F44'
OR diag_icd10 = 'F440'
OR diag_icd10 = 'F441'
OR diag_icd10 = 'F442'
OR diag_icd10 = 'F443'
OR diag_icd10 = 'F444'
OR diag_icd10 = 'F445'
OR diag_icd10 = 'F446'
OR diag_icd10 = 'F447'
OR diag_icd10 = 'F448'
OR diag_icd10 = 'F449'
OR diag_icd10 = 'F45'
OR diag_icd10 = 'F450'
OR diag_icd10 = 'F451'
OR diag_icd10 = 'F452'
OR diag_icd10 = 'F453'
OR diag_icd10 = 'F454'
OR diag_icd10 = 'F458'
OR diag_icd10 = 'F459'
OR diag_icd10 = 'F48'
OR diag_icd10 = 'F480'
OR diag_icd10 = 'F481'
OR diag_icd10 = 'F488'
OR diag_icd10 = 'F489') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

```


```{r download_coding_icd_self_medication, include=FALSE}

# downloaded ICD coding from UKB website
# Description: ICD10 - WHO International Classification of Diseases = "coding10.tsv"
coding_icd10 <- as.data.frame(fread("/file_path/coding19.tsv", header = T))
coding_icd10 <- coding_icd10[,1:2]

# self-report UKB codes
coding_dep <- as.data.frame(fread("/file_path/self_report_depression.txt", header = F))
coding_scz <- as.data.frame(fread("/file_path/self_report_scz.txt", header = F))
coding_bpd <- as.data.frame(fread("/file_path/self_report_bipolar.txt", header = F))

# read mhq screening coding
coding_mhq <- as.data.frame(fread("/file_path/mhq_screening_codes.txt", header = T))

# downloaded medication coding from UKB website "coding4.tsv"
coding_meds <- as.data.frame(fread("/file_path/coding4.tsv", header = T))

antidepressant_codes <- as.data.frame(fread("/file_path/antidepressants.txt", header = F))
colnames(antidepressant_codes) <- c("coding", "preparation","class")

antipsychotic_codes <- as.data.frame(fread("/file_path/antipsychotics.txt", header = F))
colnames(antipsychotic_codes) <- c("coding", "preparation","class")

```

```{r download_phenotypes_qc_ids_hes_self-report_touchscreen_mhq_medication_demographics_smith, include=FALSE} 

# read in participants IDs passing genetic QC
qc_ids <- read.table("/file_path/post_qc_id_list.fam", header = F) 
qc_ids <- qc_ids %>% 
  mutate(ID = coalesce(as.integer(V1)))

# read in HES output 
psych_pri <- read_tsv("/file_path/psychiatric_pri_hes.tsv")
psych_sec <- read_tsv("/file_path/psychiatric_sec_hes.tsv")

# read in self-report
self_report_dep <- fread("/file_path/self_report_dep", header = T) 
self_report_scz <- fread("/file_path/self_report_schizophrenia", header = T) 
self_report_bpd <- fread("/file_path/self_report_bpd", header = T) 

# read in touchscreen 
touchscreen <- fread("/file_path/touchscreen", header = T) 

# read in mhq 
mhq <- fread("/file_path/ever_depressed_pheno_final.txt", header = T) 

# read in mhq screening
mhq_screen <- fread("/file_path/mhq_screening_recoded", header = T) 

# read in mhq participants/completers
mhq_participants <- fread("/file_path/mhq_participants", header = T) 

# read in medication data:
meds_antidepressants <- as.data.frame(fread("/file_path/meds_antidepressants", header = T))
meds_antipsychotics <- as.data.frame(fread("/file_path/meds_antipsychotics", header = T))

# read in demographics
demographics <- as.data.frame(fread("/file_path/demographics", header = T))

# read in D.Smith phenotypes
smith <- as.data.frame(fread("/file_path/smith", header = T))

# read in smith participants/completers
smith_participants <- fread("/file_path/smith_participants", header = T) 

```


```{r format_hes_data_into_columns, include=FALSE}

# QC HES data
# Step 1: remove duplicated lines. Code removes duplicate lines where all cells are identical 
# Step 2: group by participant ID and remove lines where the admission date is repeated for the same ICD code. This will retain lines where >1 primary dx has been assigned under the same admission. 
# create QC function for steps 1 and 2 and run over dataframes for each disease, first for primary then secondary

qc_hes_pri <- function(x) {
  dplyr::distinct(x, .keep_all = TRUE) %>% 
  group_by(eid) %>% 
  dplyr::distinct(admidate, primary_diag, .keep_all = TRUE)}
  
psych_pri_qc1 <- qc_hes_pri(psych_pri)

qc_hes_sec <- function(x) {
  dplyr::distinct(x, .keep_all = TRUE) %>% 
  group_by(eid) %>% 
  dplyr::distinct(admidate, secondary_diag, .keep_all = TRUE)}

psych_sec_qc1 <- qc_hes_sec(psych_sec)

# create function to put hes data into column format and loop over dataframes. 

column_format_pri <- function(x) {x %>% 
  group_by(eid) %>%
  count(primary_diag) %>%
  spread(key=primary_diag,value=n) %>%
  replace(., is.na(.), 0)
}

psych_pri_format<-column_format_pri(psych_pri_qc1)

column_format_sec <- function(x) {x %>% 
  group_by(eid) %>%
  count(secondary_diag) %>%
  spread(key=secondary_diag,value=n) %>%
  replace(., is.na(.), 0)
}

psych_sec_format<-column_format_sec(psych_sec_qc1)

# merge with all UKB IDs so cases can be checked against website
# make fam file first

fam <- left_join(self_report_dep, qc_ids, by = c("f.eid" = "ID"))
fam <- fam[,-2]
colnames(fam)[1] <- "eid"
                 
merge_fam <- function(x) {
  left_join(fam, x, by = "eid") %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)
}

psych_pri_format_qc2 <- merge_fam(psych_pri_format)
psych_sec_format_qc2 <- merge_fam(psych_sec_format)

# make df for different psychiatric disorders

# 1) for screening

# F20-F29 Schizophrenia, schizotypal and delusional disorders
scz_pri <- psych_pri_format_qc2[,c(1:29)]
scz_sec <- psych_sec_format_qc2[,c(1:28)]

# Mood [affective] disorders F30, F31, F34, F38, F39
mood_pri <- psych_pri_format_qc2[,c(1,30:44,59:64)]
mood_sec <- psych_sec_format_qc2[,c(1,29:43,57:64)]

# 2) for depression 

# Mood [affective] disorders - F32-F33
dep_pri <- psych_pri_format_qc2[,c(1,45:58)]
dep_sec <- psych_sec_format_qc2[,c(1,44:56)]

```

```{r hes_count_pp_and_summarise, include=FALSE}

# Count per participant

# make function to count primary and secondary dx for each individual
count_dx_pp <- function(x,y) {
  data.frame("X1" = x[,1], "X2" = rowSums(x[,2:ncol(x)]), "X3"= rowSums(y[,2:ncol(y)])) %>% 
  rename("ID" = X1, "Primary count" = X2, "Secondary count" = X3)  
}

# run function across primary and secondary columns to get count per person
scz_count_pp <- count_dx_pp(scz_pri,scz_sec)
colnames(scz_count_pp) <- c("ID", "Primary_scz", "Secondary_scz")

mood_count_pp <- count_dx_pp(mood_pri,mood_sec)
colnames(mood_count_pp) <- c("ID", "Primary_bipolar", "Secondary_bipolar")

dep_count_pp <- count_dx_pp(dep_pri,dep_sec)
colnames(dep_count_pp) <- c("ID", "Primary_depression", "Secondary_depression")

# Count across participants

# make function to count dx across people
count_dx <- function(x) {
  data.frame("x1" = names(x[,-1]), "x2" = colSums(x[,-1] > 0), stringsAsFactors = FALSE, row.names = NULL)
}

# make function to merge ICD codes with the disorder name of each code
merge_dx <- function(x) {
  merge(coding_icd10, x, by.x = "coding", by.y = "x1")
}

# create primary and secondary tables with count across participants
scz_dx_pri <- count_dx(scz_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)
mood_dx_pri <- count_dx(mood_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)
# stress_dx_pri <- count_dx(stress_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)
dep_dx_pri <- count_dx(dep_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)

scz_dx_sec <- count_dx(scz_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)
mood_dx_sec <- count_dx(mood_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)
# stress_dx_sec <- count_dx(stress_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)
dep_dx_sec <- count_dx(dep_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)

# make function to merge primary and secondary columns
merge_pri_sec <- function(x,y) {
  merge(x, y, by = "ICD-10 sub-codes", all = TRUE) %>% 
  replace(., is.na(.), 0)
}

# merge primary and secondary using function

# F20-F29 Schizophrenia, schizotypal and delusional disorders
summary_scz_icd <- merge_pri_sec(scz_dx_pri,scz_dx_sec)

# Mood [affective] disorders F30, F31, F34, F38, F39
summary_mood_icd <- merge_pri_sec(mood_dx_pri,mood_dx_sec)

# Mood [affective] disorders - F32-F33
summary_dep_icd <- merge_pri_sec(dep_dx_pri,dep_dx_sec)

# Merge 
summary_screen_icd <- rbind(summary_scz_icd,summary_mood_icd)

```

```{r self_report_summarise, include=FALSE}

# add column names to count per person
colnames(self_report_dep) <- c("ID",coding_dep$V2)
colnames(self_report_scz) <- c("ID",coding_scz$V2)
colnames(self_report_bpd) <- c("ID",coding_bpd$V2)

# make summary tables
summary_self_report_dep <- data.frame("x1" = names(self_report_dep)[2], "x2" = as.numeric(format(sum(self_report_dep[,2]>0))), stringsAsFactors = F, row.names = NULL) %>% rename("temp" = x1, "Number" = x2) 
summary_self_report_dep$'Self-reported disorder' <- capitalize(summary_self_report_dep$temp)
summary_self_report_dep <- summary_self_report_dep[,c(3,2)]

summary_self_report_scz <- data.frame("x1" = names(self_report_scz)[2], "x2" = as.numeric(format(sum(self_report_scz[,2]>0))), stringsAsFactors = F, row.names = NULL) %>% rename("temp" = x1, "Number" = x2) 
summary_self_report_scz$'Self-reported disorder' <- capitalize(summary_self_report_scz$temp)
summary_self_report_scz <- summary_self_report_scz[,c(3,2)]

summary_self_report_bpd <- data.frame("x1" = names(self_report_bpd)[2], "x2" = as.numeric(format(sum(self_report_bpd[,2]>0))), stringsAsFactors = F, row.names = NULL) %>% rename("temp" = x1, "Number" = x2) 
summary_self_report_bpd$'Self-reported disorder' <- capitalize(summary_self_report_bpd$temp)
summary_self_report_bpd <- summary_self_report_bpd[,c(3,2)]

summary_screen_self_report <- rbind(summary_self_report_scz,summary_self_report_bpd)
rownames(summary_screen_self_report) <- NULL

# Rename columns for self-report count per person
colnames(self_report_dep) <- c("ID","SR_depression")
colnames(self_report_scz) <- c("ID","SR_scz")
colnames(self_report_bpd) <- c("ID","SR_bipolar")

table(self_report_dep$SR_depression)

```

```{r medications_count_pp_and_summarise, include=FALSE}

# change medication code to medication name in columns
(names( meds_antidepressants ) <- coding_meds[ match( 
names( meds_antidepressants ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_antidepressants) [1]  <- "f.eid"

(names( meds_antipsychotics ) <- coding_meds[ match( 
names( meds_antipsychotics ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_antipsychotics) [1]  <- "f.eid"

# count meds per participant (collapse medication classes into one column)
count_meds_pp <- function(x) {
  data.frame("X1" = x[,1], "X2" = rowSums(x[,2:ncol(x)]))
}

meds_antidepressants_count_pp <- count_meds_pp(meds_antidepressants)
names(meds_antidepressants_count_pp) <- c("ID","Antidepressant")

meds_antipsychotics_count_pp <- count_meds_pp(meds_antipsychotics)
names(meds_antipsychotics_count_pp) <- c("ID","Antipsychotic")

# make summary tables
summary_meds_antidepressants <- data.frame("x1" = names(meds_antidepressants[,-1]), "x2" = as.integer(format(colSums(meds_antidepressants[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Medication = x1, Number = x2)

summary_meds_antipsychotics <- data.frame("x1" = names(meds_antipsychotics[,-1]), "x2" = as.integer(format(colSums(meds_antipsychotics[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Medication = x1, Count = x2)

```

```{r mhq_copy_of_script_for_deriving_cases, echo=FALSE, eval=FALSE}

The code used to define MHQ depression cases and controls is copied below for reference. This code was used to define "lifetime depression" in our study "Classical Human Leukocyte Antigen Alleles and C4 Haplotypes Are Not Significantly Associated With Depression" https://www.biologicalpsychiatryjournal.com/article/S0006-3223(19)31558-6/abstract). Origial code written by JRIColeman in Feb 2018 and edited by KPGlanville in March 2018.

The following shell script was submitted: 
###################################################################################################################

module add general/R/3.4.1
Rscript kg_edit_march2018_Depressed_Ever_JRIC_160218.R icd10_treatment_srinterview_Depressed_Ever_Phenotype_File.txt

###################################################################################################################

icd10_treatment_srinterview_Depressed_Ever_Phenotype_File.txt is a text file containing phenotypic columns, i.e.: 
###################################################################################################################
# [1] "ID"                                                                                     
# [2] "Diagnoses.main.ICD10.0.0" to  [381] "Diagnoses.main.ICD10.0.379"                                                              
#  [382] "Non.cancer.illness.code.self.reported.0.0" - [410] "Non.cancer.illness.code.self.reported.0.28"                          
#  [411] "Treatment.medication.code.0.0" to [458] "Treatment.medication.code.0.47"                                                    
#  [459] "Bipolar.and.major.depression.status.0.0"                                                
#  [460] "Date.of.completing.mental.health.questionnaire"                                         
#  [461] "Ever.addicted.to.any.substance.or.behaviour"                                            
#  [462] "Amount.of.alcohol.drunk.on.a.typical.drinking.day"                                      
#  [463] "Ever.physically.dependent.on.alcohol"                                                   
#  [464] "Ever.had.known.person.concerned.about.or.recommend.reduction.of.alcohol.consumption"    
#  [465] "Ever.addicted.to.alcohol"                                                               
#  [466] "Frequency.of.failure.to.fulfil.normal.expectations.due.to.drinking.alcohol.in.last.year"
#  [467] "Frequency.of.memory.loss.due.to.drinking.alcohol.in.last.year"                          
#  [468] "Frequency.of.feeling.guilt.or.remorse.after.drinking.alcohol.in.last.year"              
#  [469] "Age.when.known.person.last.commented.about.drinking.habits"                             
#  [470] "Ever.been.injured.or.injured.someone.else.through.drinking.alcohol"                     
#  [471] "Frequency.of.needing.morning.drink.of.alcohol.after.heavy.drinking.session.in.last.year"
#  [472] "Frequency.of.inability.to.cease.drinking.in.last.year"                                  
#  [473] "Frequency.of.drinking.alcohol"                                                          
#  [474] "Ongoing.addiction.to.alcohol"                                                           
#  [475] "Frequency.of.consuming.six.or.more.units.of.alcohol"                                    
#  [476] "Tense.sore.or.aching.muscles.during.worst.period.of.anxiety"                            
#  [477] "Impact.on.normal.roles.during.worst.period.of.anxiety"                                  
#  [478] "Difficulty.concentrating.during.worst.period.of.anxiety"                                
#  [479] "Longest.period.spent.worried.or.anxious"                                                
#  [480] "Ever.felt.worried.tense.or.anxious.for.most.of.a.month.or.longer"                       
#  [481] "More.irritable.than.usual.during.worst.period.of.anxiety"                               
#  [482] "Keyed.up.or.on.edge.during.worst.period.of.anxiety"                                     
#  [483] "Ever.worried.more.than.most.people.would.in.similar.situation"                          
#  [484] "Restless.during.period.of.worst.anxiety"                                                
#  [485] "Frequent.trouble.falling.or.staying.asleep.during.worst.period.of.anxiety"              
#  [486] "Professional.informed.about.anxiety"                                                    
#  [487] "Easily.tired.during.worst.period.of.anxiety"                                            
#  [488] "Ever.addicted.to.a.behaviour.or.miscellanous"                                           
#  [489] "Ongoing.behavioural.or.miscellanous.addiction"                                          
#  [490] "Age.at.first.episode.of.depression"                                                     
#  [491] "Age.at.last.episode.of.depression"                                                      
#  [492] "Difficulty.concentrating.during.worst.depression"                                       
#  [493] "Fraction.of.day.affected.during.worst.episode.of.depression"                            
#  [494] "Thoughts.of.death.during.worst.depression"                                              
#  [495] "Duration.of.worst.depression"                                                           
#  [496] "Frequency.of.depressed.days.during.worst.episode.of.depression"                         
#  [497] "Impact.on.normal.roles.during.worst.period.of.depression"                               
#  [498] "Ever.had.prolonged.loss.of.interest.in.normal.activities"                               
#  [499] "Lifetime.number.of.depressed.periods"                                                   
#  [500] "Depression.possibly.related.to.childbirth"                                              
#  [501] "Ever.had.prolonged.feelings.of.sadness.or.depression"                                   
#  [502] "Depression.possibly.related.to.stressful.or.traumatic.event"                            
#  [503] "Professional.informed.about.depression"                                                 
#  [504] "Feelings.of.tiredness.during.worst.episode.of.depression"                               
#  [505] "Feelings.of.worthlessness.during.worst.period.of.depression"                            
#  [506] "Ever.taken.cannabis"                                                                    
#  [507] "Maximum.frequency.of.taking.cannabis"                                                   
#  [508] "Age.when.last.took.cannabis"                                                            
#  [509] "Ever.addicted.to.illicit.or.recreational.drugs"                                         
#  [510] "Illicit.or.recreational.drugs.addiction.or.dependence.ongoing"                          
#  [511] "General.happiness"                                                                      
#  [512] "General.happiness.with.own.health"                                                      
#  [513] "Belief.that.own.life.is.meaningful"                                                     
#  [514] "Age.when.first.had.unusual.or.psychotic.experience"                                     
#  [515] "Distress.caused.by.unusual.or.psychotic.experiences"                                    
#  [516] "Ever.heard.an.unreal.voice"                                                             
#  [517] "Number.of.times.heard.an.unreal.voice"                                                  
#  [518] "Ever.prescribed.a.medication.for.unusual.or.psychotic.experiences"                      
#  [519] "Frequency.of.unusual.or.psychotic.experiences.in.past.year"                             
#  [520] "Ever.believed.in.an.unreal.conspiracy.against.self"                                     
#  [521] "Number.of.times.believed.in.an.unreal.conspiracy.against.self"                          
#  [522] "Ever.seen.an.unreal.vision"                                                             
#  [523] "Number.of.times.seen.an.unreal.vision"                                                  
#  [524] "Ever.believed.in.unreal.communications.or.signs"                                        
#  [525] "Number.of.times.believed.in.unreal.communications.or.signs"                             
#  [526] "Ever.talked.to.a.health.professional.about.unusual.or.psychotic.experiences"            
#  [527] "Felt.loved.as.a.child"                                                                  
#  [528] "Someone.to.take.to.doctor.when.needed.as.a.child"                                       
#  [529] "Longest.period.of.mania.or.irritability"                                                
#  [530] "Severity.of.problems.due.to.mania.or.irritability"                                      
#  [531] "Felt.irritable.or.had.angry.outbursts.in.past.month"                                    
#  [532] "Avoided.activities.or.situations.because.of.previous.stressful.experience.in.past.month"
#  [533] "Felt.distant.from.other.people.in.past.month"                                           
#  [534] "Repeated.disturbing.thoughts.of.stressful.experience.in.past.month"                     
#  [535] "Felt.very.upset.when.reminded.of.stressful.experience.in.past.month"                    
#  [536] "Ever.sought.or.received.professional.help.for.mental.distress"                          
#  [537] "Ever.suffered.mental.distress.preventing.usual.activities"                              
#  [538] "Ever.had.period.of.mania.excitability"                                                  
#  [539] "Ever.had.period.extreme.irritability"                                                   
#  [540] "Ever.addicted.to.prescription.or.over.the.counter.medication"                           
#  [541] "Prescription.or.over.the.counter.medication.addiction.or.dependence.ongoing"            
#  [542] "Recent.easy.annoyance.or.irritability"                                                  
#  [543] "Recent.feelings.of.nervousness.or.anxiety"                                              
#  [544] "Recent.feelings.of.inadequacy"                                                          
#  [545] "Recent.trouble.concentrating.on.things"                                                 
#  [546] "Recent.inability.to.stop.or.control.worrying"                                           
#  [547] "Recent.feelings.of.depression"                                                          
#  [548] "Recent.poor.appetite.or.overeating"                                                     
#  [549] "Recent.feelings.of.foreboding"                                                          
#  [550] "Recent.thoughts.of.suicide.or.self.harm"                                                
#  [551] "Recent.lack.of.interest.or.pleasure.in.doing.things"                                    
#  [552] "Recent.trouble.relaxing"                                                                
#  [553] "Recent.restlessness"                                                                    
#  [554] "Trouble.falling.or.staying.asleep.or.sleeping.too.much"                                 
#  [555] "Recent.changes.in.speed.amount.of.moving.or.speaking"                                   
#  [556] "Recent.feelings.of.tiredness.or.low.energy"                                             
#  [557] "Recent.worrying.too.much.about.different.things"                                        
#  [558] "Been.in.a.confiding.relationship.as.an.adult"                                           
#  [559] "Able.to.pay.rent.mortgage.as.an.adult"                                                  
#  [560] "Been.in.serious.accident.believed.to.be.life.threatening"                               
#  [561] "Been.involved.in.combat.or.exposed.to.war.zone"                                         
#  [562] "Diagnosed.with.life.threatening.illness"                                                
#  [563] "Victim.of.physically.violent.crime"                                                     
#  [564] "Witnessed.sudden.violent.death"                                                         
#  [565] "Did.your.sleep.change"                                                                  
#  [566] "Trouble.falling.asleep"                                                                 
#  [567] "Sleeping.too.much"                                                                      
#  [568] "Waking.too.early"                                                                       
#  [569] "Weight.change.during.worst.episode.of.depression"                                       
#  [570] "Frequency.of.difficulty.controlling.worry.during.worst.period.of.anxiety"               
#  [571] "Worried.most.days.during.period.of.worst.anxiety"                                       
#  [572] "Frequency.of.inability.to.stop.worrying.during.worst.period.of.anxiety"                 
#  [573] "Multiple.worries.during.worst.period.of.anxiety"                                        
#  [574] "Difficulty.stopping.worrying.during.worst.period.of.anxiety"                            
#  [575] "Stronger.worrying.than.other.people.during.period.of.worst.anxiety"                     
#  [576] "Number.of.things.worried.about.during.worst.period.of.anxiety"                          
#  [577] "Mental.health.problems.ever.diagnosed.by.a.professional.1"                              
#  [578] "Mental.health.problems.ever.diagnosed.by.a.professional.2"                              
#  [579] "Mental.health.problems.ever.diagnosed.by.a.professional.3"                              
#  [580] "Mental.health.problems.ever.diagnosed.by.a.professional.4"                              
#  [581] "Mental.health.problems.ever.diagnosed.by.a.professional.5"                              
#  [582] "Mental.health.problems.ever.diagnosed.by.a.professional.6"                              
#  [583] "Mental.health.problems.ever.diagnosed.by.a.professional.7"                              
#  [584] "Mental.health.problems.ever.diagnosed.by.a.professional.8"                              
#  [585] "Mental.health.problems.ever.diagnosed.by.a.professional.9"                              
#  [586] "Mental.health.problems.ever.diagnosed.by.a.professional.10"                             
#  [587] "Mental.health.problems.ever.diagnosed.by.a.professional.11"                             
#  [588] "Mental.health.problems.ever.diagnosed.by.a.professional.12"                             
#  [589] "Mental.health.problems.ever.diagnosed.by.a.professional.13"                             
#  [590] "Mental.health.problems.ever.diagnosed.by.a.professional.14"                             
#  [591] "Mental.health.problems.ever.diagnosed.by.a.professional.15"                             
#  [592] "Mental.health.problems.ever.diagnosed.by.a.professional.16"                             
#  [593] "Substances.taken.for.depression.1"                                                      
#  [594] "Substances.taken.for.depression.2"                                                      
#  [595] "Substances.taken.for.depression.3"                                                      
#  [596] "Activities.undertaken.to.treat.depression.1"                                            
#  [597] "Activities.undertaken.to.treat.depression.2"                                            
#  [598] "Manifestations.of.mania.or.irritability.1"                                              
#  [599] "Manifestations.of.mania.or.irritability.2"                                              
#  [600] "Manifestations.of.mania.or.irritability.3"                                              
#  [601] "Manifestations.of.mania.or.irritability.4"                                              
#  [602] "Manifestations.of.mania.or.irritability.5"                                              
#  [603] "Manifestations.of.mania.or.irritability.6"                                              
#  [604] "Manifestations.of.mania.or.irritability.7"                                              
#  [605] "Manifestations.of.mania.or.irritability.8"                                              
#  [606] "Substances.taken.for.anxiety.1"                                                         
#  [607] "Substances.taken.for.anxiety.2"                                                         
#  [608] "Substances.taken.for.anxiety.3"                                                         
#  [609] "Activities.undertaken.to.treat.anxiety.1"                                               
#  [610] "Activities.undertaken.to.treat.anxiety.2"                                               
#  [611] "Medical.substance.addictions.1"                                                         
#  [612] "Medical.substance.addictions.2"                                                         
#  [613] "Medical.substance.addictions.3"                                                         
#  [614] "Behavioural.and.miscellaneous.addictions.1"                                             
#  [615] "Behavioural.and.miscellaneous.addictions.2"                                             
#  [616] "Non.cancer.illness.code.self.reported.1.0" to  [673] "Non.cancer.illness.code.self.reported.2.28"                           
#  [674] "Treatment.medication.code.1.0" to  [769] "Treatment.medication.code.2.47"                                                   
#  [770] "Diagnoses.secondary.ICD10.0.0" to [1204] "Diagnoses.secondary.ICD10.0.434"

###################################################################################################################
###################################################################################################################

kg_edit_march2018_Depressed_Ever_JRIC_160218.R is the Rscript for deriving cases and controls from the phenotypic columns:  
###################################################################################################################

###Check command line arguments are present

args<-commandArgs(trailingOnly=TRUE)

if(length(args)==0){
    stop("Please provide an input file containing all fields listed in the script (full path)", call.=FALSE)
}

### Load dependencies

library(data.table)
library(Hmisc)
library(stringi)
library(dplyr)
library(fmsb)

### Load data 

MHQ_Full<-fread(args[1], data.table=F)

#####Create MHQ dataset with only respondants

MHQ<-MHQ_Full[!is.na(MHQ_Full$Ever.sought.or.received.professional.help.for.mental.distress), ]
rm(MHQ_Full)

#####Physician Diagnosed Disorders

MHQ$SRSocPhobia<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
  			   ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 1) |
		                  (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 1), 1, 0)))

MHQ$SRSchizophrenia<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			       ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 2) |
		                      (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 2), 1, 0)))

MHQ$SRPsychosisOther<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			        ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 3) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 3), 1, 0)))


MHQ$SRPsychosisAny<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
					 ifelse((!is.na(SRSchizophrenia) & SRSchizophrenia == 1) | (!is.na(SRPsychosisOther) & SRPsychosisOther == 1), 1, 0)))


MHQ$SRPersonalityDisorder<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
				     ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 4) |
		                            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 4) |
		      		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 4), 1, 0)))

MHQ$SROtherPhobia<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			     ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 5) |
		    	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 5) |
		      	            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 5), 1, 0)))

MHQ$SRPanicAttacks<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			      ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 6) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 6), 1, 0)))

MHQ$SROCD<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
		     ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 7) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 7), 1, 0)))

MHQ$SRManiaBIP<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			  ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 10) |
		                 (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 10), 1, 0)))

MHQ$SRDepression<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			    ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 11) |
		                   (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 11), 1, 0)))

MHQ$SRMood<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
		      ifelse(!is.na(SRManiaBIP) & SRManiaBIP == 1 |
		      	     !is.na(SRDepression) & SRDepression == 1, 1, 0)))

MHQ$SRBulimiaNervosa<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			        ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 12) |
		                       (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 12), 1, 0)))

MHQ$SRBingeEating<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			     ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 13) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 13), 1, 0)))

MHQ$SRASD<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
		     ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 14) |
		            (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 14), 1, 0)))

MHQ$SRGADandOthers<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			      ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 15) |
		                     (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 15), 1, 0)))

MHQ$SRAnorexiaNervosa<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			         ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 16) |
		                        (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 16), 1, 0)))

MHQ$SREatingDisorderAny<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
				   ifelse((!is.na(SRAnorexiaNervosa) & SRAnorexiaNervosa == 1) |
			         	  (!is.na(SRBulimiaNervosa) & SRBulimiaNervosa == 1) |
			         	  (!is.na(SRBingeEating) & SRBingeEating == 1), 1, 0)))

MHQ$SRAgoraphobia<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			     ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 17) |
		                    (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 17), 1, 0)))

MHQ$SRAnxietyAny<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
			    ifelse((!is.na(SRSocPhobia) & SRSocPhobia == 1) |
			           (!is.na(SRGADandOthers) & SRGADandOthers == 1) |
			           (!is.na(SRPanicAttacks) & SRPanicAttacks == 1) |
			           (!is.na(SRAgoraphobia) & SRAgoraphobia == 1) |
			           (!is.na(SROtherPhobia) & SROtherPhobia == 1) |
			           (!is.na(SROCD) & SROCD == 1), 1, 0)))

MHQ$SRADHD<-with(MHQ, ifelse(is.na(Ever.sought.or.received.professional.help.for.mental.distress), NA,
		      ifelse((!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.1) & Mental.health.problems.ever.diagnosed.by.a.professional.1 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.2) & Mental.health.problems.ever.diagnosed.by.a.professional.2 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.3) & Mental.health.problems.ever.diagnosed.by.a.professional.3 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.4) & Mental.health.problems.ever.diagnosed.by.a.professional.4 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.5) & Mental.health.problems.ever.diagnosed.by.a.professional.5 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.6) & Mental.health.problems.ever.diagnosed.by.a.professional.6 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.7) & Mental.health.problems.ever.diagnosed.by.a.professional.7 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.8) & Mental.health.problems.ever.diagnosed.by.a.professional.8 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.9) & Mental.health.problems.ever.diagnosed.by.a.professional.9 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.10) & Mental.health.problems.ever.diagnosed.by.a.professional.10 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.11) & Mental.health.problems.ever.diagnosed.by.a.professional.11 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.12) & Mental.health.problems.ever.diagnosed.by.a.professional.12 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.13) & Mental.health.problems.ever.diagnosed.by.a.professional.13 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.14) & Mental.health.problems.ever.diagnosed.by.a.professional.14 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.15) & Mental.health.problems.ever.diagnosed.by.a.professional.15 == 18) |
		             (!is.na(Mental.health.problems.ever.diagnosed.by.a.professional.16) & Mental.health.problems.ever.diagnosed.by.a.professional.16 == 18), 1, 0)))

MHQ$SRConditionsSum<-0
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRSchizophrenia == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRPsychosisOther == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRDepression == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRManiaBIP == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRGADandOthers == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRPanicAttacks == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRAgoraphobia == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRSocPhobia == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SROtherPhobia == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SROCD == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRPersonalityDisorder == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRAnorexiaNervosa == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRBulimiaNervosa == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRBingeEating == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRASD == 1, SRConditionsSum + 1, SRConditionsSum))
MHQ$SRConditionsSum<-with(MHQ, ifelse(SRADHD == 1, SRConditionsSum + 1, SRConditionsSum))

MHQ$NoSRConditions<-with(MHQ,ifelse(SRConditionsSum == 0, 1, 0))

MHQ$MultipleSRConditions<-with(MHQ,ifelse(SRConditionsSum > 1, 1, 0))

#####Interview Diagnoses At Baseline######

InterviewDepression<-apply(MHQ[,grep("Non.cancer.illness.code.self.reported", colnames(MHQ))], 1, function(row) "1286" %in% row)

MHQ$InterviewDepression[c(InterviewDepression)]<-1

####HES Inpatient Diagnoses######

ICDFields<-c("F300","F308","F311","F314","F317","F320","F323","F32","F332","F338","F341","F380","F39","F301","F309","F312","F315","F318","F321","F328","F330","F333","F339","F348","F381","F302","F310","F313","F316","F319","F322","F329","F331","F334","F340","F349","F388")

HESMood<-apply(MHQ[,grep("ICD10", colnames(MHQ))], 1, function(row) ICDFields %in% row)

HESMood_Combined<-numeric()

for(indiv in 1:dim(MHQ)[1]){
    HESMood_Combined[indiv]<-ifelse(sum(HESMood[(((indiv-1)*length(ICDFields))+1):(indiv*length(ICDFields))]) == 0, 0, 1)
}

MHQ$HESMood[c(HESMood_Combined==1)]<-1

#####Drug-derived Diagnoses######

antidepcodes<-c("1140879616","1140921600","1140879540","1140867878","1140916282","1140909806","1140867888","1141152732","1141180212","1140879634","1140867876","1140882236",
		"1141190158","1141200564","1140867726","1140879620","1140867818","1140879630","1140879628","1141151946","1140867948","1140867624","1140867756","1140867884",
		"1141151978","1141152736","1141201834","1140867690","1140867640","1140867920","1140867850","1140879544","1141200570","1140867934","1140867758","1140867914",
		"1140867820","1141151982","1140882244","1140879556","1140867852","1140867860","1140917460","1140867938","1140867856","1140867922","1140910820","1140882312",
		"1140867944","1140867784","1140867812","1140867668","1140867940")

DrugDepression<-apply(MHQ[,grep("Treatment.medication.code", colnames(MHQ))], 1, function(row) antidepcodes %in% row)

DrugDepression_Combined<-numeric()

for(indiv in 1:dim(MHQ)[1]){
    DrugDepression_Combined[indiv]<-ifelse(sum(DrugDepression[(((indiv-1)*length(antidepcodes))+1):(indiv*length(antidepcodes))]) == 0, 0, 1)
}

MHQ$DrugDepression[DrugDepression_Combined==1]<-1

###Smith Depression Diagnosis 2013

MHQ$SmithMood<-with(MHQ, ifelse(is.na(Bipolar.and.major.depression.status.0.0), NA,
			 ifelse(!is.na(Bipolar.and.major.depression.status.0.0) & Bipolar.and.major.depression.status.0.0 > 0, 1, 0)))

#####Recent depression: PHQ-9######

MHQ$PHQ9.No.Info<-with(MHQ,ifelse((is.na(Recent.lack.of.interest.or.pleasure.in.doing.things) | Recent.lack.of.interest.or.pleasure.in.doing.things < 0) &
                                  (is.na(Recent.feelings.of.depression) | Recent.feelings.of.depression < 0),1,0))

MHQ$PHQ9.Screen<-with(MHQ,ifelse(((!is.na(Recent.lack.of.interest.or.pleasure.in.doing.things) & Recent.lack.of.interest.or.pleasure.in.doing.things >= 2) |
				  (!is.na(Recent.feelings.of.depression) & Recent.feelings.of.depression >= 2)) &
				 (!is.na(PHQ9.No.Info) & PHQ9.No.Info == 0),1,0))

MHQ$PHQ9.Response<-0

MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.lack.of.interest.or.pleasure.in.doing.things) & Recent.lack.of.interest.or.pleasure.in.doing.things > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.feelings.of.inadequacy) & Recent.feelings.of.inadequacy > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.feelings.of.depression) & Recent.feelings.of.depression > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.trouble.concentrating.on.things) & Recent.trouble.concentrating.on.things > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Trouble.falling.or.staying.asleep.or.sleeping.too.much) & Trouble.falling.or.staying.asleep.or.sleeping.too.much > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.changes.in.speed.amount.of.moving.or.speaking) & Recent.changes.in.speed.amount.of.moving.or.speaking > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.feelings.of.tiredness.or.low.energy) & Recent.feelings.of.tiredness.or.low.energy > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.poor.appetite.or.overeating) & Recent.poor.appetite.or.overeating > 0, PHQ9.Response + 1, PHQ9.Response))
MHQ$PHQ9.Response<-with(MHQ, ifelse(!is.na(Recent.thoughts.of.suicide.or.self.harm) & Recent.thoughts.of.suicide.or.self.harm > 0, PHQ9.Response + 1, PHQ9.Response))

MHQ$PHQ9.Items<-0

MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.lack.of.interest.or.pleasure.in.doing.things) & Recent.lack.of.interest.or.pleasure.in.doing.things >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.feelings.of.inadequacy) & Recent.feelings.of.inadequacy >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.feelings.of.depression) & Recent.feelings.of.depression >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.trouble.concentrating.on.things) & Recent.trouble.concentrating.on.things >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Trouble.falling.or.staying.asleep.or.sleeping.too.much) & Trouble.falling.or.staying.asleep.or.sleeping.too.much >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.changes.in.speed.amount.of.moving.or.speaking) & Recent.changes.in.speed.amount.of.moving.or.speaking >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.feelings.of.tiredness.or.low.energy) & Recent.feelings.of.tiredness.or.low.energy >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.poor.appetite.or.overeating) & Recent.poor.appetite.or.overeating >= 3, PHQ9.Items + 1, PHQ9.Items))
MHQ$PHQ9.Items<-with(MHQ, ifelse(!is.na(Recent.thoughts.of.suicide.or.self.harm) & Recent.thoughts.of.suicide.or.self.harm >= 2, PHQ9.Items + 1, PHQ9.Items))

MHQ$PHQ9.Severity<-with(MHQ, ifelse(Recent.lack.of.interest.or.pleasure.in.doing.things < 0 | is.na(Recent.lack.of.interest.or.pleasure.in.doing.things), 0, Recent.lack.of.interest.or.pleasure.in.doing.things-1) +
			     ifelse(Recent.feelings.of.inadequacy < 0 | is.na(Recent.feelings.of.inadequacy), 0, Recent.feelings.of.inadequacy-1) +
			     ifelse(Recent.feelings.of.depression < 0 | is.na(Recent.feelings.of.depression), 0, Recent.feelings.of.depression-1) +
			     ifelse(Recent.trouble.concentrating.on.things < 0 | is.na(Recent.trouble.concentrating.on.things), 0, Recent.trouble.concentrating.on.things-1) +
			     ifelse(Trouble.falling.or.staying.asleep.or.sleeping.too.much < 0 | is.na(Trouble.falling.or.staying.asleep.or.sleeping.too.much), 0, Trouble.falling.or.staying.asleep.or.sleeping.too.much-1) +
			     ifelse(Recent.changes.in.speed.amount.of.moving.or.speaking < 0 | is.na(Recent.changes.in.speed.amount.of.moving.or.speaking), 0, Recent.changes.in.speed.amount.of.moving.or.speaking-1) +
			     ifelse(Recent.feelings.of.tiredness.or.low.energy < 0 | is.na(Recent.feelings.of.tiredness.or.low.energy), 0, Recent.feelings.of.tiredness.or.low.energy-1) +
			     ifelse(Recent.thoughts.of.suicide.or.self.harm < 0 | is.na(Recent.thoughts.of.suicide.or.self.harm), 0, Recent.thoughts.of.suicide.or.self.harm-1) +
			     ifelse(Recent.poor.appetite.or.overeating < 0 | is.na(Recent.poor.appetite.or.overeating), 0, Recent.poor.appetite.or.overeating-1))

#####Lifetime depression:  CIDI-MDD######

MHQ$CIDI.MDD.No.Info<-with(MHQ,ifelse(((is.na(Ever.had.prolonged.feelings.of.sadness.or.depression) | Ever.had.prolonged.feelings.of.sadness.or.depression < 0) &
				       (is.na(Ever.had.prolonged.loss.of.interest.in.normal.activities) | Ever.had.prolonged.loss.of.interest.in.normal.activities < 0 )), 1, 0))

MHQ$CIDI.MDD.Screen<-with(MHQ,ifelse(((!is.na(Ever.had.prolonged.feelings.of.sadness.or.depression) & Ever.had.prolonged.feelings.of.sadness.or.depression == 1) |
			              (!is.na(Ever.had.prolonged.loss.of.interest.in.normal.activities) & Ever.had.prolonged.loss.of.interest.in.normal.activities == 1)) &
				     (!is.na(Fraction.of.day.affected.during.worst.episode.of.depression) & Fraction.of.day.affected.during.worst.episode.of.depression > 2) &
				     (!is.na(Frequency.of.depressed.days.during.worst.episode.of.depression) & Frequency.of.depressed.days.during.worst.episode.of.depression > 1) &
				     (!is.na(Impact.on.normal.roles.during.worst.period.of.depression) & Impact.on.normal.roles.during.worst.period.of.depression > 1), 1, 0))

MHQ$CIDI.MDD.Response<-0

MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Ever.had.prolonged.feelings.of.sadness.or.depression) & Ever.had.prolonged.feelings.of.sadness.or.depression > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Ever.had.prolonged.loss.of.interest.in.normal.activities) & Ever.had.prolonged.loss.of.interest.in.normal.activities > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Feelings.of.tiredness.during.worst.episode.of.depression) & Feelings.of.tiredness.during.worst.episode.of.depression > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Weight.change.during.worst.episode.of.depression) & Weight.change.during.worst.episode.of.depression > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Did.your.sleep.change) & Did.your.sleep.change > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Difficulty.concentrating.during.worst.depression) & Difficulty.concentrating.during.worst.depression > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Feelings.of.worthlessness.during.worst.period.of.depression) & Feelings.of.worthlessness.during.worst.period.of.depression > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))
MHQ$CIDI.MDD.Response<-with(MHQ, ifelse(!is.na(Thoughts.of.death.during.worst.depression) & Thoughts.of.death.during.worst.depression > 0, CIDI.MDD.Response + 1, CIDI.MDD.Response))

MHQ$CIDI.MDD.Severity<-with(MHQ, ifelse(Ever.had.prolonged.feelings.of.sadness.or.depression < 0 | is.na(Ever.had.prolonged.feelings.of.sadness.or.depression), 0, Ever.had.prolonged.feelings.of.sadness.or.depression) +
				 ifelse(Ever.had.prolonged.loss.of.interest.in.normal.activities < 0 | is.na(Ever.had.prolonged.loss.of.interest.in.normal.activities), 0, Ever.had.prolonged.loss.of.interest.in.normal.activities) +
				 ifelse(Feelings.of.tiredness.during.worst.episode.of.depression < 0 | is.na(Feelings.of.tiredness.during.worst.episode.of.depression), 0, Feelings.of.tiredness.during.worst.episode.of.depression) +
				 ifelse(Weight.change.during.worst.episode.of.depression < 1 | is.na(Weight.change.during.worst.episode.of.depression), 0, 1) +
				 ifelse(Did.your.sleep.change < 0 | is.na(Did.your.sleep.change), 0, Did.your.sleep.change) +
				 ifelse(Difficulty.concentrating.during.worst.depression < 0 | is.na(Difficulty.concentrating.during.worst.depression), 0, Difficulty.concentrating.during.worst.depression) +
				 ifelse(Feelings.of.worthlessness.during.worst.period.of.depression < 0 | is.na(Feelings.of.worthlessness.during.worst.period.of.depression), 0, Feelings.of.worthlessness.during.worst.period.of.depression) +
				 ifelse(Thoughts.of.death.during.worst.depression < 0 | is.na(Thoughts.of.death.during.worst.depression), 0, Thoughts.of.death.during.worst.depression))

#####Depression#####

### Depressed ever is equivalent to screening for caseness on the CIDI

MHQ$Depressed.Ever<-with(MHQ, ifelse(CIDI.MDD.No.Info == 1, NA,
			      ifelse(CIDI.MDD.Screen == 1 & (!is.na(CIDI.MDD.Response) & CIDI.MDD.Response > 4), 1,
			      ifelse(CIDI.MDD.Screen == 0 &
			             (!is.na(PHQ9.Severity) & PHQ9.Severity < 5) &
				     (is.na(SRDepression) | (!is.na(SRDepression) & SRDepression == 0)) &
				     (is.na(InterviewDepression) | (!is.na(InterviewDepression) & InterviewDepression == 0)), 0, NA))))

##Remove individuals with psychosis or mania

MHQ$Depressed.Ever.Clean<-with(MHQ, ifelse(is.na(Depressed.Ever), NA,
				    ifelse((!is.na(MHQ$SRPsychosisAny) & MHQ$SRPsychosisAny == 1) | (!is.na(MHQ$SRManiaBIP) & MHQ$SRManiaBIP == 1), NA, Depressed.Ever)))

##Remove controls with previous reports of mental illness or anti-depressant use

MHQ$Depressed.Ever.Clean<-with(MHQ, ifelse(is.na(Depressed.Ever.Clean), NA,
                                    ifelse(Depressed.Ever.Clean == 1, 1,
				    ifelse(NoSRConditions == 0 | (!is.na(DrugDepression) & DrugDepression == 1) | (!is.na(HESMood) & HESMood == 1) | (!is.na(SmithMood) & SmithMood == 1), NA, Depressed.Ever.Clean))))

# make phenotype file with IID, FID and status (0,1,NA)

Depresed.Ever.Clean.Only<-MHQ[, c("ID","Depressed.Ever.Clean")]
names(Depresed.Ever.Clean.Only)[1]<-"FID"
Depresed.Ever.Clean.Only$IID<-Depresed.Ever.Clean.Only$FID
Depresed.Ever.Clean.Only<-Depresed.Ever.Clean.Only[,c(1,3,2)]

# Print Depresed.Ever.Clean.Only

write.table(Depresed.Ever.Clean.Only, "/file_path/ever_depressed_pheno_v5.txt", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r mhq_and_screen_summarise, include=FALSE, eval=TRUE}

# DEPRESSED EVER

# format mhq depressed ever columns
mhq <- mhq[,2:3]
colnames(mhq) <- c("f.eid","MHQ")

# join mhq depressed ever with all IDs
mhq <- left_join(fam, mhq, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

# add column names to mhq depressed ever
colnames(mhq) <- c("ID","MHQ_depressed")

# make summary table
summary_mhq_dep <- data.frame("x1" = "Lifetime depression (post-QC)", "x2" = sum(mhq$MHQ_depressed>0))
colnames(summary_mhq_dep) <- c("MHQ category","Number")
rownames(summary_mhq_dep) <- NULL


# MHQ SCREEN 

# rename columns headings for mhq screen
(names( mhq_screen ) <- coding_mhq[ match( 
names( mhq_screen ) , coding_mhq[ , 'Coding' ] ) , 'Meaning' ])
colnames(mhq_screen) [1]  <- "f.eid"

# look at whole table with all disorders in screen
summary_mhq_all_screen <- data.frame("x1" = names(mhq_screen[,3:ncol(mhq_screen)]), "x2" = as.integer(format(colSums(mhq_screen[,3:ncol(mhq_screen)] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename("MHQ screening" = x1, Number = x2)

# extract required disorders for screening (psychosis). also include depression for screening controls only.
mhq_screen <- mhq_screen[,c(1,4,5,10,11)]
colnames(mhq_screen)[1] <- "ID"

# make summary table
summary_mhq_screen <- data.frame("x1" = names(mhq_screen[,2:ncol(mhq_screen)]), "x2" = as.integer(format(colSums(mhq_screen[,2:ncol(mhq_screen)] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename("MHQ screening" = x1, Number = x2)
summary_mhq_screen[4,1] <- "Depression (screen controls only)"

# change column headings for count per person columns
colnames(mhq_screen) <- c("ID","MHQ_scz","MHQ_psychosis","MHQ_bipolar","MHQ_dep_screen_controls")


# MHQ COMPLETERS

mhq_participants

table(mhq_participants$f.20446.0.0)
sum(is.na(mhq_participants$f.20446.0.0))

mhq_participants$MHQ_participants <- ifelse(is.na(mhq_participants$f.20446.0.0), 0, 1)
table(mhq_participants$MHQ_participants)

mhq_participants <- mhq_participants[,c(1,3)]
table(mhq_participants$MHQ_participants)

colnames(mhq_participants)[1] <- "ID"

```

```{r touchscreen_summarise, include=FALSE}

# format touchscreen responses
# 2090	Seen doctor (GP) for nerves, anxiety, tension or depression
# 2100	Seen a psychiatrist for nerves, anxiety, tension or depression
# Coding	Meaning
# 1	Yes
# 0	No
# -1	Do not know
# -3	Prefer not to answer

touchscreen$Seen_GP <- ifelse(touchscreen$f.2090.0.0==1 | touchscreen$f.2090.1.0==1 | touchscreen$f.2090.2.0==1, 1, 0)
touchscreen$Seen_GP <- ifelse(is.na(touchscreen$Seen_GP), 0, touchscreen$Seen_GP)
table(touchscreen$Seen_GP)
table(touchscreen$f.2090.0.0)

touchscreen$Seen_Psychiatrist <- ifelse(touchscreen$f.2100.0.0==1 | touchscreen$f.2100.1.0==1 | touchscreen$f.2100.2.0==1, 1, 0)
touchscreen$Seen_Psychiatrist <- ifelse(is.na(touchscreen$Seen_Psychiatrist), 0, touchscreen$Seen_Psychiatrist)
table(touchscreen$Seen_Psychiatrist)
table(touchscreen$f.2100.0.0)

touchscreen <- touchscreen[,c(1,8:9)]
colnames(touchscreen)[1] <- "ID"

x <- data.frame("x1" = "Have you ever seen a GP for nerves, anxiety, tension or depression?", "x2" = sum(touchscreen$Seen_GP>0, na.rm = T))
y <- data.frame("x1" = "Have you ever seen a psychiatrist for nerves, anxiety, tension or depression?", "x2" = sum(touchscreen$Seen_Psychiatrist>0, na.rm = T))
summary_touchscreen_dep <- rbind(x,y)
colnames(summary_touchscreen_dep) <- c("Touchscreen Question","Number")
rownames(summary_touchscreen_dep) <- NULL

```

```{r smith_summarise, include=FALSE}

# put D.Smith definition into columns for each phenotype

# Coding	Meaning
# 0	No Bipolar or Depression
# 1	Bipolar I Disorder
# 2	Bipolar II Disorder
# 3	Probable Recurrent major depression (severe)
# 4	Probable Recurrent major depression (moderate)
# 5	Single Probable major depression episode

smith$Smith_bipolar1 <- ifelse(smith$f.20126.0.0==1, 1, 0)
smith$Smith_bipolar1 <- ifelse(is.na(smith$Smith_bipolar1), 0, smith$Smith_bipolar1)

smith$Smith_bipolar2 <- ifelse(smith$f.20126.0.0==2, 1, 0)
smith$Smith_bipolar2 <- ifelse(is.na(smith$Smith_bipolar2), 0, smith$Smith_bipolar2)

smith$Smith_Dep_Severe <- ifelse(smith$f.20126.0.0==3, 1, 0)
smith$Smith_Dep_Severe <- ifelse(is.na(smith$Smith_Dep_Severe), 0, smith$Smith_Dep_Severe)

smith$Smith_Dep_Moderate <- ifelse(smith$f.20126.0.0==4, 1, 0)
smith$Smith_Dep_Moderate <- ifelse(is.na(smith$Smith_Dep_Moderate), 0, smith$Smith_Dep_Moderate)

smith$Smith_Dep_Single <- ifelse(smith$f.20126.0.0==5, 1, 0)
smith$Smith_Dep_Single <- ifelse(is.na(smith$Smith_Dep_Single), 0, smith$Smith_Dep_Single)

colnames(smith)[1] <- "ID"

# make summary table for depression

x <- data.frame("x1" = "Single episode of probable major depression", "x2" = sum(smith$Smith_Dep_Single))
y <- data.frame("x1" = "Probable recurrent major depression (moderate)", "x2" = sum(smith$Smith_Dep_Moderate))
z <- data.frame("x1" = "Probable recurrent major depression (severe)", "x2" = sum(smith$Smith_Dep_Severe))

summary_smith_dep <- rbind(x,y,z)
colnames(summary_smith_dep) <- c("Danny Smith Depression Phenotypes","Number")
rownames(summary_smith_dep) <- NULL

# make summary table for bipolar

x <- data.frame("x1" = "Probable bipolar disorder (type I)", "x2" = sum(smith$Smith_bipolar1))
y <- data.frame("x1" = "Probable bipolar disorder (type II)", "x2" = sum(smith$Smith_bipolar2))

summary_smith_bd <- rbind(x,y)
colnames(summary_smith_bd) <- c("Danny Smith Bipolar Phenotypes","Number")
rownames(summary_smith_bd) <- NULL

# SMITH COMPLETERS

smith_participants
smith_participants$Smith_participants <- ifelse(is.na(smith_participants$f.4598.0.0), 0, 1)
smith_participants <- smith_participants[,c(1,5)]
table(smith_participants$Smith_participants)
colnames(smith_participants)[1] <- "ID"

```

# Psychiatric phenotyping

---

</br>

# Schizophrenia & Bipolar

</br>

## Counts

</br>

### Pre-QC 

</br>

**MHQ screening: Count of individuals who report schizophrenia, psychosis, bipolar or depression**

```{r scz_print_mhq, echo=FALSE, results = 'asis'}

kable(summary_mhq_screen, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Touchscreen: Count of bipolar cases according to Danny Smith definition**

```{r scz_print_smith, echo=FALSE, results = 'asis'}

kable(summary_smith_bd, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Count of self-reported schizophrenia, bipolar**

```{r scz_print_self_report, echo=FALSE, results = 'asis'}

kable(summary_screen_self_report, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Antipsychotic Medications**

```{r scz_print_medication, echo=FALSE}

kable(summary_meds_antipsychotics, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**HES: Count of individuals who have received at least one ICD-10 dx**

```{r scz_print_icd, echo=FALSE, results = 'asis'}

kable(summary_screen_icd, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%  
  pack_rows("F20-F29 Schizophrenia, schizotypal and delusional disorders", 1, 29, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("F30-F39 Mood [affective] disorders", 30, 52, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Summary Pre-QC

</br>

```{r scz_derive_summary, include=FALSE}

# Summarise all measures in one table

total_count_antipsychotics <- data.frame(Medication = "Antipsychotic medication", Number = sum(meds_antipsychotics_count_pp$Antipsychotic>0))

scz_count_pp$ICD_any_scz <- rowSums(scz_count_pp[,-1]) 
count_icd_primary_scz <- data.frame(HES = "Primary: F20-F29 Schizophrenia, schizotypal and delusional disorders", Number = sum(scz_count_pp$Primary_scz>0))
count_icd_secondary_scz <- data.frame(HES = "Secondary: F20-F29 Schizophrenia, schizotypal and delusional disorders", Number = sum(scz_count_pp$Secondary_scz>0))
count_icd_any_scz <- data.frame(HES = "Primary and/or secondary: F20-F29 Schizophrenia, schizotypal and delusional disorders", Number = sum(scz_count_pp$ICD_any_scz>0))
count_icd_scz <- rbind(count_icd_primary_scz,count_icd_secondary_scz,count_icd_any_scz)
rownames(count_icd_scz) <- NULL

mood_count_pp$ICD_any_bipolar <- rowSums(mood_count_pp[,-1]) 
count_icd_primary_mood <- data.frame(HES = "Primary: F30-F39 Mood [affective] disorders (excl F32, F33)", Number = sum(mood_count_pp$Primary_bipolar>0))
count_icd_secondary_mood <- data.frame(HES = "Secondary: F30-F39 Mood [affective] disorders (excl F32, F33)", Number = sum(mood_count_pp$Secondary_bipolar>0))
count_icd_any_mood <- data.frame(HES = "Primary and/or secondary: F30-F39 Mood [affective] disorders (excl F32, F33)", Number = sum(mood_count_pp$ICD_any_bipolar>0))
count_icd_mood <- rbind(count_icd_primary_mood,count_icd_secondary_mood,count_icd_any_mood)
rownames(count_icd_mood) <- NULL

count_icd_psychosis <- rbind(count_icd_scz,count_icd_mood)

ChangeNames <- function(x) {
    names(x) <- c("Measure", "Number" )
    return(x)
}

dfs <- lapply(list(summary_mhq_screen,summary_smith_bd,summary_screen_self_report,total_count_antipsychotics,count_icd_psychosis), ChangeNames)

scz_summary <- rbind(dfs[[1]],dfs[[2]],dfs[[3]],dfs[[4]],dfs[[5]])

```

```{r scz_print_summary, echo=FALSE, results = 'asis'}

kable(scz_summary, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  pack_rows("MHQ screening", 1, 4, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Danny Smith Bipolar Phenotypes", 5, 6, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Self-reported disorder", 7, 8, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Medication", 9, 9, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("HES", 10, 15, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(12, bold = TRUE) %>% 
  row_spec(15, bold = TRUE) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Post-QC 

</br>

```{r scz_filter_on_qc_ids, include=FALSE}

# HES #

# Join count per person with QC'd IDs
scz_count_pp <- left_join(qc_ids, scz_count_pp, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

mood_count_pp <- left_join(qc_ids, mood_count_pp, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# Join count per person per ICD code with QC'd IDs
scz_pri <- left_join(qc_ids, scz_pri, by = c("ID" = "eid")) %>% 
  dplyr::select(-(V1:V6)) 

mood_pri <- left_join(qc_ids, mood_pri, by = c("ID" = "eid")) %>% 
  dplyr::select(-(V1:V6)) 

scz_sec <- left_join(qc_ids, scz_sec, by = c("ID" = "eid")) %>% 
  dplyr::select(-(V1:V6)) 

mood_sec <- left_join(qc_ids, mood_sec, by = c("ID" = "eid")) %>% 
  dplyr::select(-(V1:V6)) 

# Count across participants

# make function to count dx across participants
count_dx <- function(x) {
  data.frame("x1" = names(x[,-1]), "x2" = colSums(x[,-1] > 0, na.rm = TRUE), stringsAsFactors = FALSE, row.names = NULL)
}

# make function to merge ICD codes with the disorder name of each code
merge_dx <- function(x) {
  merge(coding_icd10, x, by.x = "coding", by.y = "x1")
}

# create primary and secondary tables with count across participants
scz_dx_pri <- count_dx(scz_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)
mood_dx_pri <- count_dx(mood_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)

scz_dx_sec <- count_dx(scz_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)
mood_dx_sec <- count_dx(mood_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)

# make function to merge primary and secondary columns
merge_pri_sec <- function(x,y) {
  merge(x, y, by = "ICD-10 sub-codes", all = TRUE) %>% 
  replace(., is.na(.), 0)
}

# merge primary and secondary using function

# F20-F29 Schizophrenia, schizotypal and delusional disorders
summary_scz_icd <- merge_pri_sec(scz_dx_pri,scz_dx_sec)

# Mood [affective] disorders F30, F31, F34, F38, F39
summary_mood_icd <- merge_pri_sec(mood_dx_pri,mood_dx_sec)

# Merge 
summary_screen_icd <- rbind(summary_scz_icd,summary_mood_icd)

rownames(summary_screen_icd) <- NULL

# SELF-REPORT

self_report_scz <- fread("/file_path/self_report_schizophrenia", header = T) 
self_report_bpd <- fread("/file_path/self_report_bpd", header = T) 

colnames(self_report_scz) <- c("ID",coding_scz$V2)
colnames(self_report_bpd) <- c("ID",coding_bpd$V2)

self_report_scz <- left_join(qc_ids, self_report_scz, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

self_report_bpd <- left_join(qc_ids, self_report_bpd, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# make summary tables
summary_self_report_scz <- data.frame("x1" = names(self_report_scz)[2], "x2" = as.numeric(format(sum(self_report_scz[,2]>0, na.rm = TRUE))), stringsAsFactors = F, row.names = NULL) %>% rename("temp" = x1, "Number" = x2) 
summary_self_report_scz$'Self-reported disorder' <- capitalize(summary_self_report_scz$temp)
summary_self_report_scz <- summary_self_report_scz[,c(3,2)]

summary_self_report_bpd <- data.frame("x1" = names(self_report_bpd)[2], "x2" = as.numeric(format(sum(self_report_bpd[,2]>0, na.rm = TRUE))), stringsAsFactors = F, row.names = NULL) %>% rename("temp" = x1, "Number" = x2) 
summary_self_report_bpd$'Self-reported disorder' <- capitalize(summary_self_report_bpd$temp)
summary_self_report_bpd <- summary_self_report_bpd[,c(3,2)]

summary_screen_self_report <- rbind(summary_self_report_scz,summary_self_report_bpd)
rownames(summary_screen_self_report) <- NULL

# Rename columns for self-report count per person
colnames(self_report_scz) <- c("ID","SR_scz")
colnames(self_report_bpd) <- c("ID","SR_bipolar")

# MEDICATIONS

meds_antipsychotics_count_pp <- left_join(qc_ids, meds_antipsychotics_count_pp, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

meds_antipsychotics <- left_join(qc_ids, meds_antipsychotics, by = c("ID" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) 

# make summary tables
summary_meds_antipsychotics <- data.frame("x1" = names(meds_antipsychotics[,-1]), "x2" = as.integer(format(colSums(meds_antipsychotics[,-1] > 0, na.rm = T)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Medication = x1, Count = x2)

# MHQ SCREEN 

mhq_screen <- fread("/file_path/mhq_screening_recoded", header = T) 

# rename columns headings for mhq screen
(names( mhq_screen ) <- coding_mhq[ match( 
names( mhq_screen ) , coding_mhq[ , 'Coding' ] ) , 'Meaning' ])
colnames(mhq_screen) [1]  <- "ID"

mhq_screen <- left_join(qc_ids, mhq_screen, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# extract required disorders for screening (psychosis). also include depression for screening controls only.
mhq_screen <- mhq_screen[,c(1,4,5,10,11)]

# make summary table
summary_mhq_screen <- data.frame("x1" = names(mhq_screen[,2:ncol(mhq_screen)]), "x2" = as.integer(format(colSums(mhq_screen[,2:ncol(mhq_screen)] > 0, na.rm = TRUE)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename("MHQ screening" = x1, Number = x2)
summary_mhq_screen[4,1] <- "Depression (screen controls only)"

# change column headings for count per participants columns
colnames(mhq_screen) <- c("ID","MHQ_scz","MHQ_psychosis","MHQ_bipolar","MHQ_dep_screen_controls")

# MHQ COMPLETERS

mhq_participants <- left_join(qc_ids, mhq_participants, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# SMITH

smith <- left_join(qc_ids, smith, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# make summary table for bipolar

x <- data.frame("x1" = "Probable bipolar disorder (type I)", "x2" = sum(smith$Smith_bipolar1, na.rm = TRUE))
y <- data.frame("x1" = "Probable bipolar disorder (type II)", "x2" = sum(smith$Smith_bipolar2, na.rm = TRUE))

summary_smith_bd <- rbind(x,y)
colnames(summary_smith_bd) <- c("Danny Smith Bipolar Phenotypes","Number")
rownames(summary_smith_bd) <- NULL

# SMITH COMPLETERS

smith_participants <- left_join(qc_ids, smith_participants, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

```

```{r make_one_df_scz_all_measures_on_qc_ids, include=FALSE}

# Make one df across measures
scz_all_measures <- plyr::join_all(list(scz_count_pp, mood_count_pp, 
                                               self_report_scz, self_report_bpd,
                                               meds_antipsychotics_count_pp,
                                               smith[,c(1,3:4)], 
                                               mhq_screen), by='ID')

scz_all_measures$Count_any_psychosis_measure <- rowSums(scz_all_measures[,2:15])

scz_all_measures$Any_psychosis_measure <- ifelse(scz_all_measures$Count_any_psychosis_measure>0, 1, 0)

scz_all_measures <- scz_all_measures[,-17]

scz_all_measures$Count_any_psychosis_measure_and_mhq_depression_screen <- rowSums(scz_all_measures[,2:16])

scz_all_measures$Any_psychosis_measure_and_MHQ_dep_screen <- ifelse(scz_all_measures$Count_any_psychosis_measure_and_mhq_depression_screen>0, 1, 0)

scz_all_measures <- scz_all_measures[,-18]

# Make case/control columns to replace counts 
scz_all_measures$Primary_scz <- ifelse(scz_all_measures$Primary_scz>0,1,0)
scz_all_measures$Secondary_scz <- ifelse(scz_all_measures$Secondary_scz>0,1,0)
scz_all_measures$ICD_any_scz <- ifelse(scz_all_measures$ICD_any_scz>0,1,0)
scz_all_measures$Primary_bipolar <- ifelse(scz_all_measures$Primary_bipolar>0,1,0)
scz_all_measures$Secondary_bipolar <- ifelse(scz_all_measures$Secondary_bipolar>0,1,0)
scz_all_measures$ICD_any_bipolar <- ifelse(scz_all_measures$ICD_any_bipolar>0,1,0)

scz_all_measures$SR_scz <- ifelse(scz_all_measures$SR_scz>0,1,0)
scz_all_measures$SR_bipolar <- ifelse(scz_all_measures$SR_bipolar>0,1,0)

scz_all_measures$Antipsychotic <- ifelse(scz_all_measures$Antipsychotic>0,1,0)

names(scz_all_measures)

write.table(scz_all_measures, file="/file_path/psychosis_phenotypes", row.names=F, col.name=T, quote=F, sep="\t")

```

**MHQ screening: Count of individuals who report schizophrenia, psychosis, bipolar or depression**

```{r qc_scz_print_mhq, echo=FALSE, results = 'asis'}

kable(summary_mhq_screen, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Touchscreen: Count of bipolar cases according to Danny Smith definition**

```{r qc_scz_print_smith, echo=FALSE, results = 'asis'}

kable(summary_smith_bd, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Count of self-reported schizophrenia, bipolar**

```{r qc_scz_print_self_report, echo=FALSE, results = 'asis'}

kable(summary_screen_self_report, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Antipsychotic Medications**

```{r qc_scz_print_medication, echo=FALSE}

kable(summary_meds_antipsychotics, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**HES: Count of individuals who have received at least one ICD-10 dx**

```{r qc_scz_print_icd, echo=FALSE, results = 'asis'}

kable(summary_screen_icd, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%  
  pack_rows("F20-F29 Schizophrenia, schizotypal and delusional disorders", 1, 29, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("F30-F39 Mood [affective] disorders", 30, 52, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Summary Post-QC

</br>

```{r qc_scz_derive_summary, include=FALSE}

# Summary table across measures

total_count_antipsychotics <- data.frame(Medication = "Antipsychotic medication", Number = sum(meds_antipsychotics_count_pp$Antipsychotic>0, na.rm = T))

count_icd_primary_scz <- data.frame(HES = "Primary: F20-F29 Schizophrenia, schizotypal and delusional disorders", Number = sum(scz_count_pp$Primary_scz>0, na.rm = T))
count_icd_secondary_scz <- data.frame(HES = "Secondary: F20-F29 Schizophrenia, schizotypal and delusional disorders", Number = sum(scz_count_pp$Secondary_scz>0, na.rm = T))
count_icd_any_scz <- data.frame(HES = "Primary and/or secondary: F20-F29 Schizophrenia, schizotypal and delusional disorders", Number = sum(scz_count_pp$ICD_any_scz>0, na.rm = T))
count_icd_scz <- rbind(count_icd_primary_scz,count_icd_secondary_scz,count_icd_any_scz)
rownames(count_icd_scz) <- NULL

count_icd_primary_mood <- data.frame(HES = "Primary: F30-F39 Mood [affective] disorders (excl F32, F33)", Number = sum(mood_count_pp$Primary_bipolar>0, na.rm = T))
count_icd_secondary_mood <- data.frame(HES = "Secondary: F30-F39 Mood [affective] disorders (excl F32, F33)", Number = sum(mood_count_pp$Secondary_bipolar>0, na.rm = T))
count_icd_any_mood <- data.frame(HES = "Primary and/or secondary: F30-F39 Mood [affective] disorders (excl F32, F33)", Number = sum(mood_count_pp$ICD_any_bipolar>0, na.rm = T))
count_icd_mood <- rbind(count_icd_primary_mood,count_icd_secondary_mood,count_icd_any_mood)
rownames(count_icd_mood) <- NULL

count_icd_psychosis <- rbind(count_icd_scz,count_icd_mood)

count_any_psychosis_no_mhq_screen <- data.frame(Total = "Any psychosis measure (MHQ screen, Smith, Self Report, Medication, HES)", Number = sum(scz_all_measures$Any_psychosis_measure, na.rm = T))
count_any_psychosis_mhq_screen <- data.frame(Total = "Any psychosis measure and/or report depression in MHQ screen (for screening controls)", Number = sum(scz_all_measures$Any_psychosis_measure_and_MHQ_dep_screen, na.rm = T))

count_any_psychosis <- rbind(count_any_psychosis_no_mhq_screen,count_any_psychosis_mhq_screen)
rownames(count_any_psychosis) <- NULL

ChangeNames <- function(x) {
    names(x) <- c("Measure", "Number" )
    return(x)
}

dfs <- lapply(list(summary_mhq_screen,summary_smith_bd,summary_screen_self_report,total_count_antipsychotics,count_icd_psychosis,count_any_psychosis), ChangeNames)

scz_summary <- rbind(dfs[[1]],dfs[[2]],dfs[[3]],dfs[[4]],dfs[[5]],dfs[[6]])

```

```{r qc_scz_print_summary, echo=FALSE, results = 'asis'}

kable(scz_summary, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  pack_rows("MHQ screening", 1, 4, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Danny Smith Bipolar Phenotypes", 5, 6, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Self-reported disorder", 7, 8, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Medication", 9, 9, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("HES", 10, 15, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Any Measure", 16, 17, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(12, bold = TRUE) %>% 
  row_spec(15, bold = TRUE) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

# Depression

</br>

## Counts

</br>

### Pre-QC 

</br>

**MHQ: Count of individuals meeting criteria for lifetime depression**

```{r dep_print_mhq, echo=FALSE, results = 'asis'}

kable(summary_mhq_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Touchscreen: Count of "help-seeking" individuals**

```{r dep_print_touchscreen, echo=FALSE, results = 'asis'}

kable(summary_touchscreen_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Touchscreen: Count of cases according to Danny Smith definition**

```{r dep_print_smith, echo=FALSE, results = 'asis'}

kable(summary_smith_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Count of self-reported depression**

```{r dep_print_self_report, echo = FALSE, results = 'asis'}

kable(summary_self_report_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Antidepressant Medications**

```{r dep_print_medication, echo=FALSE}

kable(summary_meds_antidepressants, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**HES: Count of individuals who have received at least one ICD-10 dx**

```{r dep_print_icd, echo = FALSE, results = 'asis'}

kable(summary_dep_icd, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  pack_rows("F32 Depressive episode", 1, 7, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("F33 Recurrent depressive disorder", 8, 14, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Summary Pre-QC

</br>

```{r dep_derive_summary, include=FALSE}

# Make summary table of all measures

total_count_antidepressants <- data.frame(Medication = "Antidepressant medication", Number = sum(meds_antidepressants_count_pp$Antidepressant>0))

dep_count_pp$ICD_any_depression <- rowSums(dep_count_pp[,-1]) 
count_icd_primary_dep <- data.frame(HES = "Primary: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$Primary_depression>0))
count_icd_secondary_dep <- data.frame(HES = "Secondary: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$Secondary_depression>0))
count_icd_any_dep <- data.frame(HES = "Primary and/or secondary: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$ICD_any_depression>0))
count_icd_dep <- rbind(count_icd_primary_dep,count_icd_secondary_dep,count_icd_any_dep)
rownames(count_icd_dep) <- NULL

ChangeNames <- function(x) {
    names(x) <- c("Measure", "Number" )
    return(x)
}

dfs <- lapply(list(summary_mhq_dep,summary_smith_dep,summary_touchscreen_dep,summary_self_report_dep,count_icd_dep,total_count_antidepressants), ChangeNames)

dep_summary <- rbind(dfs[[1]],dfs[[2]],dfs[[3]],dfs[[4]],dfs[[5]],dfs[[6]])

rownames(dep_summary) <- NULL

```

```{r dep_print_summary, echo=FALSE, results = 'asis'}

kable(dep_summary, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  pack_rows("MHQ", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Danny Smith Depression Phenotypes", 2, 4, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Touchscreen", 5, 6, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Self-reported disorder", 7, 7, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("HES", 8, 10, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Medication", 11, 11, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(10, bold = TRUE) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>


### Post-QC 

</br>

```{r dep_filter_on_qc_ids, include=FALSE}

# Filter on participants passing genetic QC

# HES #

# Join count per person with QC'd IDs
dep_count_pp <- left_join(qc_ids, dep_count_pp, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# Join count per person per ICD code with QC'd IDs
dep_pri <- left_join(qc_ids, dep_pri, by = c("ID" = "eid")) %>% 
  dplyr::select(-(V1:V6)) 

dep_sec <- left_join(qc_ids, dep_sec, by = c("ID" = "eid")) %>% 
  dplyr::select(-(V1:V6)) 

# Count across people

# make function to count dx across people
count_dx <- function(x) {
  data.frame("x1" = names(x[,-1]), "x2" = colSums(x[,-1] > 0, na.rm = TRUE), stringsAsFactors = FALSE, row.names = NULL)
}

# make function to merge ICD codes with the disorder name of each code
merge_dx <- function(x) {
  merge(coding_icd10, x, by.x = "coding", by.y = "x1")
}

# create primary and secondary tables with count across people
dep_dx_pri <- count_dx(dep_pri) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary" = x2)
dep_dx_sec <- count_dx(dep_sec) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary" = x2)

# make function to merge primary and secondary columns
merge_pri_sec <- function(x,y) {
  merge(x, y, by = "ICD-10 sub-codes", all = TRUE) %>% 
  replace(., is.na(.), 0)
}

# merge primary and secondary using function
summary_dep_icd <- merge_pri_sec(dep_dx_pri,dep_dx_sec)

# SELF-REPORT

self_report_dep <- fread("/file_path/self_report_dep", header = T) 

colnames(self_report_dep) <- c("ID",coding_dep$V2)

self_report_dep <- left_join(qc_ids, self_report_dep, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# make summary tables
summary_self_report_dep <- data.frame("x1" = names(self_report_dep)[2], "x2" = as.numeric(format(sum(self_report_dep[,2]>0, na.rm = TRUE))), stringsAsFactors = F, row.names = NULL) %>% rename("temp" = x1, "Number" = x2) 
summary_self_report_dep$'Self-reported disorder' <- capitalize(summary_self_report_dep$temp)
summary_self_report_dep <- summary_self_report_dep[,c(3,2)]

# Rename columns for self-report count per person
colnames(self_report_dep) <- c("ID","SR_depression")

# MEDICATIONS

meds_antidepressants_count_pp <- left_join(qc_ids, meds_antidepressants_count_pp, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

meds_antidepressants <- left_join(qc_ids, meds_antidepressants, by = c("ID" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) 

# make summary tables
summary_meds_antidepressants <- data.frame("x1" = names(meds_antidepressants[,-1]), "x2" = as.integer(format(colSums(meds_antidepressants[,-1] > 0, na.rm = T)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Medication = x1, Count = x2)

# MHQ 

mhq <- left_join(qc_ids, mhq, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

# SMITH

# make summary table for depression

smith$Smith_Depression <- rowSums(smith[,5:7])

x <- data.frame("x1" = "Single episode of probable major depression", "x2" = sum(smith$Smith_Dep_Single, na.rm = T))
y <- data.frame("x1" = "Probable recurrent major depression (moderate)", "x2" = sum(smith$Smith_Dep_Moderate, na.rm = T))
z <- data.frame("x1" = "Probable recurrent major depression (severe)", "x2" = sum(smith$Smith_Dep_Severe, na.rm = T))
w <- data.frame("x1" = "Total: Danny Smith derived depression phenotypes", "x2" = sum(smith$Smith_Depression, na.rm = T))

summary_smith_dep <- rbind(x,y,z,w)
colnames(summary_smith_dep) <- c("Danny Smith Depression Phenotypes","Number")
rownames(summary_smith_dep) <- NULL

# TOUCHSCREEN

touchscreen$Seen_GP_or_Psychiatrist <- rowSums(touchscreen[,2:3])

touchscreen <- left_join(qc_ids, touchscreen, by = "ID") %>% 
  dplyr::select(-(V1:V6)) 

x <- data.frame("x1" = "Have you ever seen a GP for nerves, anxiety, tension or depression?", "x2" = sum(touchscreen$Seen_GP>0, na.rm = T))
y <- data.frame("x1" = "Have you ever seen a psychiatrist for nerves, anxiety, tension or depression?", "x2" = sum(touchscreen$Seen_Psychiatrist>0, na.rm = T))
z <- data.frame("x1" = "Total: Seen a GP or psychiatrist for nerves, anxiety, tension or depression?", "x2" = sum(touchscreen$Seen_GP_or_Psychiatrist>0, na.rm = T))

summary_touchscreen_dep <- rbind(x,y,z)
colnames(summary_touchscreen_dep) <- c("Touchscreen Question","Number")
rownames(summary_touchscreen_dep) <- NULL

```

```{r make_one_df_dep_all_measures_on_qc_ids_and_one_df_all_psych, include=FALSE}

# merge depression columns  
dep_all_measures <- plyr::join_all(list(dep_count_pp, 
                                               self_report_dep,
                                               meds_antidepressants_count_pp,
                                               smith[,c(1,5:8)],
                                               touchscreen,  
                                               mhq), by='ID')

dep_all_measures$Count_any_depression_measure <- rowSums(dep_all_measures[,2:ncol(dep_all_measures)])
dep_all_measures$Any_depression_measure <- ifelse(dep_all_measures$Count_any_depression_measure>0, 1, 0)
dep_all_measures <- dep_all_measures[,-15]
dep_all_measures$Count_any_depression_measure_no_touchscreen <- rowSums(dep_all_measures[,c(2:10,14)])
dep_all_measures$Any_depression_measure_no_touchscreen <- ifelse(dep_all_measures$Count_any_depression_measure_no_touchscreen>0, 1, 0)
dep_all_measures <- dep_all_measures[,-16]

# Make case/control columns to replace counts 
dep_all_measures$Primary_depression <- ifelse(dep_all_measures$Primary_depression>0,1,0)
dep_all_measures$Secondary_depression <- ifelse(dep_all_measures$Secondary_depression>0,1,0)
dep_all_measures$ICD_any_depression <- ifelse(dep_all_measures$ICD_any_depression>0,1,0)
dep_all_measures$SR_depression <- ifelse(dep_all_measures$SR_depression>0,1,0)
dep_all_measures$Antidepressant <- ifelse(dep_all_measures$Antidepressant>0,1,0)
dep_all_measures$Seen_GP_or_Psychiatrist <- ifelse(dep_all_measures$Seen_GP_or_Psychiatrist>0,1,0)

write.table(dep_all_measures, file="/file_path/depression_phenotypes", row.names=F, col.name=T, quote=F, sep="\t")

# merge with psychosis dataframe, MHQ participants, Smith participants
psych_all_measures <- merge(dep_all_measures,scz_all_measures, by.x = "ID", by.y = "ID")

# make supercontrols
psych_all_measures$Super_controls <- ifelse(psych_all_measures$Any_depression_measure>0 | psych_all_measures$Any_psychosis_measure_and_MHQ_dep_screen>0, 0, 1)

super_controls <- psych_all_measures[,c(1,34)]
super_controls <- super_controls[super_controls$Super_controls==1,]
super_controls <- super_controls[!is.na(super_controls$Super_controls),]
super_controls$Super_controls <- 0
rownames(super_controls) <- NULL
sum(psych_all_measures$Super_controls, na.rm = T)

# For flowchart - calculate number with any depression measure, removing those excluded for psychosis
psych_all_measures$Any_depression_measure_no_psychosis <-
  ifelse(psych_all_measures$Any_depression_measure>0 & psych_all_measures$Any_psychosis_measure == 0, 1, 0)
sum(psych_all_measures$Any_depression_measure_no_psychosis==1, na.rm = T)

# Count negative IDs (indicating withdrawls)
sum(psych_all_measures$ID < 0)

# Count individuals with indication for depression on MHQ screening section, but no other indication of depression or psychosis in UKB (excluded from analysis)
psych_all_measures$MHQ_dep_screen_no_other_dep_or_psychosis <- ifelse(
  psych_all_measures$MHQ_dep_screen_controls==1 &
    psych_all_measures$Any_depression_measure==0 &
    psych_all_measures$Any_psychosis_measure==0, 1, 0)

sum(psych_all_measures$MHQ_dep_screen_no_other_dep_or_psychosis==1, na.rm = T)

```

</br>

**MHQ: Count of individuals meeting criteria for lifetime depression**

```{r qc_dep_print_mhq, echo=FALSE, results = 'asis'}

kable(summary_mhq_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Touchscreen: Count of "help-seeking" individuals**

```{r qc_dep_print_touchscreen, echo=FALSE, results = 'asis'}

kable(summary_touchscreen_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  row_spec(3, bold = TRUE)

```

</br>

**Touchscreen: Count of cases according to Danny Smith definition**

```{r qc_dep_print_smith, echo=FALSE, results = 'asis'}

kable(summary_smith_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  row_spec(4, bold = TRUE)

```

</br>

**Nurse interview: Count of self-reported depression**

```{r qc_dep_print_self_report, echo = FALSE, results = 'asis'}

kable(summary_self_report_dep, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Nurse interview: Antidepressant Medications**

```{r qc_dep_print_medication, echo=FALSE}

kable(summary_meds_antidepressants, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**HES: Count of individuals who have received at least one ICD-10 dx**

```{r qc_dep_print_icd, echo = FALSE, results = 'asis'}

kable(summary_dep_icd, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  pack_rows("F32 Depressive episode", 1, 7, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("F33 Recurrent depressive disorder", 8, 14, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Summary Post-QC

</br>

```{r qc_dep_derive_summary, echo=FALSE}

# Make summary table
total_count_antidepressants <- data.frame(Medication = "Antidepressant medication", Number = sum(meds_antidepressants_count_pp$Antidepressant>0, na.rm = T))

count_icd_primary_dep <- data.frame(HES = "Primary: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$Primary_depression>0, na.rm = T))
count_icd_secondary_dep <- data.frame(HES = "Secondary: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$Secondary_depression>0, na.rm = T))
count_icd_single <- data.frame(HES = "Single ICD-10 dx: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$ICD_any_depression==1, na.rm = T))
count_icd_multiple <- data.frame(HES = "Multiple ICD-10 dx: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$ICD_any_depression>1, na.rm = T))
count_icd_any_dep <- data.frame(HES = "Primary and/or secondary: F32, F33 Mood [affective] disorders", Number = sum(dep_count_pp$ICD_any_depression>0, na.rm = T))

count_icd_dep <- rbind(count_icd_primary_dep,count_icd_secondary_dep,count_icd_single,count_icd_multiple, count_icd_any_dep)
rownames(count_icd_dep) <- NULL

count_any_depression <- data.frame(Total = "Any depression measure (MHQ, Smith, Help-Seeking, Self Report, HES, Medication)", Number = sum(psych_all_measures$Any_depression_measure, na.rm = T))
count_any_depression_no_touchscreen <- data.frame(Total = "Any depression measure wihtout Help-Seeking (MHQ, Smith, Self Report, HES, Medication)", Number = sum(psych_all_measures$Any_depression_measure_no_touchscreen, na.rm = T))

count_any_depression <- rbind(count_any_depression,count_any_depression_no_touchscreen)
rownames(count_any_depression) <- NULL

ChangeNames <- function(x) {
    names(x) <- c("Measure", "Number" )
    return(x)
}

dfs <- lapply(list(summary_mhq_dep,summary_smith_dep,summary_touchscreen_dep,summary_self_report_dep,count_icd_dep,total_count_antidepressants,count_any_depression), ChangeNames)

dep_summary <- rbind(dfs[[1]],dfs[[2]],dfs[[3]],dfs[[4]],dfs[[5]],dfs[[6]],dfs[[7]])

rownames(dep_summary) <- NULL

```

```{r qc_dep_print_summary, echo=FALSE, results = 'asis'}

kable(dep_summary, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  pack_rows("MHQ", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Danny Smith Depression Phenotypes", 2, 5, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Touchscreen", 6, 8, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Self-reported disorder", 9, 9, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("HES", 10, 14, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Medication", 15, 15, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Any Measure", 16, 17, label_row_css = "background-color: #888; color: #fff;") %>%
  row_spec(5, bold = TRUE) %>% 
  row_spec(8, bold = TRUE) %>% 
  row_spec(14, bold = TRUE) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Summary after screening for psychosis

</br>

```{r add_mhq_and_smith_participant_and_icd_single_multiple_columns_to_psych_all_measures, echo=FALSE}

# merge with all psych dataframe with MHQ participants, Smith participants, ICD depression single and multiple

psych_all_measures <- merge(psych_all_measures,mhq_participants, by.x = "ID", by.y = "ID")
# test <- merge(psych_all_measures,mhq_participants, by.x = "ID", by.y = "ID")
psych_all_measures <- merge(psych_all_measures,smith_participants, by.x = "ID", by.y = "ID")

dep_count_pp$Single_ICD_depression <- ifelse(dep_count_pp$ICD_any_depression==1,1,0)
dep_count_pp$Multiple_ICD_depression <- ifelse(dep_count_pp$ICD_any_depression>1,1,0)

psych_all_measures <- merge(psych_all_measures,dep_count_pp[,c(1,5)], by.x = "ID", by.y = "ID")
psych_all_measures <- merge(psych_all_measures,dep_count_pp[,c(1,6)], by.x = "ID", by.y = "ID")

psych_all_measures <- psych_all_measures %>%
  drop_na()

rownames(psych_all_measures) <- NULL

```

```{r screen_dep_cases_for_psychosis, echo=FALSE}

# screen depression cases for any psychosis measure
# make copy of df before screening
psych_all_measures_pre_psychosis_screen_in_dep_cases <- psych_all_measures

psych_all_measures$Remove_psychosis_NA <- ifelse(psych_all_measures$Any_psychosis_measure==1, NA, psych_all_measures$Any_psychosis_measure)

psych_all_measures <- psych_all_measures %>%
  drop_na()

comparison_before_after_screen_depression_for_psychosis <- data.frame("Measure" = names(psych_all_measures[,c(14,10,11,12,5,4,6)]), "Pre Psychosis Screen" = colSums(psych_all_measures_pre_psychosis_screen_in_dep_cases[,c(14,10,11,12,5,4,6)]), "Post Psychosis Screen" = colSums(psych_all_measures[,c(14,10,11,12,5,4,6)]))

rownames(comparison_before_after_screen_depression_for_psychosis) <- NULL
colnames(comparison_before_after_screen_depression_for_psychosis) <- c("Measure","Pre Psychosis Screen","Post Psychosis Screen")

```

</br>

```{r print_before_and_after_psychosis_screen, echo=FALSE, results = 'asis'}

kable(comparison_before_after_screen_depression_for_psychosis, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Overlap, Post-QC and psychosis screen

</br>

```{r upset_overlap_including_mhq_participants, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=15}

overlap_including_mhq_participants <- upset(psych_all_measures, 
                                            sets = c("ICD_any_depression", "SR_depression", "Antidepressant", "Smith_Depression", "Seen_GP_or_Psychiatrist", "MHQ_depressed"), 
                                            mb.ratio = c(0.55, 0.45), order.by = "freq")

overlap_including_mhq_participants

```

**Plot:** Overlap between HES, Self-report, Medication, MHQ, Smith and Help-Seeking in the UKB

</br>

```{r make_df_without_mhq_participants, include=FALSE}

# Make df without MHQ participants to show participants with 1 to 5 non-MHQ measures
psych_all_measures_without_mhq_participants <- psych_all_measures

psych_all_measures_without_mhq_participants$Remove_mhq_NA <- ifelse(psych_all_measures_without_mhq_participants$MHQ_participants==1, NA, psych_all_measures_without_mhq_participants$MHQ_participants)

psych_all_measures_without_mhq_participants <- psych_all_measures_without_mhq_participants %>%
  drop_na()

```

```{r upset_overlap_without_mhq_participants, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

# Plot measures in non-MHQ participants only
colnames(psych_all_measures_without_mhq_participants)[4] <- "Hospital (ICD-10)"
colnames(psych_all_measures_without_mhq_participants)[5] <- "Self-reported depression"
colnames(psych_all_measures_without_mhq_participants)[10] <- "Depression (Smith)"
colnames(psych_all_measures_without_mhq_participants)[13] <- "Help-seeking"

overlap_without_mhq_participants <- upset(psych_all_measures_without_mhq_participants, 
      sets = c("Hospital (ICD-10)", "Self-reported depression", "Antidepressant",
               "Depression (Smith)", "Help-seeking"), 
      mb.ratio = c(0.6, 0.4), 
      order.by = c("freq","degree"),
      mainbar.y.label = "N people per intersection",
      sets.x.label = "\nN people per measure",
      point.size = 2.5,
      line.size = 0.5,
      text.scale = c(1.3, 1.3, 1.3, 1.3, 1.5, 1.2),
      main.bar.color = c("red","red","red","red",
                         "orange","orange","orange","orange","orange","orange","orange",
                         "green3","green3","green3","green3","green3","green3","green3",
                         "blue","blue","blue","blue",
                         "orchid4"))

overlap_without_mhq_participants

```

**Plot:** Overlap between HES, Self-report, Medication, Smith and Help-Seeking AFTER removing MHQ participants

</br>

For paper:

```{r upstart_overlap_without_mhq_participants_for_paper, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=12}

colnames(psych_all_measures_without_mhq_participants)[5] <- "Self-reported Depression"
colnames(psych_all_measures_without_mhq_participants)[6] <- "Antidepressant Usage"

overlap_without_mhq_participants <- upset(psych_all_measures_without_mhq_participants, 
      sets = c("Hospital (ICD-10)", "Self-reported Depression", "Antidepressant Usage",
               "Depression (Smith)", "Help-seeking"), 
      mb.ratio = c(0.6, 0.4), 
      order.by = c("freq","degree"),
      mainbar.y.label = "No. individuals per intersection",
      sets.x.label = "\nNo. individuals per measure",
      point.size = 2.5,
      line.size = 0.5,
      text.scale = c(1.5, 1.5, 1.5, 1.5, 1.7, 1.35),
      main.bar.color = c("red","red","red","red",
                         "darkorange1","darkorange1","darkorange1","darkorange1","darkorange1","darkorange1","darkorange1",
                         "green4","green4","green4","green4","green4","green4","green4",
                         "blue","blue","blue","blue",
                         "deeppink"))

overlap_without_mhq_participants

# Change back to original column in case df used again in subsequent chunks
colnames(psych_all_measures_without_mhq_participants)[6] <- "Antidepressant"
colnames(psych_all_measures_without_mhq_participants)[5] <- "Self reported depression"

```

**Plot:** Overlap between HES, Self-report, Medication, Smith and Help-Seeking AFTER removing MHQ participants

</br>

# Polygenic Risk Scores

</br>

```{r prs_pheno_prep_main_and_sub_categories_vs_232k_controls, include=FALSE}

# make file for measures groups and [sub groups] versus super_controls (232k): MHQ, Help-Seeking [GP, psychiatrist], Smith [One, Moderate, Severe] Self-report, Antidepressants, HES [primary, secondary, single, multiple]

mhq <- super_controls %>% 
  rename(MHQ_depressed = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$MHQ_depressed==1,c(1,14)])

help_seeking <- super_controls %>% 
  rename(Seen_GP_or_Psychiatrist = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Seen_GP_or_Psychiatrist==1,c(1,13)])

seen_gp <- super_controls %>% 
  rename(Seen_GP = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Seen_GP==1,c(1,11)])

seen_psychiatrist <- super_controls %>% 
  rename(Seen_Psychiatrist = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Seen_Psychiatrist==1,c(1,12)])

smith <- super_controls %>% 
  rename(Smith_Depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Depression==1,c(1,10)])

smith_single <- super_controls %>% 
  rename(Smith_Dep_Single = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Dep_Single==1,c(1,9)])

smith_moderate <- super_controls %>% 
  rename(Smith_Dep_Moderate = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Dep_Moderate==1,c(1,8)])

smith_severe <- super_controls %>% 
  rename(Smith_Dep_Severe = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Dep_Severe==1,c(1,7)])

self_report <- super_controls %>% 
  rename(SR_depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$SR_depression==1,c(1,5)])

antidepressants <- super_controls %>% 
  rename(Antidepressant = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Antidepressant==1,c(1,6)])

hes <- super_controls %>% 
  rename(ICD_any_depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$ICD_any_depression==1,c(1,4)])

hes_primary <- super_controls %>% 
  rename(Primary_depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Primary_depression==1,c(1,2)])

hes_secondary <- super_controls %>% 
  rename(Secondary_depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Secondary_depression==1,c(1,3)])

hes_single <- super_controls %>% 
  rename(Single_ICD_depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Single_ICD_depression==1,c(1,37)])

hes_multiple <- super_controls %>% 
  rename(Multiple_ICD_depression = Super_controls) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Multiple_ICD_depression==1,c(1,38)])

dep_main_sub_cat <- plyr::join_all(list(mhq, help_seeking, seen_gp, seen_psychiatrist, smith, smith_single, smith_moderate, smith_severe, self_report, antidepressants, hes, hes_primary, hes_secondary, hes_single, hes_multiple), by = "ID", type = 'full')

write.table(dep_main_sub_cat, file="/file_path/dep_main_sub_cat", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r create_mhq_controls, include=FALSE}

# make MHQ controls

mhq_controls <- fread("/file_path/ever_depressed_pheno_final.txt", header = T) 
mhq_controls <- mhq_controls[mhq_controls$Depressed.Ever.Clean==0,]

super_controls

mhq_controls <- merge(mhq_controls, super_controls, by.x = "IID", by.y = "ID")
mhq_controls <- mhq_controls[,c(1,3)]
colnames(mhq_controls) <- c("ID","MHQ_controls_extra_screen")

```

```{r prs_pheno_prep_main_and_sub_categories_vs_58k_controls, include=FALSE}

# make file for measures and [sub groups] versus mhq controls (58k): MHQ, Help-Seeking [GP, psychiatrist], Smith [One, Moderate, Severe] Self-report, Antidepressants, HES [primary, secondary, single, multiple]

mhq <- mhq_controls %>% 
  rename(MHQ_depressed = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$MHQ_depressed==1,c(1,14)])

help_seeking <- mhq_controls %>% 
  rename(Seen_GP_or_Psychiatrist = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Seen_GP_or_Psychiatrist==1,c(1,13)])

seen_gp <- mhq_controls %>% 
  rename(Seen_GP = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Seen_GP==1,c(1,11)])

seen_psychiatrist <- mhq_controls %>% 
  rename(Seen_Psychiatrist = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Seen_Psychiatrist==1,c(1,12)])

smith <- mhq_controls %>% 
  rename(Smith_Depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Depression==1,c(1,10)])

smith_single <- mhq_controls %>% 
  rename(Smith_Dep_Single = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Dep_Single==1,c(1,9)])

smith_moderate <- mhq_controls %>% 
  rename(Smith_Dep_Moderate = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Dep_Moderate==1,c(1,8)])

smith_severe <- mhq_controls %>% 
  rename(Smith_Dep_Severe = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Smith_Dep_Severe==1,c(1,7)])

self_report <- mhq_controls %>% 
  rename(SR_depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$SR_depression==1,c(1,5)])

antidepressants <- mhq_controls %>% 
  rename(Antidepressant = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Antidepressant==1,c(1,6)])

hes <- mhq_controls %>% 
  rename(ICD_any_depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$ICD_any_depression==1,c(1,4)])

hes_primary <- mhq_controls %>% 
  rename(Primary_depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Primary_depression==1,c(1,2)])

hes_secondary <- mhq_controls %>% 
  rename(Secondary_depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Secondary_depression==1,c(1,3)])

hes_single <- mhq_controls %>% 
  rename(Single_ICD_depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Single_ICD_depression==1,c(1,37)])

hes_multiple <- mhq_controls %>% 
  rename(Multiple_ICD_depression = MHQ_controls_extra_screen) %>% 
  bind_rows(., psych_all_measures[psych_all_measures$Multiple_ICD_depression==1,c(1,38)])

dep_main_sub_cat_mhq_controls <- plyr::join_all(list(mhq, help_seeking, seen_gp, seen_psychiatrist, smith, smith_single, smith_moderate, smith_severe, self_report, antidepressants, hes, hes_primary, hes_secondary, hes_single, hes_multiple), by = "ID", type = 'full')

write.table(dep_main_sub_cat_mhq_controls, file="/file_path/dep_main_sub_cat_mhq_controls", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r prs_pheno_prep_overlapping_vs_232k_controls, include=FALSE}

# calculate number of measures in non-mhq participants, between one and five
psych_all_measures_without_mhq_participants$Overlapping <- rowSums(psych_all_measures_without_mhq_participants[,c(4,5,6,10,13)])
table(psych_all_measures_without_mhq_participants$Overlapping)

# make columns for each set of cases
psych_all_measures_without_mhq_participants$One_Measure <- ifelse(psych_all_measures_without_mhq_participants$Overlapping==1, 1, 0)
psych_all_measures_without_mhq_participants$Two_Measures <- ifelse(psych_all_measures_without_mhq_participants$Overlapping==2, 1, 0)
psych_all_measures_without_mhq_participants$Three_Measures <- ifelse(psych_all_measures_without_mhq_participants$Overlapping==3, 1, 0)
psych_all_measures_without_mhq_participants$Four_Measures <- ifelse(psych_all_measures_without_mhq_participants$Overlapping==4, 1, 0)
psych_all_measures_without_mhq_participants$Five_Measures <- ifelse(psych_all_measures_without_mhq_participants$Overlapping==5, 1, 0)
rownames(psych_all_measures_without_mhq_participants) <- NULL

# select required columns
no_mhq_cases_overlap <- psych_all_measures_without_mhq_participants[,c(1,44:48)]

# make files for overlapping groups with supercontrols and merge
one_measure <- super_controls %>% 
  rename(One_Measure = Super_controls) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$One_Measure==1,c(1:2)])

two_measure <- super_controls %>% 
  rename(Two_Measures = Super_controls) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Two_Measures==1,c(1,3)])

three_measure <- super_controls %>% 
  rename(Three_Measures = Super_controls) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Three_Measures==1,c(1,4)])

four_measure <- super_controls %>% 
  rename(Four_Measures = Super_controls) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Four_Measures==1,c(1,5)])

five_measure <- super_controls %>% 
  rename(Five_Measures = Super_controls) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Five_Measures==1,c(1,6)])

overlapping_measures <- plyr::join_all(list(one_measure, two_measure, three_measure, four_measure, five_measure), by = "ID", type = 'full')

write.table(overlapping_measures, file="/file_path/overlapping_measures", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r prs_pheno_prep_overlapping_vs_58k_controls, include=FALSE}

# make file for number of measures in non-mhq participants vs mhq controls

one_measure <- mhq_controls %>% 
  rename(One_Measure = MHQ_controls_extra_screen) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$One_Measure==1,c(1:2)])

two_measure <- mhq_controls %>% 
  rename(Two_Measures = MHQ_controls_extra_screen) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Two_Measures==1,c(1,3)])

three_measure <- mhq_controls %>% 
  rename(Three_Measures = MHQ_controls_extra_screen) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Three_Measures==1,c(1,4)])

four_measure <- mhq_controls %>% 
  rename(Four_Measures = MHQ_controls_extra_screen) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Four_Measures==1,c(1,5)])

five_measure <- mhq_controls %>% 
  rename(Five_Measures = MHQ_controls_extra_screen) %>% 
  bind_rows(., no_mhq_cases_overlap[no_mhq_cases_overlap$Five_Measures==1,c(1,6)])

overlapping_measures_mhq_controls <- plyr::join_all(list(one_measure, two_measure, three_measure, four_measure, five_measure), by = "ID", type = 'full')

write.table(overlapping_measures_mhq_controls, file="/file_path/overlapping_measures_mhq_controls", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r make_pheno_file_overlapping_and_mhq, echo=FALSE, eval=FALSE}

# Make phenotype files with one to five measures and mhq

# 58k controls
overlapping_measures_mhq_controls <- read.table("/file_path/overlapping_measures_mhq_controls", header = T) 
dep_main_sub_cat_mhq_controls <- read.table("/file_path/dep_main_sub_cat_mhq_controls", header = T) 

mhq_58k <- dep_main_sub_cat_mhq_controls[,1:2] %>%
  na.omit()

one_to_five_and_mhq_58k <- full_join(overlapping_measures_mhq_controls,mhq_58k)

write.table(one_to_five_and_mhq_58k, "/file_path/one_to_five_and_mhq_58k", col.names=T, row.names=F, quote=F)

one_to_five_and_mhq_58k <- read.table("/file_path/one_to_five_and_mhq_58k", header = T) 

# 232k controls
overlapping_measures <- read.table("/file_path/overlapping_measures", header = T) 

dep_main_sub_cat <- read.table("/file_path/dep_main_sub_cat", header = T) 

mhq_232k <- dep_main_sub_cat[,1:2] %>%
  na.omit()

one_to_five_and_mhq_232k <- full_join(overlapping_measures,mhq_232k)

write.table(one_to_five_and_mhq_232k, "/file_path/one_to_five_and_mhq_232k", col.names=T, row.names=F, quote=F)

one_to_five_and_mhq_232k <- read.table("/file_path/one_to_five_and_mhq_232k", header = T) 

```

```{r prs_scripts_PRSice, echo=FALSE, eval=FALSE}

# MDD PRS predicting depression phenotypes, 58k controls, summary statistic including 23andMe
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/one_to_five_and_mhq_58k \
--pheno-col One_Measure,Two_Measures,Three_Measures,Four_Measures,Five_Measures,MHQ_depressed \
--ignore-fid \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/NEW_58k_with23

# MDD PRS predicting depression phenotypes, 58k controls, summary statistic excluding 23andMe
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR07.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/one_to_five_and_mhq_58k \
--pheno-col One_Measure,Two_Measures,Three_Measures,Four_Measures,Five_Measures,MHQ_depressed \
--ignore-fid \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/NEW_58k_no23

# MDD PRS predicting depression phenotypes, 232k controls, summary statistic including 23andMe
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/one_to_five_and_mhq_232k \
--pheno-col One_Measure,Two_Measures,Three_Measures,Four_Measures,Five_Measures,MHQ_depressed \
--ignore-fid \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/NEW_232k_with23

# MDD PRS predicting depression phenotypes, 232k controls, summary statistic excluding 23andMe
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/cleaned/DEPR07.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/one_to_five_and_mhq_232k \
--pheno-col One_Measure,Two_Measures,Three_Measures,Four_Measures,Five_Measures,MHQ_depressed \
--ignore-fid \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/prs/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/NEW_232k_no23

```

## PRS at all p-value thresholds

</br>

```{r prsice_plots_one_measure, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots at all p-value thresholds 

# Read all results
NEW_232k_no23.prsice <- read.table("/file_path/NEW_232k_no23.prsice", header = T) 
NEW_58k_no23.prsice <- read.table("/file_path/NEW_58k_no23.prsice", header = T) 
NEW_232k_with23.prsice <- read.table("/file_path/NEW_232k_with23.prsice", header = T) 
NEW_58k_with23.prsice <- read.table("/file_path/NEW_58k_with23.prsice", header = T) 

# One Measure

one_232k_no23 <- NEW_232k_no23.prsice[grep("One_Measure", NEW_232k_no23.prsice$Pheno), ]
one_232k_no23$Set <- rep("One Measure, Excl. 23andMe, Controls")
one_58k_no23 <- NEW_58k_no23.prsice[grep("One_Measure", NEW_58k_no23.prsice$Pheno), ]
one_58k_no23$Set <- rep("One Measure, Excl. 23andMe, MHQ Controls")
one_232k_with23 <- NEW_232k_with23.prsice[grep("One_Measure", NEW_232k_with23.prsice$Pheno), ] 
one_232k_with23$Set <- rep("One Measure, Incl. 23andMe, Controls")
one_58k_with23 <- NEW_58k_with23.prsice[grep("One_Measure", NEW_58k_with23.prsice$Pheno), ] 
one_58k_with23$Set <- rep("One Measure, Incl. 23andMe, MHQ Controls")

one <- list(one_232k_no23, one_58k_no23, one_232k_with23, one_58k_with23)

# format p-value column for plot

for (i in 1:(length(one))) {
  one[[i]]$print.p <- formatC(one[[i]]$P, digits = 2)
}
for (i in 1:(length(one))) {
  one[[i]]$print.p <- sub("e", "*x*10^", one[[i]]$print.p)
}

# Facet grid
one_facet <- rbind(data.frame(one[1]),
                            data.frame(one[2]),
                            data.frame(one[3]),
                            data.frame(one[4]))

plot_one_facet <- ggplot(one_facet, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set) +
  ylim(0,.025)

plot_one_facet

```

**Plot:** MDD PRS predicting One Measure cases vs controls. 

</br>

```{r prsice_plots_two_measure, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots at all p-value thresholds 

# Two Measures

two_232k_no23 <- NEW_232k_no23.prsice[grep("Two_Measures", NEW_232k_no23.prsice$Pheno), ]
two_232k_no23$Set <- rep("Two Measures, Excl. 23andMe, Controls")

two_58k_no23 <- NEW_58k_no23.prsice[grep("Two_Measures", NEW_58k_no23.prsice$Pheno), ]
two_58k_no23$Set <- rep("Two Measures, Excl. 23andMe, MHQ Controls")

two_232k_with23 <- NEW_232k_with23.prsice[grep("Two_Measures", NEW_232k_with23.prsice$Pheno), ] 
two_232k_with23$Set <- rep("Two Measures, Incl. 23andMe, Controls")

two_58k_with23 <- NEW_58k_with23.prsice[grep("Two_Measures", NEW_58k_with23.prsice$Pheno), ] 
two_58k_with23$Set <- rep("Two Measures, Incl. 23andMe, MHQ Controls")

two <- list(two_232k_no23, two_58k_no23, two_232k_with23, two_58k_with23)

# format p-value column for plot

for (i in 1:(length(two))) {
  two[[i]]$print.p <- formatC(two[[i]]$P, digits = 2)
}
for (i in 1:(length(two))) {
  two[[i]]$print.p <- sub("e", "*x*10^", two[[i]]$print.p)
}

# Facet grid
two_facet <- rbind(data.frame(two[1]),
                            data.frame(two[2]),
                            data.frame(two[3]),
                            data.frame(two[4]))

plot_two_facet <- ggplot(two_facet, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set) +
  ylim(0,.025)

plot_two_facet

```

**Plot:** MDD PRS predicting Two Measures cases vs controls.

</br>

```{r prsice_plots_three_measure, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots at all p-value thresholds 

# Three Measures

three_232k_no23 <- NEW_232k_no23.prsice[grep("Three_Measures", NEW_232k_no23.prsice$Pheno), ]
three_232k_no23$Set <- rep("Three Measures, Excl. 23andMe, Controls")

three_58k_no23 <- NEW_58k_no23.prsice[grep("Three_Measures", NEW_58k_no23.prsice$Pheno), ]
three_58k_no23$Set <- rep("Three Measures, Excl. 23andMe, MHQ Controls")

three_232k_with23 <- NEW_232k_with23.prsice[grep("Three_Measures", NEW_232k_with23.prsice$Pheno), ] 
three_232k_with23$Set <- rep("Three Measures, Incl. 23andMe, Controls")

three_58k_with23 <- NEW_58k_with23.prsice[grep("Three_Measures", NEW_58k_with23.prsice$Pheno), ] 
three_58k_with23$Set <- rep("Three Measures, Incl. 23andMe, MHQ Controls")

three <- list(three_232k_no23, three_58k_no23, three_232k_with23, three_58k_with23)

# format p-value column for plot

for (i in 1:(length(three))) {
  three[[i]]$print.p <- formatC(three[[i]]$P, digits = 2)
}
for (i in 1:(length(three))) {
  three[[i]]$print.p <- sub("e", "*x*10^", three[[i]]$print.p)
}

# Facet grid
three_facet <- rbind(data.frame(three[1]),
                            data.frame(three[2]),
                            data.frame(three[3]),
                            data.frame(three[4]))

plot_three_facet <- ggplot(three_facet, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set) +
  ylim(0,.025)

plot_three_facet

```

**Plot:** MDD PRS predicting Three Measures cases vs controls. 

</br>

```{r prsice_plots_four_measure, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots at all p-value thresholds 

# Four Measures

four_232k_no23 <- NEW_232k_no23.prsice[grep("Four_Measures", NEW_232k_no23.prsice$Pheno), ]
four_232k_no23$Set <- rep("Four Measures, Excl. 23andMe, Controls")

four_58k_no23 <- NEW_58k_no23.prsice[grep("Four_Measures", NEW_58k_no23.prsice$Pheno), ]
four_58k_no23$Set <- rep("Four Measures, Excl. 23andMe, MHQ Controls")

four_232k_with23 <- NEW_232k_with23.prsice[grep("Four_Measures", NEW_232k_with23.prsice$Pheno), ] 
four_232k_with23$Set <- rep("Four Measures, Incl. 23andMe, Controls")

four_58k_with23 <- NEW_58k_with23.prsice[grep("Four_Measures", NEW_58k_with23.prsice$Pheno), ] 
four_58k_with23$Set <- rep("Four Measures, Incl. 23andMe, MHQ Controls")

four <- list(four_232k_no23, four_58k_no23, four_232k_with23, four_58k_with23)

# format p-value column for plot

for (i in 1:(length(four))) {
  four[[i]]$print.p <- formatC(four[[i]]$P, digits = 2)
}
for (i in 1:(length(four))) {
  four[[i]]$print.p <- sub("e", "*x*10^", four[[i]]$print.p)
}

# Facet grid
four_facet <- rbind(data.frame(four[1]),
                            data.frame(four[2]),
                            data.frame(four[3]),
                            data.frame(four[4]))

plot_four_facet <- ggplot(four_facet, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set) +
  ylim(0,.025)

plot_four_facet

```

**Plot:** MDD PRS predicting Four Measures cases vs controls. 

</br>

```{r prsice_plots_five_measure, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots at all p-value thresholds 

# Five Measures

five_232k_no23 <- NEW_232k_no23.prsice[grep("Five_Measures", NEW_232k_no23.prsice$Pheno), ]
five_232k_no23$Set <- rep("Five Measures, Excl. 23andMe, Controls")

five_58k_no23 <- NEW_58k_no23.prsice[grep("Five_Measures", NEW_58k_no23.prsice$Pheno), ]
five_58k_no23$Set <- rep("Five Measures, Excl. 23andMe, MHQ Controls")

five_232k_with23 <- NEW_232k_with23.prsice[grep("Five_Measures", NEW_232k_with23.prsice$Pheno), ] 
five_232k_with23$Set <- rep("Five Measures, Incl. 23andMe, Controls")

five_58k_with23 <- NEW_58k_with23.prsice[grep("Five_Measures", NEW_58k_with23.prsice$Pheno), ] 
five_58k_with23$Set <- rep("Five Measures, Incl. 23andMe, MHQ Controls")

five <- list(five_232k_no23, five_58k_no23, five_232k_with23, five_58k_with23)

# format p-value column for plot

for (i in 1:(length(five))) {
  five[[i]]$print.p <- formatC(five[[i]]$P, digits = 2)
}
for (i in 1:(length(five))) {
  five[[i]]$print.p <- sub("e", "*x*10^", five[[i]]$print.p)
}

# Facet grid
five_facet <- rbind(data.frame(five[1]),
                            data.frame(five[2]),
                            data.frame(five[3]),
                            data.frame(five[4]))

plot_five_facet <- ggplot(five_facet, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set) +
  ylim(0,.025)

plot_five_facet

```

**Plot:** MDD PRS predicting Five Measures cases vs controls. 

</br>

```{r prsice_plots_mhq_measure, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots at all p-value thresholds 

# Lifetime Depression (MHQ)

mhq_232k_no23 <- NEW_232k_no23.prsice[grep("MHQ_depressed", NEW_232k_no23.prsice$Pheno), ]
mhq_232k_no23$Set <- rep("Lifetime Dep (MHQ), Excl. 23andMe, Controls")

mhq_58k_no23 <- NEW_58k_no23.prsice[grep("MHQ_depressed", NEW_58k_no23.prsice$Pheno), ]
mhq_58k_no23$Set <- rep("Lifetime Dep (MHQ), Excl. 23andMe, MHQ Controls")

mhq_232k_with23 <- NEW_232k_with23.prsice[grep("MHQ_depressed", NEW_232k_with23.prsice$Pheno), ] 
mhq_232k_with23$Set <- rep("Lifetime Dep (MHQ), Incl. 23andMe, Controls")

mhq_58k_with23 <- NEW_58k_with23.prsice[grep("MHQ_depressed", NEW_58k_with23.prsice$Pheno), ] 
mhq_58k_with23$Set <- rep("Lifetime Dep (MHQ), Incl. 23andMe, MHQ Controls")

mhq <- list(mhq_232k_no23, mhq_58k_no23, mhq_232k_with23, mhq_58k_with23)

# format p-value column for plot

for (i in 1:(length(mhq))) {
  mhq[[i]]$print.p <- formatC(mhq[[i]]$P, digits = 2)
}
for (i in 1:(length(mhq))) {
  mhq[[i]]$print.p <- sub("e", "*x*10^", mhq[[i]]$print.p)
}

# Facet grid
mhq_facet <- rbind(data.frame(mhq[1]),
                            data.frame(mhq[2]),
                            data.frame(mhq[3]),
                            data.frame(mhq[4]))

plot_mhq_facet <- ggplot(mhq_facet, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set) +
  ylim(0,.025)

plot_mhq_facet

```

**Plot:** MDD PRS predicting Lifetime Depression (MHQ) cases vs controls. 

</br>

## PRS summary

</br>

```{r plot_observed, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=15}

# combine data
NEW_232k_no23.summary <- read.table("/file_path/NEW_232k_no23.summary", header = T) 
NEW_232k_no23.summary$Group <- rep("232k Controls, No 23andMe")
NEW_58k_no23.summary <- read.table("/file_path/NEW_58k_no23.summary", header = T) 
NEW_58k_no23.summary$Group <- rep("58k Controls, No 23andMe")
NEW_232k_with23.summary <- read.table("/file_path/NEW_232k_with23.summary", header = T) 
NEW_232k_with23.summary$Group <- rep("232k Controls, With 23andMe")
NEW_58k_with23.summary <- read.table("/file_path/NEW_58k_with23.summary", header = T) 
NEW_58k_with23.summary$Group <- rep("58k Controls, With 23andMe")

new_overlapping <- rbind(NEW_232k_no23.summary,NEW_58k_no23.summary,NEW_232k_with23.summary,NEW_58k_with23.summary)

# Change empirical P values to scientific notation
new_overlapping$print.p <- format(new_overlapping$P, digits = 1)
new_overlapping$print.p <- sub("e", "*x*10^", new_overlapping$print.p)
new_overlapping$print.empirical <- format(new_overlapping$Empirical.P, digits = 1)
new_overlapping$print.empirical <- sub("e", "*x*10^", new_overlapping$print.empirical)

new_overlapping$Phenotype <- factor(new_overlapping$Phenotype, levels=c("One_Measure","Two_Measures","Three_Measures","Four_Measures","Five_Measures","MHQ_depressed"))
new_overlapping$Group <- factor(new_overlapping$Group, levels=c("232k Controls, No 23andMe","58k Controls, No 23andMe","232k Controls, With 23andMe","58k Controls, With 23andMe"))

## Plot
new_prs_overlap <- ggplot(data = new_overlapping, aes(x = Phenotype, y = PRS.R2.adj, fill = Group)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.9) +
  geom_text(aes(label=paste(print.p)), position=position_dodge(width=0.9), 
                              vjust=-1, hjust=0, size = 3, angle = 45, parse = T) +
  ylab(expression(paste("\nR"^"2", " Liability"))) +
  xlab("\nDepression Phenotype") +
  scale_fill_manual(values=c("firebrick2", "firebrick4", "royalblue1", "royalblue4"), labels = c("Controls / excl. 23andMe", "MHQ controls / excl. 23andMe", "Controls / incl. 23andMe", "MHQ controls / incl. 23andMe")) +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16),
        panel.grid.minor.x=element_blank(), 
        panel.grid.major.x=element_blank(), 
        panel.grid.minor.y=element_blank(), 
        panel.grid.major.y=element_blank(),
        panel.background = element_rect(fill = "white"),
        axis.line.y = element_line(color="grey"),
        axis.line.x = element_line(color="grey"),
        legend.title = element_text(size = 14),
        legend.text = element_text(size = 14)) +
  labs(fill = 'Control Group / Summary Statistics\n') +
  scale_x_discrete(labels = c("One Measure","Two Measures","Three Measures","Four Measures","Five Measures","Lifetime Depression\n(MHQ)")) +
  annotate("text", x = 1:6, y = Inf,  vjust=4, fontface=4, colour = "grey40",
           size=5, label = c(paste(format(sum(overlapping_measures$One_Measure==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Two_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Three_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Four_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Five_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(dep_main_sub_cat$MHQ_depressed==1, na.rm = T), big.mark = ",")))) + 
  annotate("text", x = 1:6, y = Inf,  vjust=2, fontface=4, colour = "grey30",
           size=5, label = rep("Cases", 6)) +
  ylim(0,0.06)

new_prs_overlap

```

**Plot:** MDD PRS predicting UKB MDD. OBSERVED p-values shown atop bars. 

</br>

```{r print_results_table, echo=FALSE, results = 'asis'}

# Print full table of results
new_overlapping_table <- new_overlapping[,c(14,1,3,12,8,5,9,11,13)]
new_overlapping_table$P <- format(new_overlapping_table$P, digits = 1)
new_overlapping_table$Empirical.P <- format(new_overlapping_table$Empirical.P, digits = 1)
new_overlapping_table$PRS.R2.adj <- percent(new_overlapping_table$PRS.R2.adj, digits = 2)
new_overlapping_table$Coefficient <- format(new_overlapping_table$Coefficient, digits = 2)

kable(new_overlapping_table, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  collapse_rows(columns = 1, valign = "top") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666", align = "centre")

```

</br>

# GWAS and Heritability

</br>

```{r residualise_all_pheno, eval=FALSE, include=FALSE}

# Residualise phenotypes for linear regression in BGENIE

control232ka <- fread("/file_path/dep_main_sub_cat", data.table=F)
control232kb <- fread("/file_path/overlapping_measures", data.table=F)

control232k <- merge(control232ka, control232kb, by = "ID", all = TRUE)
control232k$Four_Five_Measures <- ifelse(control232k$Four_Measures==1 | control232k$Five_Measures==1, 1, 0)
control232k <- control232k[,c(1:2,17:19,22)]

control58ka <- fread("/file_path/dep_main_sub_cat_mhq_controls", data.table=F)
control58kb <- fread("/file_path/overlapping_measures_mhq_controls", data.table=F)

control58k <- merge(control58ka, control58kb, by = "ID", all = TRUE)
control58k$Four_Five_Measures <- ifelse(control58k$Four_Measures==1 | control58k$Five_Measures==1, 1, 0)
control58k <- control58k[,c(1:2,17:19,22)]

# Regress covariates out of phenotypes

Cov <- fread("/file_path/covariates.txt", data.table=F)
Covariates <- Cov[,2:12]
names(Covariates) <- c("ID","age","sex","PC1","PC2","PC3","PC4","PC5","PC6","batch","centre")

mhq_232k_cov <- na.omit(merge(control232k[,1:2], Covariates))
one_232k_cov <- na.omit(merge(control232k[,c(1,3)], Covariates))
two_232k_cov <- na.omit(merge(control232k[,c(1,4)], Covariates))
three_232k_cov <- na.omit(merge(control232k[,c(1,5)], Covariates))
four_five_232k_cov <- na.omit(merge(control232k[,c(1,6)], Covariates))

mhq_232k_resid <- data.frame(mhq_232k_cov$ID)
names(mhq_232k_resid) <- "ID"
mhq_232k_resid$mhq_232k_resid <- with(mhq_232k_cov, residuals(glm(MHQ_depressed ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

one_232k_resid <- data.frame(one_232k_cov$ID)
names(one_232k_resid) <- "ID"
one_232k_resid$one_232k_resid <- with(one_232k_cov, residuals(glm(One_Measure ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

two_232k_resid <- data.frame(two_232k_cov$ID)
names(two_232k_resid) <- "ID"
two_232k_resid$two_232k_resid <- with(two_232k_cov, residuals(glm(Two_Measures ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

three_232k_resid <- data.frame(three_232k_cov$ID)
names(three_232k_resid) <- "ID"
three_232k_resid$three_232k_resid <- with(three_232k_cov, residuals(glm(Three_Measures ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

four_five_232k_resid <- data.frame(four_five_232k_cov$ID)
names(four_five_232k_resid) <- "ID"
four_five_232k_resid$four_five_232k_resid <- with(four_five_232k_cov, residuals(glm(Four_Five_Measures ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

write.table(mhq_232k_resid, "/file_path/mhq_232k_resid", col.names=T, row.names=F, quote=F)
write.table(one_232k_resid, "/file_path/one_232k_resid", col.names=T, row.names=F, quote=F)
write.table(two_232k_resid, "/file_path/two_232k_resid", col.names=T, row.names=F, quote=F)
write.table(three_232k_resid, "/file_path/three_232k_resid", col.names=T, row.names=F, quote=F)
write.table(four_five_232k_resid, "/file_path/four_five_232k_resid", col.names=T, row.names=F, quote=F)

mhq_58k_cov <- na.omit(merge(control58k[,1:2], Covariates))
one_58k_cov <- na.omit(merge(control58k[,c(1,3)], Covariates))
two_58k_cov <- na.omit(merge(control58k[,c(1,4)], Covariates))
three_58k_cov <- na.omit(merge(control58k[,c(1,5)], Covariates))
four_five_58k_cov <- na.omit(merge(control58k[,c(1,6)], Covariates))

mhq_58k_resid <- data.frame(mhq_58k_cov$ID)
names(mhq_58k_resid) <- "ID"
mhq_58k_resid$mhq_58k_resid <- with(mhq_58k_cov, residuals(glm(MHQ_depressed ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

one_58k_resid <- data.frame(one_58k_cov$ID)
names(one_58k_resid) <- "ID"
one_58k_resid$one_58k_resid <- with(one_58k_cov, residuals(glm(One_Measure ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

two_58k_resid <- data.frame(two_58k_cov$ID)
names(two_58k_resid) <- "ID"
two_58k_resid$two_58k_resid <- with(two_58k_cov, residuals(glm(Two_Measures ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

three_58k_resid <- data.frame(three_58k_cov$ID)
names(three_58k_resid) <- "ID"
three_58k_resid$three_58k_resid <- with(three_58k_cov, residuals(glm(Three_Measures ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

four_five_58k_resid <- data.frame(four_five_58k_cov$ID)
names(four_five_58k_resid) <- "ID"
four_five_58k_resid$four_five_58k_resid <- with(four_five_58k_cov, residuals(glm(Four_Five_Measures ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 + as.factor(batch) + as.factor(centre), family="binomial"),type="pearson"))

write.table(mhq_58k_resid, "/file_path/mhq_58k_resid", col.names=T, row.names=F, quote=F)
write.table(one_58k_resid, "/file_path/one_58k_resid", col.names=T, row.names=F, quote=F)
write.table(two_58k_resid, "/file_path/two_58k_resid", col.names=T, row.names=F, quote=F)
write.table(three_58k_resid, "/file_path/three_58k_resid", col.names=T, row.names=F, quote=F)
write.table(four_five_58k_resid, "/file_path/four_five_58k_resid", col.names=T, row.names=F, quote=F)

```

```{bash bgenie_gwas_all_pheno, echo=FALSE, eval=FALSE}

# Prepare for phenotypes for BGENIE

dat1 <- read.table("/file_path/mhq_232k_resid", head=T)
dat2 <- read.table("/file_path/one_232k_resid", head=T)
dat3 <- read.table("/file_path/two_232k_resid", head=T)
dat4 <- read.table("/file_path/three_232k_resid", head=T)
dat5 <- read.table("/file_path/four_five_232k_resid", head=T)
dat6 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(dat1, dat2, dat3, dat4, dat5))

bgen_id <- read.table("/file_path/bgen_ordered_ids.txt", head=F)
names(bgen_id)[1] <- "ID"
dat <- merge(bgen_id, dat6, by = "ID", all.x=TRUE)
dat <- dat[order(match(dat$ID,bgen_id$ID)),] 
dat <- dat[,2:ncol(dat)]
write.table(dat, "/file_path/bgenie_232k_controls", col.names=T, row.names=F, quote=F, sep=" ")

dat1 <- read.table("/file_path/mhq_58k_resid", head=T)
dat2 <- read.table("/file_path/one_58k_resid", head=T)
dat3 <- read.table("/file_path/two_58k_resid", head=T)
dat4 <- read.table("/file_path/three_58k_resid", head=T)
dat5 <- read.table("/file_path/four_five_58k_resid", head=T)
dat6 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(dat1, dat2, dat3, dat4, dat5))

dat <- merge(bgen_id, dat6, by = "ID", all.x=TRUE)
dat <- dat[order(match(dat$ID,bgen_id$ID)),] 
dat <- dat[,2:ncol(dat)]
write.table(dat, "/file_path/bgenie_58k_controls", col.names=T, row.names=F, quote=F, sep=" ")

# Run BGENIE regression on phenotypes: 232k controls

for i in {1..22}; do qsub /file_path/bgenie_232k.sh $i; done

/file_path/bgenie/v1.2/bgenie_v1.2_static1 \
--bgen /file_path/imp_chr${1}_MAF1_INFO4_v1.bgen \
--pheno /file_path/bgenie_232k_controls \
--miss NA \
--scale_phenotypes \
--scale_genotypes \
--pvals \
--out /file_path/bgenie_232k_chr${1} \
--thread 8

# Run BGENIE regression on phenotypes: 232k controls

for i in {1..22}; do qsub /file_path/bgenie_58k.sh $i; done

/file_path/bgenie/v1.2/bgenie_v1.2_static1 \
--bgen /file_path/imp_chr${1}_MAF1_INFO4_v1.bgen \
--pheno /file_path/bgenie_58k_controls \
--miss NA \
--scale_phenotypes \
--scale_genotypes \
--pvals \
--out /file_path/bgenie_58k_chr${1} \
--thread 8

# Create separate summary stats for each depression phenotype

for chr in {1..22}; do
  for j in {0..4}; do 
    a=$(( 8 + (4 * $j)))
    b=$(( 9 + (4 * $j)))
    c=$(( 10 + (4 * $j)))
    d=$(( 11 + (4 * $j)))
      zcat /file_path/bgenie_232k_chr${chr}.gz |\
       awk '{print $1, $2, $3, $4, $5, $6, $7, $'$a', $'$b', $'$c', $'$d', 10^($'$d' * -1)}'\
       > /file_path/bgenie_232k_chr${chr}_pheno_${j}
  done
done


for chr in {1..22}; do
  for j in {0..4}; do 
    a=$(( 8 + (4 * $j)))
    b=$(( 9 + (4 * $j)))
    c=$(( 10 + (4 * $j)))
    d=$(( 11 + (4 * $j)))
      zcat /file_path/bgenie_58k_chr${chr}.gz |\
       awk '{print $1, $2, $3, $4, $5, $6, $7, $'$a', $'$b', $'$c', $'$d', 10^($'$d' * -1)}'\
       > /file_path/bgenie_58k_chr${chr}_pheno_${j}
  done
done

# Join chromosomes for each of five phenotypes

head -n1 bgenie_232k_chr1_pheno_0 > bgenie_sumstat_mhq_232k
for fname in bgenie_232k_chr*_pheno_0
do
    tail -n+2 $fname >> bgenie_sumstat_mhq_232k
done

head -n1 bgenie_232k_chr1_pheno_1 > bgenie_sumstat_one_232k
for fname in bgenie_232k_chr*_pheno_1
do
    tail -n+2 $fname >> bgenie_sumstat_one_232k
done

head -n1 bgenie_232k_chr1_pheno_2 > bgenie_sumstat_two_232k
for fname in bgenie_232k_chr*_pheno_2
do
    tail -n+2 $fname >> bgenie_sumstat_two_232k
done

head -n1 bgenie_232k_chr1_pheno_3 > bgenie_sumstat_three_232k
for fname in bgenie_232k_chr*_pheno_3
do
    tail -n+2 $fname >> bgenie_sumstat_three_232k
done

head -n1 bgenie_232k_chr1_pheno_4 > bgenie_sumstat_four_five_232k
for fname in bgenie_232k_chr*_pheno_4
do
    tail -n+2 $fname >> bgenie_sumstat_four_five_232k
done


head -n1 bgenie_58k_chr1_pheno_0 > bgenie_sumstat_mhq_58k
for fname in bgenie_58k_chr*_pheno_0
do
    tail -n+2 $fname >> bgenie_sumstat_mhq_58k
done

head -n1 bgenie_58k_chr1_pheno_1 > bgenie_sumstat_one_58k
for fname in bgenie_58k_chr*_pheno_1
do
    tail -n+2 $fname >> bgenie_sumstat_one_58k
done

head -n1 bgenie_58k_chr1_pheno_2 > bgenie_sumstat_two_58k
for fname in bgenie_58k_chr*_pheno_2
do
    tail -n+2 $fname >> bgenie_sumstat_two_58k
done

head -n1 bgenie_58k_chr1_pheno_3 > bgenie_sumstat_three_58k
for fname in bgenie_58k_chr*_pheno_3
do
    tail -n+2 $fname >> bgenie_sumstat_three_58k
done

head -n1 bgenie_58k_chr1_pheno_4 > bgenie_sumstat_four_five_58k
for fname in bgenie_58k_chr*_pheno_4
do
    tail -n+2 $fname >> bgenie_sumstat_four_five_58k
done

# Change column headings

# NB from bgenie wiki: In the regression model we code the first and second alleles as 0 and 1 respectively, so the beta coefficient refers to the effect of having an extra copy of the second allele.

pheno_mhq_232 <- fread("/file_path/bgenie_sumstat_mhq_232k", head=T)
pheno_one_232 <- fread("/file_path/bgenie_sumstat_one_232k", head=T)
pheno_two_232 <- fread("/file_path/bgenie_sumstat_two_232k", head=T)
pheno_three_232 <- fread("/file_path/bgenie_sumstat_three_232k", head=T)
pheno_four_five_232 <- fread("/file_path/bgenie_sumstat_four_five_232k", head=T)

pheno_mhq_58 <- fread("/file_path/bgenie_sumstat_mhq_58k", head=T)
pheno_one_58 <- fread("/file_path/bgenie_sumstat_one_58k", head=T)
pheno_two_58 <- fread("/file_path/bgenie_sumstat_two_58k", head=T)
pheno_three_58 <- fread("/file_path/bgenie_sumstat_three_58k", head=T)
pheno_four_five_58 <- fread("/file_path/bgenie_sumstat_four_five_58k", head=T)

colnames(pheno_mhq_232) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_one_232) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_two_232) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_three_232) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_four_five_232) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")

colnames(pheno_mhq_58) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_one_58) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_two_58) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_three_58) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(pheno_four_five_58) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")

    
write.table(pheno_mhq_58, gzfile("/file_path/bgenie_sumstat_mhq_58k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_one_58, gzfile("/file_path/bgenie_sumstat_one_58k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_two_58, gzfile("/file_path/bgenie_sumstat_two_58k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_three_58, gzfile("/file_path/bgenie_sumstat_three_58k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_four_five_58, gzfile("/file_path/bgenie_sumstat_four_five_58k.gz"), col.names=T, row.names=F, quote=F, sep=" ")

write.table(pheno_mhq_232, gzfile("/file_path/bgenie_sumstat_mhq_232k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_one_232, gzfile("/file_path/bgenie_sumstat_one_232k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_two_232, gzfile("/file_path/bgenie_sumstat_two_232k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_three_232, gzfile("/file_path/bgenie_sumstat_three_232k.gz"), col.names=T, row.names=F, quote=F, sep=" ")
write.table(pheno_four_five_232, gzfile("/file_path/bgenie_sumstat_four_five_232k.gz"), col.names=T, row.names=F, quote=F, sep=" ")

# Submit summary statistics to FUMA

```

```{bash bgenie_munge_all_pheno, echo=FALSE, eval=FALSE}

# munge summary statistics
/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_mhq_58k.gz \
--N 86806 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_mhq_58k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_mhq_232k.gz \
--N 261604 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_mhq_232k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_one_58k.gz \
--N 115159 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_one_58k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_one_232k.gz \
--N 289957 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_one_232k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_two_58k.gz \
--N 79288 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_two_58k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_two_232k.gz \
--N 254086 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_two_232k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_three_58k.gz \
--N 67561 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_three_58k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_three_232k.gz \
--N 242359 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_three_232k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_four_five_58k.gz \
--N 62710 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_four_five_58k

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_four_five_232k.gz \
--N 237508 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--out /file_path/munged_bgenie_sumstat_four_five_232k

```

```{bash bgenie_heritability_all_pheno, echo=FALSE, eval=FALSE}

# Calculate heritability 

# Calculate sample prevalences: 57818 controls, 232616 controls

# mhq 28983/(57818+28983) = 0.3339017
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_mhq_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.333 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_mhq_58k

# mhq 28983/(232616+28983) = 0.1107917
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_mhq_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.110 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_mhq_232k

# one 57326/(57818+57326) = 0.4978635
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_one_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.497 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_one_58k

# one 57326/(232616+57326) = 0.1977154
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_one_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.197 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_one_232k

# two 21468/(57818+21468) = 0.2707666
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_two_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.270 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_two_58k

# two 21468/(232616+21468) = 0.08449174
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_two_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.084 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_two_232k

# three 9739/(57818+9739) = 0.1441597
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_three_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.144 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_three_58k

# three 9739/(232616+9739) = 0.04018485
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_three_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.040 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_three_232k

# four/five (4247+642)/(57818+4247+642) = 0.07796578
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_four_five_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.077 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_four_five_58k

# four/five (4247+642)/(232616+4247+642) = 0.02058483
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_four_five_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.020 \
--pop-prev 0.15 \
--out h2_munged_bgenie_sumstat_four_five_232k

# Calculate on observed scale and transform using Lee and Wray script across population prevalences

# 58k controls

# mhq 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_mhq_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_mhq_58k

# one 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_one_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_one_58k

# two 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_two_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_two_58k

# three 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_three_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_three_58k

# four/five 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_four_five_58k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_four_five_58k

# 232k controls

# mhq 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_mhq_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_mhq_232k

# one 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_one_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_one_232k

# two 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_two_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_two_232k

# three 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_three_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_three_232k

# four/five 
/file_path/ldsc/ldsc.py \
--h2 munged_bgenie_sumstat_four_five_232k.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--out OBS_h2_munged_bgenie_sumstat_four_five_232k

```

```{r bgenie_make_heritability_results_table_58k, include=FALSE}

h2_mhq_58k <- read.table("/file_path/h2_munged_bgenie_sumstat_mhq_58k.log", sep = ',')
h2_one_58k <- read.table("/file_path/h2_munged_bgenie_sumstat_one_58k.log", sep = ',')
h2_two_58k <- read.table("/file_path/h2_munged_bgenie_sumstat_two_58k.log", sep = ',')
h2_three_58k <- read.table("/file_path/h2_munged_bgenie_sumstat_three_58k.log", sep = ',')
h2_four_five_58k <- read.table("/file_path/h2_munged_bgenie_sumstat_four_five_58k.log", sep = ',')

gwas_mhq_58k <- read.table("/file_path/munged_bgenie_sumstat_mhq_58k.log", sep = ',')
gwas_one_58k <- read.table("/file_path/munged_bgenie_sumstat_one_58k.log", sep = ',')
gwas_two_58k <- read.table("/file_path/munged_bgenie_sumstat_two_58k.log", sep = ',')
gwas_three_58k <- read.table("/file_path/munged_bgenie_sumstat_three_58k.log", sep = ',')
gwas_four_five_58k <- read.table("/file_path/munged_bgenie_sumstat_four_five_58k.log", sep = ',')

rowname <- data.frame("Parameter" = c("Ncase", "Ncontrol", "Covariates", "INFO", "MAF", "N SNPs GWAS", "N SNPs post-munge", "Mean chi^2", "Lambda GC", "Max chi^2", "Significant SNPs", "Pop Prev", "Sample Prev", "N SNPs post LD merge", "Liability scale h2 (SE)", "Lambda", "Mean Chi2", "Intercept (SE)", "Ratio (SE)"))

mhq_58k <- rbind(data.frame("MHQ" = c("28k", "58k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("MHQ" = gwas_mhq_58k[c(31,42,44:47),]),
                               data.frame("MHQ" = h2_mhq_58k[c(13:14,25,29:33),]))

one_58k <- rbind(data.frame("One" = c("57k", "58k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("One" = gwas_one_58k[c(31,42,44:47),]),
                               data.frame("One" = h2_one_58k[c(13:14,25,29:33),]))

two_58k <- rbind(data.frame("Two" = c("21k", "58k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("Two" = gwas_two_58k[c(31,42,44:47),]),
                               data.frame("Two" = h2_two_58k[c(13:14,25,29:33),]))

three_58k <- rbind(data.frame("Three" = c("10k", "58k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("Three" = gwas_three_58k[c(31,42,44:47),]),
                               data.frame("Three" = h2_three_58k[c(13:14,25,29:33),]))

four_five_58k <- rbind(data.frame("Four-Five" = c("5k", "58k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("Four-Five" = gwas_four_five_58k[c(31,42,44:47),]),
                               data.frame("Four-Five" = h2_four_five_58k[c(13:14,25,29:33),]))

heritability_table <- cbind(rowname, mhq_58k, one_58k, two_58k, three_58k, four_five_58k)

rownames(heritability_table) <- NULL

heritability_table[] <- lapply(heritability_table, function(x) gsub("Mean chi\\^2 = ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Lambda GC = ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Max chi\\^2 = ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Genome-wide significant SNPs \\(some may have been removed by filtering\\)\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("--pop-prev ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("--samp-prev ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("\\\\", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub(" SNPs remain\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Total Liability scale h2\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Lambda GC\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Mean Chi\\^2\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Intercept\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Ratio\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("\\(usually indicates GC correction\\)\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Read ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs from --sumstats file\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Writing summary statistics for ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_mhq_58k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_one_58k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_two_58k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_three_58k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_four_five_58k.sumstats.gz.", "", x))

heritability_table_58k <- heritability_table

```

Heritability statistics for depression cases using **58k controls**

```{r bgenie_print_heritability_table_all_pheno_58k, echo=FALSE, results = 'asis'}

kable(heritability_table_58k, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  pack_rows("Data Summary", 1, 5, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Munge Stats", 6, 11, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Heritability Stats", 12, 19, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

```{r bgenie_make_heritability_results_table_232k, include=FALSE}

h2_mhq_232k <- read.table("/file_path/h2_munged_bgenie_sumstat_mhq_232k.log", sep = ',')
h2_one_232k <- read.table("/file_path/h2_munged_bgenie_sumstat_one_232k.log", sep = ',')
h2_two_232k <- read.table("/file_path/h2_munged_bgenie_sumstat_two_232k.log", sep = ',')
h2_three_232k <- read.table("/file_path/h2_munged_bgenie_sumstat_three_232k.log", sep = ',')
h2_four_five_232k <- read.table("/file_path/h2_munged_bgenie_sumstat_four_five_232k.log", sep = ',')

gwas_mhq_232k <- read.table("/file_path/munged_bgenie_sumstat_mhq_232k.log", sep = ',')
gwas_one_232k <- read.table("/file_path/munged_bgenie_sumstat_one_232k.log", sep = ',')
gwas_two_232k <- read.table("/file_path/munged_bgenie_sumstat_two_232k.log", sep = ',')
gwas_three_232k <- read.table("/file_path/munged_bgenie_sumstat_three_232k.log", sep = ',')
gwas_four_five_232k <- read.table("/file_path/munged_bgenie_sumstat_four_five_232k.log", sep = ',')

rowname <- data.frame("Parameter" = c("Ncase", "Ncontrol", "Covariates", "INFO", "MAF", "N SNPs GWAS", "N SNPs post-munge", "Mean chi^2", "Lambda GC", "Max chi^2", "Significant SNPs", "Pop Prev", "Sample Prev", "N SNPs post LD merge", "Liability scale h2 (SE)", "Lambda", "Mean Chi2", "Intercept (SE)", "Ratio (SE)"))

mhq_232k <- rbind(data.frame("MHQ" = c("28k", "232k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("MHQ" = gwas_mhq_232k[c(31,42,44:47),]),
                               data.frame("MHQ" = h2_mhq_232k[c(13:14,25,29:33),]))

one_232k <- rbind(data.frame("One" = c("57k", "232k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("One" = gwas_one_232k[c(31,42,44:47),]),
                               data.frame("One" = h2_one_232k[c(13:14,25,29:33),]))

two_232k <- rbind(data.frame("Two" = c("21k", "232k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("Two" = gwas_two_232k[c(31,42,44:47),]),
                               data.frame("Two" = h2_two_232k[c(13:14,25,29:33),]))

three_232k <- rbind(data.frame("Three" = c("10k", "232k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("Three" = gwas_three_232k[c(31,42,44:47),]),
                               data.frame("Three" = h2_three_232k[c(13:14,25,29:33),]))

four_five_232k <- rbind(data.frame("Four-Five" = c("5k", "232k", "PC1-6, batch, centre", ">=0.9", ">=0.01")),
                               data.frame("Four-Five" = gwas_four_five_232k[c(31,42,44:47),]),
                               data.frame("Four-Five" = h2_four_five_232k[c(13:14,25,29:33),]))

heritability_table <- cbind(rowname, mhq_232k, one_232k, two_232k, three_232k, four_five_232k)

rownames(heritability_table) <- NULL

heritability_table[] <- lapply(heritability_table, function(x) gsub("Mean chi\\^2 = ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Lambda GC = ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Max chi\\^2 = ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Genome-wide significant SNPs \\(some may have been removed by filtering\\)\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("--pop-prev ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("--samp-prev ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("\\\\", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub(" SNPs remain\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Total Liability scale h2\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Lambda GC\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Mean Chi\\^2\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Intercept\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Ratio\\: ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("\\(usually indicates GC correction\\)\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Read ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs from --sumstats file\\.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("Writing summary statistics for ", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_mhq_232k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_one_232k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_two_232k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_three_232k.sumstats.gz.", "", x))
heritability_table[] <- lapply(heritability_table, function(x) gsub("SNPs \\(7353982 with nonmissing beta) to /file_path/munged_bgenie_sumstat_four_five_232k.sumstats.gz.", "", x))

heritability_table_232k <- heritability_table

```

Heritability statistics for depression cases using **232k controls**

```{r bgenie_print_heritability_table_all_pheno_232k, echo=FALSE, results = 'asis'}

kable(heritability_table_232k, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  pack_rows("Data Summary", 1, 5, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Munge Stats", 6, 11, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("Heritability Stats", 12, 19, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

```{r make_heritabilities_summary, echo=FALSE}

h2_58k <- tidyr::gather(heritability_table_58k[15,]) %>% 
  slice(-1) %>% 
  rename(Phenotype = key, Heritability = value) %>%
  separate(Heritability, c("Heritability", "SE"), " ")

h2_58k[] <- lapply(h2_58k, function(x) gsub("\\(", "", x))
h2_58k[] <- lapply(h2_58k, function(x) gsub("\\)", "", x))

h2_58k$Heritability <- as.numeric(as.character(h2_58k$Heritability))
h2_58k$SE <- as.numeric(as.character(h2_58k$SE))
h2_58k$Controls <- rep("58k Controls")

h2_232k <- tidyr::gather(heritability_table_232k[15,]) %>% 
  slice(-1) %>% 
  rename(Phenotype = key, Heritability = value) %>%
  separate(Heritability, c("Heritability", "SE"), " ")

h2_232k[] <- lapply(h2_232k, function(x) gsub("\\(", "", x))
h2_232k[] <- lapply(h2_232k, function(x) gsub("\\)", "", x))

h2_232k$Heritability <- as.numeric(as.character(h2_232k$Heritability))
h2_232k$SE <- as.numeric(as.character(h2_232k$SE))
h2_232k$Controls <- rep("232k Controls")

h2 <- rbind(h2_58k, h2_232k)

h2$Phenotype <- factor(h2$Phenotype, levels=c("One","Two","Three","Four.Five","MHQ"))

h2$lowci <- h2$Heritability-1.96*h2$SE
h2$highci <- h2$Heritability+1.96*h2$SE

# make column for supplementary table
h2$table <- paste(format(round(h2$lowci, digits = 3), nsmall = 3),"-",format(round(h2$highci, digits = 3), nsmall = 3))

```

</br>

```{r plot_heritabilities, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=15}

heritability2 <- ggplot(data = h2, aes(x = Phenotype, y = Heritability, fill = Controls)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.8) +
  ylab(expression(paste("\nh"^"2"[SNP]," Liability"))) + 
  xlab("\nDepression Phenotype") +
  scale_fill_manual(values=c("royalblue1", "royalblue4"), 
                    labels = c("Controls", "MHQ controls")) +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16),
        panel.grid.minor.x=element_blank(), 
        panel.grid.major.x=element_blank(), 
        panel.grid.minor.y=element_blank(), 
        panel.grid.major.y=element_blank(),
        panel.background = element_rect(fill = "white"),
        axis.line.y = element_line(color="grey"),
        axis.line.x = element_line(color="grey"),
        legend.title = element_text(size = 14),
        legend.text = element_text(size = 14)) +
  labs(fill = 'Control Group\n') +
  scale_x_discrete(labels = c("One Measure","Two Measures","Three Measures","Four & Five Measures","Lifetime Depression (MHQ)")) +
  
  annotate("text", x = 1:5, y = Inf,  vjust=4, fontface=4, colour = "grey40",
           size=5, label = c(paste(format(sum(overlapping_measures$One_Measure==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Two_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Three_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(overlapping_measures$Four_Measures==1, na.rm = T)+sum(overlapping_measures$Five_Measures==1, na.rm = T), big.mark = ",")),
                             paste(format(sum(dep_main_sub_cat$MHQ_depressed==1, na.rm = T), big.mark = ",")))) + 
  
  annotate("text", x = 1:5, y = Inf,  vjust=2, fontface=4, colour = "grey30",
           size=5, label = rep("Cases", 5)) +
  
  geom_errorbar(aes(ymin=lowci, ymax=highci), position = position_dodge(.8), width = 0.2) +
  ylim(0,0.5) 
  
heritability2

## save plot
# ggsave("heritability_pres.pdf", width = 11, height = 5)

```

**Bar Plot:** SNP heritability of depression phenotypes in the UKB. Error bars: 95% CIs. Transformation to liability scale using population prevalence 15%.

</br>

```{r simulate_pop_prev_58k, echo=FALSE}

# create function to calculate heritabilities over different pop prevalences

h2l_R2 <- function(k, r2, p) {
  # K baseline disease risk
  # r2 from a linear regression model attributable to genomic profile risk score
  # P proportion of sample that are cases
  # calculates proportion of variance explained on the liability scale
  #from ABC at http://www.complextraitgenomics.com/software/
  #Lee SH, Goddard ME, Wray NR, Visscher PM. (2012) A better coefficient of determination for genetic profile analysis. Genet Epidemiol. 2012 Apr;36(3):214-24.
  x= qnorm(1-k)
  z= dnorm(x)
  i=z/k
  C= k*(1-k)*k*(1-k)/(z^2*p*(1-p))
  theta= i*((p-k)/(1-k))*(i*((p-k)/(1-k))-x)
  h2l_R2 = C*r2 / (1 + C*theta*r2)
}

# run function across different population prevalences for each phenotype with 58k controls and put in df

h2l_R2_58k <- NULL

for (prev in seq(0.01,0.6,0.01)) {
  temp <- data.frame(prev=prev, 
                     h2_liab_one=h2l_R2 (k=prev, r2=0.1445, p=0.4978635),
                     h2_liab_two=h2l_R2 (k=prev, r2=0.1489, p=0.2707666),
                     h2_liab_three=h2l_R2 (k=prev, r2=0.1366, p=0.1441597),
                     h2_liab_four_five=h2l_R2 (k=prev, r2=0.0799, p=0.07796578),
                     h2_liab_mhq=h2l_R2 (k=prev, r2=0.0942, p=0.3339017))
  h2l_R2_58k <- rbind(h2l_R2_58k,temp)
}

# create function to calculate SE for heritabilities over different pop prevalences

se_h2l_R2 <- function(k,h2,se, p) {
  # K baseline disease risk
  # r2 from a linear regression model attributable to genomic profile risk score
  # P proportion of sample that are cases
  # calculates proportion of variance explained on the liability scale
  #from ABC at http://www.complextraitgenomics.com/software/
  #Lee SH, Goddard ME, Wray NR, Visscher PM. (2012) A better coefficient of determination for genetic profile analysis. Genet Epidemiol. 2012 Apr;36(3):214-24.

  #SE on the liability (From a Taylor series expansion)
  #var(h2l_r2) = [d(h2l_r2)/d(R2v)]^2*var(R2v) with d being calculus differentiation
  x= qnorm(1-k)
  z= dnorm(x)
  i=z/k
  C= k*(1-k)*k*(1-k)/(z^2*p*(1-p))
  theta= i*((p-k)/(1-k))*(i*((p-k)/(1-k))-x)
  se_h2l_R2 = C*(1-h2*theta)*se
}

# run function across different population prevalences for each phenotype with 58k controls and put in df

se_h2l_R2_58k <- NULL

for (prev in seq(0.01,0.6,0.01)) {
  temp <- data.frame(prev=prev, 
                     h2_liab_one=se_h2l_R2 (k=prev, h2=0.1445, se=0.0076, p=0.4978635),
                     h2_liab_two=se_h2l_R2 (k=prev, h2=0.1489, se=0.0087, p=0.2707666),
                     h2_liab_three=se_h2l_R2 (k=prev, h2=0.1366, se=0.0087, p=0.1441597),
                     h2_liab_four_five=se_h2l_R2 (k=prev, h2=0.0799, se=0.008, p=0.07796578),
                     h2_liab_mhq=se_h2l_R2 (k=prev, h2=0.0942, se=0.0068, p=0.3339017))
  se_h2l_R2_58k <- rbind(se_h2l_R2_58k,temp)
}

# melt dfs into format required for plotting and merge h2 with SE

h2_58k_pop_h2 <- reshape2::melt(h2l_R2_58k, id.var = 'prev')
colnames(h2_58k_pop_h2) <- c("Prevalence","Phenotype","Heritability")

h2_58k_pop_se <- reshape2::melt(se_h2l_R2_58k, id.var = 'prev')
colnames(h2_58k_pop_se) <- c("Prevalence","Phenotype","SE")

h2_58k_pop <- merge(h2_58k_pop_h2,h2_58k_pop_se)

# create CIs

h2_58k_pop$lowci95 <- h2_58k_pop$Heritability-1.96*h2_58k_pop$SE
h2_58k_pop$highci95 <- h2_58k_pop$Heritability+1.96*h2_58k_pop$SE

# create population = sample prevalence df for 58k

# 58k prevalences
# one = 50%
# two = 27%
# three = 14%
# four_five = 8%
# mhq = 33%

pop_prev_58k <- rbind(
  subset(h2_58k_pop, Phenotype=="h2_liab_one" & Prevalence==0.50),
  subset(h2_58k_pop, Phenotype=="h2_liab_two" & Prevalence==0.27),
  subset(h2_58k_pop, Phenotype=="h2_liab_three" & Prevalence==0.14),
  subset(h2_58k_pop, Phenotype=="h2_liab_four_five" & Prevalence==0.08),
  subset(h2_58k_pop, Phenotype=="h2_liab_mhq" & Prevalence==0.33)
)

```

```{r simulate_pop_prev_232k, echo=FALSE}

# run function across different population prevalences for each phenotype with 232k controls and put in df

h2l_R2_232k <- NULL

for (prev in seq(0.01,0.60,0.01)) {
  temp <- data.frame(prev=prev, 
                     h2_liab_one=h2l_R2 (k=prev, r2=0.0382, p=0.1977154),
                     h2_liab_two=h2l_R2 (k=prev, r2=0.0288, p=0.08449174),
                     h2_liab_three=h2l_R2 (k=prev, r2=0.0241, p=0.04018485),
                     h2_liab_four_five=h2l_R2 (k=prev, r2=0.0137, p=0.02058483),
                     h2_liab_mhq=h2l_R2 (k=prev, r2=0.0371, p=0.1107917))
  h2l_R2_232k <- rbind(h2l_R2_232k,temp)
}

# run function across different population prevalences for each phenotype with 232k controls and put in df

se_h2l_R2_232k <- NULL

for (prev in seq(0.01,0.60,0.01)) {
  temp <- data.frame(prev=prev, 
                     h2_liab_one=se_h2l_R2 (k=prev, h2=0.0382, se=0.0025, p=0.1977154),
                     h2_liab_two=se_h2l_R2 (k=prev, h2=0.0288, se=0.0024, p=0.08449174),
                     h2_liab_three=se_h2l_R2 (k=prev, h2=0.0241, se=0.0023, p=0.04018485),
                     h2_liab_four_five=se_h2l_R2 (k=prev, h2=0.0137, se=0.0019, p=0.02058483),
                     h2_liab_mhq=se_h2l_R2 (k=prev, h2=0.0371, se=0.0026, p=0.1107917))
  se_h2l_R2_232k <- rbind(se_h2l_R2_232k,temp)
}

# melt dfs into format required for plotting and merge h2 with SE

h2_232k_pop_h2 <- reshape2::melt(h2l_R2_232k, id.var = 'prev')
colnames(h2_232k_pop_h2) <- c("Prevalence","Phenotype","Heritability")

h2_232k_pop_se <- reshape2::melt(se_h2l_R2_232k, id.var = 'prev')
colnames(h2_232k_pop_se) <- c("Prevalence","Phenotype","SE")

h2_232k_pop <- merge(h2_232k_pop_h2,h2_232k_pop_se)

# create CIs

h2_232k_pop$lowci95 <- h2_232k_pop$Heritability-1.96*h2_232k_pop$SE
h2_232k_pop$highci95 <- h2_232k_pop$Heritability+1.96*h2_232k_pop$SE

# create population = sample prevalence df for 232k

# 232k prevalences
# one = 20%
# two = 8%
# three = 4%
# four_five = 2%
# mhq = 11%

pop_prev_232k <- rbind(
  subset(h2_232k_pop, Phenotype=="h2_liab_one" & Prevalence==0.20),
  subset(h2_232k_pop, Phenotype=="h2_liab_two" & Prevalence==0.08),
  subset(h2_232k_pop, Phenotype=="h2_liab_three" & Prevalence==0.04),
  subset(h2_232k_pop, Phenotype=="h2_liab_four_five" & Prevalence==0.02),
  subset(h2_232k_pop, Phenotype=="h2_liab_mhq" & Prevalence==0.11)
)

```

```{r plot_h2_sim_232k_60max_paper, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=9}

sim_h2_232k <- ggplot(h2_232k_pop, aes(x = Prevalence, y = Heritability, col = Phenotype)) +
  geom_line(size=2) +
  scale_color_manual(values=c("red", "orange", "green3", "blue","grey40"),
                    labels=c("One Measure","Two Measures","Three Measures","Four & Five Measures","Lifetime Depression (MHQ)"), name = "Depression Phenotype\n") +
  geom_vline(xintercept = 0.15, linetype="dotted", size=1) +
  geom_point(data = pop_prev_232k, shape = 17, size = 6, colour=c("red", "orange", "green3", "blue","grey40")) +
  labs(x = "\nPopulation Prevalence", y = expression(paste("\nh"^"2"[SNP]," Liability - Controls"))) +
  geom_ribbon(aes(ymin = h2_232k_pop$lowci95, ymax = h2_232k_pop$highci95), linetype=1, alpha=0.05) +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16),
        panel.grid.minor.x=element_blank(), 
        panel.grid.major.x=element_blank(), 
        panel.grid.minor.y=element_blank(), 
        panel.grid.major.y=element_blank(),
        axis.line.y = element_line(color="grey"),
        axis.line.x = element_line(color="grey"),
        legend.text = element_text(size=14),
        legend.title = element_text(size=14),
        panel.background = element_rect(fill = "white")) +
  xlim(0,0.6) +
  ylim(0,0.45)

sim_h2_232k

```

**Line Plot:** SNP heritability of depression phenotypes at population prevalences between 1-60%. Controls = 232k. Error ribbons: 95% CIs.  

</br>

```{r plot_h2_sim_58k_60max_paper, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=9}

sim_h2_58k <- ggplot(h2_58k_pop, aes(x = Prevalence, y = Heritability, col = Phenotype)) +
  geom_line(size=2) +
  scale_color_manual(values=c("red", "orange", "green3", "blue","grey40"),
                    labels=c("One Measure","Two Measures","Three Measures","Four & Five Measures","Lifetime Depression (MHQ)"), name = "Depression Phenotype\n") +
  geom_vline(xintercept = 0.15, linetype="dotted", size=1) +
  geom_point(data = pop_prev_58k, shape = 17, size = 6, colour=c("red", "orange", "green3", "blue","grey40")) +
  labs(x = "\nPopulation Prevalence", y = expression(paste("\nh"^"2"[SNP]," Liability - MHQ controls"))) +
  geom_ribbon(aes(ymin = h2_58k_pop$lowci95, ymax = h2_58k_pop$highci95), linetype=1, alpha=0.05) +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16),
        panel.grid.minor.x=element_blank(), 
        panel.grid.major.x=element_blank(), 
        panel.grid.minor.y=element_blank(), 
        panel.grid.major.y=element_blank(),
        axis.line.y = element_line(color="grey"),
        axis.line.x = element_line(color="grey"),
        legend.text = element_text(size=14),
        legend.title = element_text(size=14),
        panel.background = element_rect(fill = "white")) +
  xlim(0,0.6) +
  ylim(0,0.45)

sim_h2_58k

```

**Line Plot:** SNP heritability of depression phenotypes at population prevalences between 1-60%. Controls = 58k. Error ribbons: 95% CIs.  

</br>

# Genetic Correlations with PGC (with and without 23andMe)

</br>

```{bash munge_ldsc, echo=FALSE, eval=FALSE}

# Munge PGC with 23andMe

module load devtools/anaconda/2019.3-python2.7.16

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/DEPR06.gz \
--info-min 0.9 \
--signed-sumstats OR,1 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_pgc_full

# Munge PGC without 23andMe

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/DEPR07.gz \
--info-min 0.9 \
--signed-sumstats OR,1 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_pgc_no23andme

# Munge UKB summary stats 

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_mhq_58k.gz \
--N 86806 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_mhq_58k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_mhq_232k.gz \
--N 261604 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_mhq_232k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_one_58k.gz \
--N 115159 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_one_58k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_one_232k.gz \
--N 289957 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_one_232k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_two_58k.gz \
--N 79288 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_two_58k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_two_232k.gz \
--N 254086 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_two_232k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_three_58k.gz \
--N 67561 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_three_58k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_three_232k.gz \
--N 242359 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_three_232k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_four_five_58k.gz \
--N 62710 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_four_five_58k_merged

/file_path/ldsc/munge_sumstats.py \
--sumstats /file_path/bgenie_sumstat_four_five_232k.gz \
--N 237508 \
--info-min 0.9 \
--signed-sumstats BETA,0 \
--snp SNP \
--a1 A1 \
--a2 A2 \
--p P \
--info INFO \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/munged_bgenie_sumstat_four_five_232k_merged

```

```{bash rg_58k, echo=FALSE, eval=FALSE}

# Calculate rG using LDSC

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz,munged_bgenie_sumstat_one_58k_merged.sumstats.gz,munged_bgenie_sumstat_two_58k_merged.sumstats.gz,munged_bgenie_sumstat_three_58k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.33,0.50,0.27,0.14,0.08,0.27,0.32 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_mhq_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_one_58k_merged.sumstats.gz,munged_bgenie_sumstat_two_58k_merged.sumstats.gz,munged_bgenie_sumstat_three_58k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.50,0.27,0.14,0.08,0.27,0.32,0.33 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_one_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_two_58k_merged.sumstats.gz,munged_bgenie_sumstat_three_58k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz,munged_bgenie_sumstat_one_58k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.27,0.14,0.08,0.27,0.32,0.33,0.50 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_two_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_three_58k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz,munged_bgenie_sumstat_one_58k_merged.sumstats.gz,munged_bgenie_sumstat_two_58k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.14,0.08,0.27,0.32,0.33,0.50,0.27 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_three_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz,munged_bgenie_sumstat_one_58k_merged.sumstats.gz,munged_bgenie_sumstat_two_58k_merged.sumstats.gz,munged_bgenie_sumstat_three_58k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.08,0.27,0.32,0.33,0.50,0.27,0.14 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_four_five_new

/file_path/ldsc/ldsc.py \
--rg munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz,munged_bgenie_sumstat_one_58k_merged.sumstats.gz,munged_bgenie_sumstat_two_58k_merged.sumstats.gz,munged_bgenie_sumstat_three_58k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.27,0.32,0.33,0.50,0.27,0.14,0.08 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_pgc

/file_path/ldsc/ldsc.py \
--rg munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz,munged_bgenie_sumstat_one_58k_merged.sumstats.gz,munged_bgenie_sumstat_two_58k_merged.sumstats.gz,munged_bgenie_sumstat_three_58k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz,munged_pgc_full.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.32,0.33,0.50,0.27,0.14,0.08,0.27 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_58k_pgc_no23andme

```

```{r make_rg_table_58k, echo=FALSE}

# Read data and make table of results for genetic correlations using 58k controls

rg_58k_mhq_new.log <- read.table("/file_path/rg_58k_mhq_new.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_one_new.log <- read.table("/file_path/rg_58k_one_new.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_two_new.log <- read.table("/file_path/rg_58k_two_new.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_three_new.log <- read.table("/file_path/rg_58k_three_new.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_four_five_new.log <- read.table("/file_path/rg_58k_four_five_new.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_pgc.log <- read.table("/file_path/rg_58k_pgc.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_pgc_no23andme.log <- read.table("/file_path/rg_58k_pgc_no23andme.log", sep = ",", stringsAsFactors = FALSE)

rg_58k_mhq <- data.frame(x = rg_58k_mhq_new.log[193:198,], stringsAsFactors = F)
rg_58k_one <- data.frame(x = rg_58k_one_new.log[193:198,], stringsAsFactors = F)
rg_58k_two <- data.frame(x = rg_58k_two_new.log[193:198,], stringsAsFactors = F)
rg_58k_three <- data.frame(x = rg_58k_three_new.log[193:198,], stringsAsFactors = F)
rg_58k_four_five <- data.frame(x = rg_58k_four_five_new.log[193:198,], stringsAsFactors = F)
rg_58k_pgc <- data.frame(x = rg_58k_pgc.log[193:198,], stringsAsFactors = F)
rg_58k_pgc_no23andme <- data.frame(x = rg_58k_pgc_no23andme.log[193:198,], stringsAsFactors = F)

head <- data.frame(x = rg_58k_mhq_new.log[192,], stringsAsFactors = F)

rg_58k <- data.frame(rbind(
  strsplit(rg_58k_mhq$x, " +")[[1]],
  strsplit(rg_58k_mhq$x, " +")[[2]],
  strsplit(rg_58k_mhq$x, " +")[[3]],
  strsplit(rg_58k_mhq$x, " +")[[4]],
  strsplit(rg_58k_mhq$x, " +")[[5]],
  strsplit(rg_58k_mhq$x, " +")[[6]],
  strsplit(rg_58k_one$x, " +")[[1]],
  strsplit(rg_58k_one$x, " +")[[2]],
  strsplit(rg_58k_one$x, " +")[[3]],
  strsplit(rg_58k_one$x, " +")[[4]],
  strsplit(rg_58k_one$x, " +")[[5]],
  strsplit(rg_58k_one$x, " +")[[6]],
  strsplit(rg_58k_two$x, " +")[[1]],
  strsplit(rg_58k_two$x, " +")[[2]],
  strsplit(rg_58k_two$x, " +")[[3]],
  strsplit(rg_58k_two$x, " +")[[4]],
  strsplit(rg_58k_two$x, " +")[[5]],
  strsplit(rg_58k_two$x, " +")[[6]],
  strsplit(rg_58k_three$x, " +")[[1]],
  strsplit(rg_58k_three$x, " +")[[2]],
  strsplit(rg_58k_three$x, " +")[[3]],
  strsplit(rg_58k_three$x, " +")[[4]],
  strsplit(rg_58k_three$x, " +")[[5]],
  strsplit(rg_58k_three$x, " +")[[6]],
  strsplit(rg_58k_four_five$x, " +")[[1]],
  strsplit(rg_58k_four_five$x, " +")[[2]],
  strsplit(rg_58k_four_five$x, " +")[[3]],
  strsplit(rg_58k_four_five$x, " +")[[4]],
  strsplit(rg_58k_four_five$x, " +")[[5]],
  strsplit(rg_58k_four_five$x, " +")[[6]],
  strsplit(rg_58k_pgc$x, " +")[[1]],
  strsplit(rg_58k_pgc$x, " +")[[2]],
  strsplit(rg_58k_pgc$x, " +")[[3]],
  strsplit(rg_58k_pgc$x, " +")[[4]],
  strsplit(rg_58k_pgc$x, " +")[[5]],
  strsplit(rg_58k_pgc$x, " +")[[6]],
  strsplit(rg_58k_pgc_no23andme$x, " +")[[1]],
  strsplit(rg_58k_pgc_no23andme$x, " +")[[2]],
  strsplit(rg_58k_pgc_no23andme$x, " +")[[3]],
  strsplit(rg_58k_pgc_no23andme$x, " +")[[4]],
  strsplit(rg_58k_pgc_no23andme$x, " +")[[5]],
  strsplit(rg_58k_pgc_no23andme$x, " +")[[6]]
))
colnames(rg_58k) <- strsplit(head$x, " +")[[1]]

rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_bgenie_sumstat_mhq_58k_merged.sumstats.gz", "MHQ", x))
rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_bgenie_sumstat_one_58k_merged.sumstats.gz", "One", x))
rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_bgenie_sumstat_two_58k_merged.sumstats.gz", "Two", x))
rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_bgenie_sumstat_three_58k_merged.sumstats.gz", "Three", x))
rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_bgenie_sumstat_four_five_58k_merged.sumstats.gz", "Four_Five", x))
rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_pgc_full.sumstats.gz", "PGC with 23andMe", x))
rg_58k[] <- lapply(rg_58k, function(x) gsub("munged_pgc_no23andme.sumstats.gz", "PGC without 23andMe", x))

# remove duplicate lines for printing table

rg_58k_distinct <- distinct(rg_58k, rg, .keep_all= TRUE)
rg_58k_distinct$rg <- round(as.numeric(rg_58k_distinct$rg), digits = 2)
rg_58k_distinct$se <- round(as.numeric(rg_58k_distinct$se), digits = 3)

rg_58k_distinct$lowci <- format(round(rg_58k_distinct$rg-1.96*rg_58k_distinct$se, digits = 2), nsmall = 2)
rg_58k_distinct$highci <- format(round(rg_58k_distinct$rg+1.96*rg_58k_distinct$se, digits = 2), nsmall = 2)
rg_58k_distinct$ci_95 <- paste(rg_58k_distinct$lowci,"-",rg_58k_distinct$highci)

rg_58k_distinct$p <- as.numeric(rg_58k_distinct$p)
rg_58k_distinct$p <- formatC(rg_58k_distinct$p, format = "e", digits = 0)

rg_58k_distinct <- rg_58k_distinct[,c(1:5,16,6:13)]

```

Genetic correlation table: depression phenotypes using **58k controls**

```{r print_rg_table_58k, echo=FALSE, results = 'asis'}

# Print table

kable(rg_58k_distinct, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

```{r make_rg_matrix_58k, include=FALSE, results = 'asis'}

# Make matrix of results

rg_58k_new <- rg_58k[,c(2:4,7)]

rg_58k_new$rg <- ifelse(rg_58k_new$rg>1, 1, rg_58k_new$rg)

rg_58k_new[nrow(rg_58k_new) + 1,] = c("MHQ","MHQ","1","NA")
rg_58k_new[nrow(rg_58k_new) + 1,] = c("One","One","1","NA")
rg_58k_new[nrow(rg_58k_new) + 1,] = c("Two","Two","1","NA")
rg_58k_new[nrow(rg_58k_new) + 1,] = c("Three","Three","1","NA")
rg_58k_new[nrow(rg_58k_new) + 1,] = c("Four_Five","Four_Five","1","NA")
rg_58k_new[nrow(rg_58k_new) + 1,] = c("PGC with 23andMe","PGC with 23andMe","1","NA")
rg_58k_new[nrow(rg_58k_new) + 1,] = c("PGC without 23andMe","PGC without 23andMe","1","NA")

rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="MHQ", 1, 0)
rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="One", 2, rg_58k_new$sort_p1)
rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="Two", 3, rg_58k_new$sort_p1)
rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="Three", 4, rg_58k_new$sort_p1)
rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="Four_Five", 5, rg_58k_new$sort_p1)
rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="PGC with 23andMe", 6, rg_58k_new$sort_p1)
rg_58k_new$sort_p1 <- ifelse(rg_58k_new$p1=="PGC without 23andMe", 7, rg_58k_new$sort_p1)

rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="MHQ", 1, 0)
rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="One", 2, rg_58k_new$sort_p2)
rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="Two", 3, rg_58k_new$sort_p2)
rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="Three", 4, rg_58k_new$sort_p2)
rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="Four_Five", 5, rg_58k_new$sort_p2)
rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="PGC with 23andMe", 6, rg_58k_new$sort_p2)
rg_58k_new$sort_p2 <- ifelse(rg_58k_new$p2=="PGC without 23andMe", 7, rg_58k_new$sort_p2)

rg_58k_new <- rg_58k_new[order(rg_58k_new$sort_p1, rg_58k_new$sort_p2),] 

rg_58k_matrix <- matrix(rg_58k_new$rg, nrow = 7, byrow = TRUE)
rg_58k_matrix <- mapply(rg_58k_matrix, FUN = as.numeric)
rg_58k_matrix <- matrix(data=rg_58k_matrix, ncol=7, nrow=7)
row.names(rg_58k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 
colnames(rg_58k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 

p_58k_matrix <- matrix(rg_58k_new$p, nrow = 7, byrow = TRUE)
p_58k_matrix <- mapply(p_58k_matrix, FUN = as.numeric)
p_58k_matrix <- matrix(data=p_58k_matrix, ncol=7, nrow=7)
row.names(p_58k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 
colnames(p_58k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 

```

```{r print_rg_matrix_58k, echo=FALSE, results = 'asis', fig.height=6, fig.width=8}

# Print correlation plot

corrplot(rg_58k_matrix, type = "upper", tl.col = "navy", tl.srt = 45, addCoef.col = "white")

```

Genetic correlation matrix: depression phenotypes using **58k controls**
        
</br>

```{bash rg_232k, echo=FALSE, eval=FALSE}

# Calculate rG using LDSC

module load devtools/anaconda/2019.3-python2.7.16

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz,munged_bgenie_sumstat_one_232k_merged.sumstats.gz,munged_bgenie_sumstat_two_232k_merged.sumstats.gz,munged_bgenie_sumstat_three_232k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.11,0.20,0.08,0.04,0.02,0.27,0.32 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_mhq_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_one_232k_merged.sumstats.gz,munged_bgenie_sumstat_two_232k_merged.sumstats.gz,munged_bgenie_sumstat_three_232k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.20,0.08,0.04,0.02,0.27,0.32,0.11 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_one_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_two_232k_merged.sumstats.gz,munged_bgenie_sumstat_three_232k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz,munged_bgenie_sumstat_one_232k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.08,0.04,0.02,0.27,0.32,0.11,0.20 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_two_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_three_232k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz,munged_bgenie_sumstat_one_232k_merged.sumstats.gz,munged_bgenie_sumstat_two_232k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.04,0.02,0.27,0.32,0.11,0.20,0.08 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_three_new

/file_path/ldsc/ldsc.py \
--rg munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz,munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz,munged_bgenie_sumstat_one_232k_merged.sumstats.gz,munged_bgenie_sumstat_two_232k_merged.sumstats.gz,munged_bgenie_sumstat_three_232k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.02,0.27,0.32,0.11,0.20,0.08,0.04 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_four_five_new

/file_path/ldsc/ldsc.py \
--rg munged_pgc_full.sumstats.gz,munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz,munged_bgenie_sumstat_one_232k_merged.sumstats.gz,munged_bgenie_sumstat_two_232k_merged.sumstats.gz,munged_bgenie_sumstat_three_232k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.27,0.32,0.11,0.20,0.08,0.04,0.02 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_pgc

/file_path/ldsc/ldsc.py \
--rg munged_pgc_no23andme.sumstats.gz,munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz,munged_bgenie_sumstat_one_232k_merged.sumstats.gz,munged_bgenie_sumstat_two_232k_merged.sumstats.gz,munged_bgenie_sumstat_three_232k_merged.sumstats.gz,munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz,munged_pgc_full.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.32,0.11,0.20,0.08,0.04,0.02,0.27 \
--pop-prev 0.15,0.15,0.15,0.15,0.15,0.15,0.15 \
--out rg_232k_pgc_no23andme

```

```{r make_rg_table_232k, echo=FALSE}

# Make table of results

rg_232k_mhq_new.log <- read.table("/file_path/rg_232k_mhq_new.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_one_new.log <- read.table("/file_path/rg_232k_one_new.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_two_new.log <- read.table("/file_path/rg_232k_two_new.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_three_new.log <- read.table("/file_path/rg_232k_three_new.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_four_five_new.log <- read.table("/file_path/rg_232k_four_five_new.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_pgc.log <- read.table("/file_path/rg_232k_pgc.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_pgc_no23andme.log <- read.table("/file_path/rg_232k_pgc_no23andme.log", sep = ",", stringsAsFactors = FALSE)

rg_232k_mhq <- data.frame(x = rg_232k_mhq_new.log[193:198,], stringsAsFactors = F)
rg_232k_one <- data.frame(x = rg_232k_one_new.log[193:198,], stringsAsFactors = F)
rg_232k_two <- data.frame(x = rg_232k_two_new.log[193:198,], stringsAsFactors = F)
rg_232k_three <- data.frame(x = rg_232k_three_new.log[193:198,], stringsAsFactors = F)
rg_232k_four_five <- data.frame(x = rg_232k_four_five_new.log[193:198,], stringsAsFactors = F)
rg_232k_pgc <- data.frame(x = rg_232k_pgc.log[193:198,], stringsAsFactors = F)
rg_232k_pgc_no23andme <- data.frame(x = rg_232k_pgc_no23andme.log[193:198,], stringsAsFactors = F)

head <- data.frame(x = rg_232k_mhq_new.log[192,], stringsAsFactors = F)

rg_232k <- data.frame(rbind(
  strsplit(rg_232k_mhq$x, " +")[[1]],
  strsplit(rg_232k_mhq$x, " +")[[2]],
  strsplit(rg_232k_mhq$x, " +")[[3]],
  strsplit(rg_232k_mhq$x, " +")[[4]],
  strsplit(rg_232k_mhq$x, " +")[[5]],
  strsplit(rg_232k_mhq$x, " +")[[6]],
  strsplit(rg_232k_one$x, " +")[[1]],
  strsplit(rg_232k_one$x, " +")[[2]],
  strsplit(rg_232k_one$x, " +")[[3]],
  strsplit(rg_232k_one$x, " +")[[4]],
  strsplit(rg_232k_one$x, " +")[[5]],
  strsplit(rg_232k_one$x, " +")[[6]],
  strsplit(rg_232k_two$x, " +")[[1]],
  strsplit(rg_232k_two$x, " +")[[2]],
  strsplit(rg_232k_two$x, " +")[[3]],
  strsplit(rg_232k_two$x, " +")[[4]],
  strsplit(rg_232k_two$x, " +")[[5]],
  strsplit(rg_232k_two$x, " +")[[6]],
  strsplit(rg_232k_three$x, " +")[[1]],
  strsplit(rg_232k_three$x, " +")[[2]],
  strsplit(rg_232k_three$x, " +")[[3]],
  strsplit(rg_232k_three$x, " +")[[4]],
  strsplit(rg_232k_three$x, " +")[[5]],
  strsplit(rg_232k_three$x, " +")[[6]],
  strsplit(rg_232k_four_five$x, " +")[[1]],
  strsplit(rg_232k_four_five$x, " +")[[2]],
  strsplit(rg_232k_four_five$x, " +")[[3]],
  strsplit(rg_232k_four_five$x, " +")[[4]],
  strsplit(rg_232k_four_five$x, " +")[[5]],
  strsplit(rg_232k_four_five$x, " +")[[6]],
  strsplit(rg_232k_pgc$x, " +")[[1]],
  strsplit(rg_232k_pgc$x, " +")[[2]],
  strsplit(rg_232k_pgc$x, " +")[[3]],
  strsplit(rg_232k_pgc$x, " +")[[4]],
  strsplit(rg_232k_pgc$x, " +")[[5]],
  strsplit(rg_232k_pgc$x, " +")[[6]],
  strsplit(rg_232k_pgc_no23andme$x, " +")[[1]],
  strsplit(rg_232k_pgc_no23andme$x, " +")[[2]],
  strsplit(rg_232k_pgc_no23andme$x, " +")[[3]],
  strsplit(rg_232k_pgc_no23andme$x, " +")[[4]],
  strsplit(rg_232k_pgc_no23andme$x, " +")[[5]],
  strsplit(rg_232k_pgc_no23andme$x, " +")[[6]]
))
colnames(rg_232k) <- strsplit(head$x, " +")[[1]]

rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_bgenie_sumstat_mhq_232k_merged.sumstats.gz", "MHQ", x))
rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_bgenie_sumstat_one_232k_merged.sumstats.gz", "One", x))
rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_bgenie_sumstat_two_232k_merged.sumstats.gz", "Two", x))
rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_bgenie_sumstat_three_232k_merged.sumstats.gz", "Three", x))
rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_bgenie_sumstat_four_five_232k_merged.sumstats.gz", "Four_Five", x))
rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_pgc_full.sumstats.gz", "PGC with 23andMe", x))
rg_232k[] <- lapply(rg_232k, function(x) gsub("munged_pgc_no23andme.sumstats.gz", "PGC without 23andMe", x))

# remove duplicate lines for printing table

rg_232k_distinct <- distinct(rg_232k, rg, .keep_all= TRUE)
rg_232k_distinct$rg <- round(as.numeric(rg_232k_distinct$rg), digits = 2)
rg_232k_distinct$se <- round(as.numeric(rg_232k_distinct$se), digits = 3)

rg_232k_distinct$lowci <- format(round(rg_232k_distinct$rg-1.96*rg_232k_distinct$se, digits = 2), nsmall = 2)
rg_232k_distinct$highci <- format(round(rg_232k_distinct$rg+1.96*rg_232k_distinct$se, digits = 2), nsmall = 2)
rg_232k_distinct$ci_95 <- paste(rg_232k_distinct$lowci,"-",rg_232k_distinct$highci)

rg_232k_distinct$p <- as.numeric(rg_232k_distinct$p)
rg_232k_distinct$p <- formatC(rg_232k_distinct$p, format = "e", digits = 0)

rg_232k_distinct <- rg_232k_distinct[,c(1:5,16,6:13)]

```

Genetic correlation table: depression phenotypes using **232k controls**

```{r print_rg_table_232k, echo=FALSE, results = 'asis'}

# Print table of results

kable(rg_232k_distinct, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

```{r make_rg_matrix_232k, include=FALSE, results = 'asis'}

# Make matrix of rg

rg_232k_new <- rg_232k[,c(2:4,7)]

rg_232k_new$rg <- ifelse(rg_232k_new$rg>1, 1, rg_232k_new$rg)

rg_232k_new[nrow(rg_232k_new) + 1,] = c("MHQ","MHQ","1","NA")
rg_232k_new[nrow(rg_232k_new) + 1,] = c("One","One","1","NA")
rg_232k_new[nrow(rg_232k_new) + 1,] = c("Two","Two","1","NA")
rg_232k_new[nrow(rg_232k_new) + 1,] = c("Three","Three","1","NA")
rg_232k_new[nrow(rg_232k_new) + 1,] = c("Four_Five","Four_Five","1","NA")
rg_232k_new[nrow(rg_232k_new) + 1,] = c("PGC with 23andMe","PGC with 23andMe","1","NA")
rg_232k_new[nrow(rg_232k_new) + 1,] = c("PGC without 23andMe","PGC without 23andMe","1","NA")

rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="MHQ", 1, 0)
rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="One", 2, rg_232k_new$sort_p1)
rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="Two", 3, rg_232k_new$sort_p1)
rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="Three", 4, rg_232k_new$sort_p1)
rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="Four_Five", 5, rg_232k_new$sort_p1)
rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="PGC with 23andMe", 6, rg_232k_new$sort_p1)
rg_232k_new$sort_p1 <- ifelse(rg_232k_new$p1=="PGC without 23andMe", 7, rg_232k_new$sort_p1)

rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="MHQ", 1, 0)
rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="One", 2, rg_232k_new$sort_p2)
rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="Two", 3, rg_232k_new$sort_p2)
rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="Three", 4, rg_232k_new$sort_p2)
rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="Four_Five", 5, rg_232k_new$sort_p2)
rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="PGC with 23andMe", 6, rg_232k_new$sort_p2)
rg_232k_new$sort_p2 <- ifelse(rg_232k_new$p2=="PGC without 23andMe", 7, rg_232k_new$sort_p2)

rg_232k_new <- rg_232k_new[order(rg_232k_new$sort_p1, rg_232k_new$sort_p2),] 

rg_232k_matrix <- matrix(rg_232k_new$rg, nrow = 7, byrow = TRUE)
rg_232k_matrix <- mapply(rg_232k_matrix, FUN = as.numeric)
rg_232k_matrix <- matrix(data=rg_232k_matrix, ncol=7, nrow=7)
row.names(rg_232k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 
colnames(rg_232k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 

p_232k_matrix <- matrix(rg_232k_new$p, nrow = 7, byrow = TRUE)
p_232k_matrix <- mapply(p_232k_matrix, FUN = as.numeric)
p_232k_matrix <- matrix(data=p_232k_matrix, ncol=7, nrow=7)
row.names(p_232k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 
colnames(p_232k_matrix) <- c("MHQ","One","Two","Three","Four & Five","PGC incl. 23andMe","PGC excl. 23andMe") 

```

```{r print_rg_matrix_232k, echo=FALSE, results = 'asis', fig.height=6, fig.width=8}

# Print correlation matrix

corrplot(rg_232k_matrix, type = "upper", tl.col = "navy", tl.srt = 45, addCoef.col = "white")

```

Genetic correlation matrix: depression phenotypes using **232k controls**

</br>

```{r make_df_for_plot, include=FALSE, results = 'asis'}

# Make dataframe for plotting rG between UKB depression phenotypes and PGC depression

rg_232k_pgc <- rg_232k_distinct[grep("PGC", rg_232k_distinct$p2), ]
rg_232k_pgc <- rg_232k_pgc[-grep("PGC", rg_232k_pgc$p1), c(2:5)]
rg_232k_pgc$p2 <- gsub("PGC with 23andMe", "PGC with 23andMe semi-screened", rg_232k_pgc$p2)
rg_232k_pgc$p2 <- gsub("PGC without 23andMe", "PGC without 23andMe semi-screened", rg_232k_pgc$p2)

rg_58k_pgc <- rg_58k_distinct[grep("PGC", rg_58k_distinct$p2), ]
rg_58k_pgc <- rg_58k_pgc[-grep("PGC", rg_58k_pgc$p1), c(2:5)]
rg_58k_pgc$p2 <- gsub("PGC with 23andMe", "PGC with 23andMe screened", rg_58k_pgc$p2)
rg_58k_pgc$p2 <- gsub("PGC without 23andMe", "PGC without 23andMe screened", rg_58k_pgc$p2)

corr <- rbind(rg_232k_pgc,rg_58k_pgc)
corr$p1  = factor(corr$p1, levels=c("One", "Two", "Three", "Four_Five", "MHQ"))
corr$p2  = factor(corr$p2, levels=c("PGC without 23andMe semi-screened", "PGC without 23andMe screened", "PGC with 23andMe semi-screened", "PGC with 23andMe screened"))

```

```{r new_plot, echo=FALSE, results = 'asis', fig.height=6, fig.width=15}

# make plot for rG between UKB depression phenotypes and PGC depression

rg_plot <- ggplot(corr, aes(x = p1, y = rg)) +
  geom_pointrange(aes(ymin = corr$rg-1.96*corr$se, ymax = corr$rg+1.96*corr$se,
                      color = p2), position = position_dodge(width = 0.5), size=1.5) +
   scale_x_discrete(labels = c("One Measure","Two Measures","Three Measures","Four & Five Measures","Lifetime Depression\n(MHQ)")) +
  ylab(expression(paste("\nr"[G], " with PGC Depression Phenotypes"))) +
  xlab("\nDepression Phenotype") +
  scale_color_manual(name = "Control Group / Summary Statistics\n", values=c("firebrick2", "firebrick4", "royalblue1", "royalblue4"), labels = c("Controls / excl. 23andMe", "MHQ Controls / excl. 23andMe", "Controls / incl. 23andMe", "MHQ Controls / incl. 23andMe")) +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16),
        panel.grid.minor.x=element_blank(),
        panel.grid.major.x=element_blank(),
        panel.grid.minor.y=element_blank(),
        panel.grid.major.y=element_line(size = 1, linetype = 'dotted', colour = "grey"),
        panel.background = element_rect(fill = "white"),
        axis.line.y = element_line(color="grey"),
        axis.line.x = element_line(color="grey"),
        legend.title = element_text(size = 14),
        legend.text = element_text(size = 14)) +
  ylim(0,NA)

rg_plot

```

</br>

# Any and stringent depression

</br>

```{r make_any_stringent_depression_phenotypes, include=FALSE}

# Read dfs containing MHQ and 1-5 measures of depression

control232ka <- fread("/file_path/dep_main_sub_cat", data.table=F)
control232kb <- fread("/file_path/overlapping_measures", data.table=F)

# merge dfs

control232k <- merge(control232ka, control232kb, by = "ID", all = TRUE)

# Make 'any' depression (termed 'broad' in df) by selecting MHQ and those with 2 or more measures

control232k$Broad <- ifelse(
  control232k$MHQ_depressed==1 |
  control232k$Two_Measures==1 |
  control232k$Three_Measures==1 |
  control232k$Four_Measures==1 |
  control232k$Five_Measures==1,
  1,0
)

# Make 'stringent' depression (termed 'narrow' in df) by selecting those with 3 or more measures

control232k$Narrow <- ifelse(
  control232k$Three_Measures==1 |
  control232k$Four_Measures==1 |
  control232k$Five_Measures==1,
  1,0
)

# Write to file

write.table(control232k, "/file_path/dep_broad_narrow", col.names=T, row.names=F, quote=F)

```

```{r print_broad_narrow_table, echo=FALSE, results = 'asis'}

# Print summary tables for any ("broad") and stringent ("narrow") depression. These will be used in subsequent study: https://www.medrxiv.org/content/10.1101/2020.12.08.20242495v2

broad_narrow <- data.frame(
  "Phenotype" = c("Broad","Narrow","Controls"),
  "Number" = c(sum(control232k$Broad==1, na.rm = T),sum(control232k$Narrow==1, na.rm = T),sum(control232k$Broad==0, na.rm = T))
)

kable(broad_narrow, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

# Demographic information

</br>

```{bash print_demographic_columns, echo=FALSE, eval=FALSE}

Find required columns in field finder (/file_path/phenotypes) and print from phenotype file using bash

Demographics
8329   f.21022.0.0    Age.at.recruitment.0.0
1655   f.31.0.0       Sex.0.0
1943   f.189.0.0      Townsend.deprivation.index.at.recruitment.0.0
8317   f.21001.0.0    Body.mass.index.BMI.0.0
7671   f.20116.0.0    Smoking.status.0.0
2518   f.2178.0.0     Overall.health.rating.0.0

cd /file_path
awk '{print $1,$8329,$1655,$1943,$8317,$7671,$2518}' /file_path/ukb18177_glanville_phenotypes.txt > demographics_health

```

```{r demographics_table_main_cat, echo=FALSE, results = 'asis'}

# Prepare table with demographic information for controls and depression according to different measures

# Read demographics
demographics <- read.table("/file_path/demographics_health", header = T)
colnames(demographics) <- c("ID","Age","Sex","SES","BMI","Current Smoker","Health Rating")

# Recode smoking: change to 1 = current, 0 = not current (never or previous), NA = prefer not to answer
# Smoking Coding
# -3	Prefer not to answer
# 0	Never
# 1	Previous
# 2	Current
demographics$`Current Smoker` <- ifelse(demographics$`Current Smoker`==-3, NA, demographics$`Current Smoker`)
demographics$`Current Smoker` <- ifelse(demographics$`Current Smoker`==1, 0, demographics$`Current Smoker`)                                 
demographics$`Current Smoker` <- ifelse(demographics$`Current Smoker`==2, 1, demographics$`Current Smoker`)

# Recode health: change to -1 and -3 to NA
# 1 = Excellent
# 2 = Good
# 3 = Fair
# 4 = Poor
# -1 = Do not know
# -3 = Prefer not to answer
demographics$`Health Rating` <- ifelse(demographics$`Health Rating`==-3 | demographics$`Health Rating`==-1, NA, demographics$`Health Rating`)

# Read phenotype file with 232k and 58k cases
main_cat_232k <- read.table("/file_path/dep_main_sub_cat", header = T)
main_cat_232k$any_dep <- ifelse(rowSums(main_cat_232k[,2:ncol(main_cat_232k)], na.rm = T) > 0, 1, 0)
main_cat_58k <- read.table("/file_path/dep_main_sub_cat_mhq_controls", header = T)
main_cat_58k$any_dep <- ifelse(rowSums(main_cat_58k[,2:ncol(main_cat_58k)], na.rm = T) > 0, 1, 0)

# Make semi-screened and screened controls dfs and merge with demographics
semi_screened <- main_cat_232k %>%
  dplyr::select(ID = ID, semi_screened = any_dep) %>%
  filter(semi_screened == 0) %>%
  left_join(., demographics, by="ID")

screened <- main_cat_58k %>%
  dplyr::select(ID = ID, screened = any_dep) %>%
  filter(screened == 0) %>%
  left_join(., demographics, by="ID")

# Make main categories dfs and merge with demographics

icd <- main_cat_232k %>%
  dplyr::select(ID, ICD_any_depression) %>%
  filter(ICD_any_depression == 1) %>%
  left_join(., demographics, by="ID")

sr <- main_cat_232k %>%
  dplyr::select(ID, SR_depression) %>%
  filter(SR_depression == 1) %>%
  left_join(., demographics, by="ID")

smith <- main_cat_232k %>%
  dplyr::select(ID, Smith_Depression) %>%
  filter(Smith_Depression == 1) %>%
  left_join(., demographics, by="ID")

antidep <- main_cat_232k %>%
  dplyr::select(ID, Antidepressant) %>%
  filter(Antidepressant == 1) %>%
  left_join(., demographics, by="ID")

mhq <- main_cat_232k %>%
  dplyr::select(ID, MHQ_depressed) %>%
  filter(MHQ_depressed == 1) %>%
  left_join(., demographics, by="ID")

helpseeking <- main_cat_232k %>%
  dplyr::select(ID, Seen_GP_or_Psychiatrist) %>%
  filter(Seen_GP_or_Psychiatrist == 1) %>%
  left_join(., demographics, by="ID")

any_dep <- main_cat_232k %>%
  dplyr::select(ID, any_dep) %>%
  filter(any_dep == 1) %>%
  left_join(., demographics, by="ID")

# Create summary table
summarise_demo <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","),
  "Prevalence (Semi-screened)" = paste(percent(round(nrow(x)/(nrow(x)+nrow(semi_screened)), digits = 4), digits = 2)),
  "Prevalence (Screened)" = paste(percent(round(nrow(x)/(nrow(x)+nrow(screened)), digits = 2), digits = 0)),
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 0), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Health Rating (SD)" = paste(format(mean(x$`Health Rating`, na.rm = T), digits = 2), paste("(", format(sd(x$`Health Rating`, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " ")
)
}

icd_desc <- summarise_demo(icd)
sr_desc <- summarise_demo(sr)
smith_desc <- summarise_demo(smith)
antidep_desc <- summarise_demo(antidep)
mhq_desc <- summarise_demo(mhq)
helpseeking_desc <- summarise_demo(helpseeking)
any_dep_desc <- summarise_demo(any_dep)

# descriptives row for controls
desc_controls <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","),
  "Prevalence (Semi-screened)" = "NA",
  "Prevalence (Screened)" = "NA",
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 0), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Health Rating (SD)" = paste(format(mean(x$`Health Rating`, na.rm = T), digits = 2), paste("(", format(sd(x$`Health Rating`, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " ")
)
}

semi_screened_desc <- desc_controls(semi_screened)
screened_desc <- desc_controls(screened)

# combine data
main_cat_summary <- rbind(semi_screened_desc,
                          screened_desc,
                          icd_desc,
                          sr_desc,
                          smith_desc,
                          antidep_desc,
                          mhq_desc,
                          helpseeking_desc,
                          any_dep_desc)

colnames(main_cat_summary) <- c("Count","Prev (semi-screened)","Prev (screened)","Mean age (SD)","Female (%)","SES* (SD)","Current smoker (%)","Mean BMI (SD)","Mean Health Rating (SD)")
rownames(main_cat_summary) <- c("Semi-screened Controls","Screened Controls","Hospital (ICD-10)","Self-reported Depression","Depression (Smith)","Antidepressant Usage","Lifetime Depression (MHQ)","Help-seeking","Any Depression Measure")

kable(main_cat_summary, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

```{r demographics_table_overlapping, echo=FALSE, results = 'asis'}

# Prepare table with demographic information for controls and depression according to degree of endorsement

# Read phenotype file
overlap <- read.table("/file_path/overlapping_measures", header = T)
overlap$any_overlap <- ifelse(rowSums(overlap[,2:ncol(overlap)], na.rm = T) > 0, 1, 0)

# Make overlapping categories dfs and merge with demographics
one <- overlap %>%
  dplyr::select(ID, One_Measure) %>%
  filter(One_Measure == 1) %>%
  left_join(., demographics, by="ID")

two <- overlap %>%
  dplyr::select(ID, Two_Measures) %>%
  filter(Two_Measures == 1) %>%
  left_join(., demographics, by="ID")

three <- overlap %>%
  dplyr::select(ID, Three_Measures) %>%
  filter(Three_Measures == 1) %>%
  left_join(., demographics, by="ID")

four <- overlap %>%
  dplyr::select(ID, Four_Measures) %>%
  filter(Four_Measures == 1) %>%
  left_join(., demographics, by="ID")

five <- overlap %>%
  dplyr::select(ID, Five_Measures) %>%
  filter(Five_Measures == 1) %>%
  left_join(., demographics, by="ID")

any_overlap <- overlap %>%
  dplyr::select(ID, any_overlap) %>%
  filter(any_overlap == 1) %>%
  left_join(., demographics, by="ID")

one_desc <- summarise_demo(one)
two_desc <- summarise_demo(two)
three_desc <- summarise_demo(three)
four_desc <- summarise_demo(four)
five_desc <- summarise_demo(five)
any_overlap_desc <- summarise_demo(any_overlap)

# combine data
overlap_summary <- rbind(semi_screened_desc,
                          screened_desc,
                          one_desc,
                          two_desc,
                          three_desc,
                          four_desc,
                          five_desc,
                          any_overlap_desc)

colnames(overlap_summary) <- c("Count","Prev (semi-screened)","Prev (screened)","Mean age (SD)","Female (%)","SES* (SD)","Current smoker (%)","Mean BMI (SD)","Mean Health Rating (SD)")
rownames(overlap_summary) <- c("Semi-screened Controls","Screened Controls","One Measure","Two Measures","Three Measures","Four Measures","Five Measures","Any Measure")

kable(overlap_summary, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## Controls (N = 232,552) vs Any Depression Phenotype (N = 146,079)

</br>

```{r ttest_demographics_ss_vs_anydep, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between cases and controls

# Semi-screened vs Any Depression Phenotype
age_ss_vs_anydep <- t.test(semi_screened$Age, any_dep$Age) 
table_age_ss_vs_anydep <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(age_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(age_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(age_ss_vs_anydep$p.value, format = "e", digits = 0)
)

sex_ss_vs_anydep <- t.test(semi_screened$Sex, any_dep$Sex) 
table_sex_ss_vs_anydep <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_ss_vs_anydep$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_ss_vs_anydep$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(sex_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(sex_ss_vs_anydep$p.value, format = "e", digits = 0)
)

ses_ss_vs_anydep <- t.test(semi_screened$SES, any_dep$SES) 
table_ses_ss_vs_anydep <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(ses_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(ses_ss_vs_anydep$p.value, format = "e", digits = 0)
)

smoke_ss_vs_anydep <- t.test(semi_screened$`Current Smoker`, any_dep$`Current Smoker`) 
table_smoke_ss_vs_anydep <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(smoke_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(smoke_ss_vs_anydep$p.value, format = "e", digits = 0)
)

bmi_ss_vs_anydep <- t.test(semi_screened$BMI, any_dep$BMI) 
table_bmi_ss_vs_anydep <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(bmi_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(bmi_ss_vs_anydep$p.value, format = "e", digits = 0)
)

health_ss_vs_anydep <- t.test(semi_screened$`Health Rating`, any_dep$`Health Rating`) 
table_health_ss_vs_anydep <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(health_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(health_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(health_ss_vs_anydep$p.value, format = "e", digits = 0)
)

ss_vs_anydep <- rbind(table_age_ss_vs_anydep,
                  table_sex_ss_vs_anydep,
                  table_ses_ss_vs_anydep,
                  table_smoke_ss_vs_anydep,
                  table_bmi_ss_vs_anydep,
                  table_health_ss_vs_anydep)

colnames(ss_vs_anydep) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

ss_vs_anydep$`Estimate (controls)` <- round(ss_vs_anydep$`Estimate (controls)`, digits = 2)
ss_vs_anydep$`Estimate (cases)` <- round(ss_vs_anydep$`Estimate (cases)`, digits = 2)
ss_vs_anydep$`t-statistic` <- round(ss_vs_anydep$`t-statistic`, digits = 0)

kable(ss_vs_anydep, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## MHQ controls (N = 57,805) vs Any Depression Phenotype (N = 146,079)

</br>

```{r ttest_demographics_s_vs_anydep, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between cases and mhq controls

# Screened vs Any Depression Phenotype
age_s_vs_anydep <- t.test(screened$Age, any_dep$Age) 
table_age_s_vs_anydep <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_s_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_s_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(age_s_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(age_s_vs_anydep$parameter)[1,1],
  "p-value" = formatC(age_s_vs_anydep$p.value, format = "e", digits = 0)
)

sex_s_vs_anydep <- t.test(screened$Sex, any_dep$Sex) 
table_sex_s_vs_anydep <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_s_vs_anydep$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_s_vs_anydep$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_s_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(sex_s_vs_anydep$parameter)[1,1],
  "p-value" = formatC(sex_s_vs_anydep$p.value, format = "e", digits = 0)
)

ses_s_vs_anydep <- t.test(screened$SES, any_dep$SES) 
table_ses_s_vs_anydep <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_s_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_s_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_s_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(ses_s_vs_anydep$parameter)[1,1],
  "p-value" = formatC(ses_s_vs_anydep$p.value, format = "e", digits = 0)
)

smoke_s_vs_anydep <- t.test(screened$`Current Smoker`, any_dep$`Current Smoker`) 
table_smoke_s_vs_anydep <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_s_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_s_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_s_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(smoke_s_vs_anydep$parameter)[1,1],
  "p-value" = formatC(smoke_s_vs_anydep$p.value, format = "e", digits = 0)
)

bmi_s_vs_anydep <- t.test(screened$BMI, any_dep$BMI) 
table_bmi_s_vs_anydep <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_s_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_s_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_s_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(bmi_s_vs_anydep$parameter)[1,1],
  "p-value" = formatC(bmi_s_vs_anydep$p.value, format = "e", digits = 0)
)

health_s_vs_anydep <- t.test(screened$`Health Rating`, any_dep$`Health Rating`) 
table_health_s_vs_anydep <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_s_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_s_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(health_s_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(health_s_vs_anydep$parameter)[1,1],
  "p-value" = formatC(health_s_vs_anydep$p.value, format = "e", digits = 0)
)

s_vs_anydep <- rbind(table_age_s_vs_anydep,
                  table_sex_s_vs_anydep,
                  table_ses_s_vs_anydep,
                  table_smoke_s_vs_anydep,
                  table_bmi_s_vs_anydep,
                  table_health_s_vs_anydep)

colnames(s_vs_anydep) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

s_vs_anydep$`Estimate (controls)` <- round(s_vs_anydep$`Estimate (controls)`, digits = 2)
s_vs_anydep$`Estimate (cases)` <- round(s_vs_anydep$`Estimate (cases)`, digits = 2)
s_vs_anydep$`t-statistic` <- round(s_vs_anydep$`t-statistic`, digits = 0)

kable(s_vs_anydep, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## Controls (N = 232,552) vs Any Depression Measure (N = 93,414)

</br>

```{r ttest_demographics_ss_vs_overlap, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between cases and controls

# Semi-screened vs Any Depression Overlap
age_ss_vs_overlap <- t.test(semi_screened$Age, any_overlap$Age) 
table_age_ss_vs_overlap <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(age_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(age_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(age_ss_vs_overlap$p.value, format = "e", digits = 0)
)

sex_ss_vs_overlap <- t.test(semi_screened$Sex, any_overlap$Sex) 
table_sex_ss_vs_overlap <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_ss_vs_overlap$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_ss_vs_overlap$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(sex_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(sex_ss_vs_overlap$p.value, format = "e", digits = 0)
)

ses_ss_vs_overlap <- t.test(semi_screened$SES, any_overlap$SES) 
table_ses_ss_vs_overlap <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(ses_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(ses_ss_vs_overlap$p.value, format = "e", digits = 0)
)

smoke_ss_vs_overlap <- t.test(semi_screened$`Current Smoker`, any_overlap$`Current Smoker`) 
table_smoke_ss_vs_overlap <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(smoke_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(smoke_ss_vs_overlap$p.value, format = "e", digits = 0)
)

bmi_ss_vs_overlap <- t.test(semi_screened$BMI, any_overlap$BMI) 
table_bmi_ss_vs_overlap <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(bmi_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(bmi_ss_vs_overlap$p.value, format = "e", digits = 0)
)

health_ss_vs_overlap <- t.test(semi_screened$`Health Rating`, any_overlap$`Health Rating`) 
table_health_ss_vs_overlap <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(health_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(health_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(health_ss_vs_overlap$p.value, format = "e", digits = 0)
)

ss_vs_overlap <- rbind(table_age_ss_vs_overlap,
                  table_sex_ss_vs_overlap,
                  table_ses_ss_vs_overlap,
                  table_smoke_ss_vs_overlap,
                  table_bmi_ss_vs_overlap,
                  table_health_ss_vs_overlap)

colnames(ss_vs_overlap) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

ss_vs_overlap$`Estimate (controls)` <- round(ss_vs_overlap$`Estimate (controls)`, digits = 2)
ss_vs_overlap$`Estimate (cases)` <- round(ss_vs_overlap$`Estimate (cases)`, digits = 2)
ss_vs_overlap$`t-statistic` <- round(ss_vs_overlap$`t-statistic`, digits = 0)

kable(ss_vs_overlap, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## MHQ controls (N = 57,805) vs Any Depression Measure (N = 93,414)

</br>

```{r ttest_demographics_s_vs_overlap, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between cases and mhq controls

# Screened vs vs Any Depression Overlap
age_s_vs_overlap <- t.test(screened$Age, any_overlap$Age) 
table_age_s_vs_overlap <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(age_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(age_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(age_s_vs_overlap$p.value, format = "e", digits = 0)
)

sex_s_vs_overlap <- t.test(screened$Sex, any_overlap$Sex) 
table_sex_s_vs_overlap <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_s_vs_overlap$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_s_vs_overlap$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(sex_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(sex_s_vs_overlap$p.value, format = "e", digits = 0)
)

ses_s_vs_overlap <- t.test(screened$SES, any_overlap$SES) 
table_ses_s_vs_overlap <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(ses_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(ses_s_vs_overlap$p.value, format = "e", digits = 0)
)

smoke_s_vs_overlap <- t.test(screened$`Current Smoker`, any_overlap$`Current Smoker`) 
table_smoke_s_vs_overlap <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(smoke_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(smoke_s_vs_overlap$p.value, format = "e", digits = 0)
)

bmi_s_vs_overlap <- t.test(screened$BMI, any_overlap$BMI) 
table_bmi_s_vs_overlap <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(bmi_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(bmi_s_vs_overlap$p.value, format = "e", digits = 0)
)

health_s_vs_overlap <- t.test(screened$`Health Rating`, any_overlap$`Health Rating`) 
table_health_s_vs_overlap <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(health_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(health_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(health_s_vs_overlap$p.value, format = "e", digits = 0)
)

s_vs_overlap <- rbind(table_age_s_vs_overlap,
                  table_sex_s_vs_overlap,
                  table_ses_s_vs_overlap,
                  table_smoke_s_vs_overlap,
                  table_bmi_s_vs_overlap,
                  table_health_s_vs_overlap)

colnames(s_vs_overlap) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

s_vs_overlap$`Estimate (controls)` <- round(s_vs_overlap$`Estimate (controls)`, digits = 2)
s_vs_overlap$`Estimate (cases)` <- round(s_vs_overlap$`Estimate (cases)`, digits = 2)
s_vs_overlap$`t-statistic` <- round(s_vs_overlap$`t-statistic`, digits = 0)

kable(s_vs_overlap, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## Calculate depression cases (by different measures) without MHQ participants, summarise demographics

</br>

```{r new_demographics_table_main_cat, echo=FALSE, results = 'asis'}

# Summarise demographic information for depression cases according to different measures in those that did not complete the MHQ

# Read demographics
demographics <- read.table("/file_path/demographics_health", header = T)
colnames(demographics) <- c("ID","Age","Sex","SES","BMI","Current Smoker","Health Rating")

# Recode smoking: change to 1 = current, 0 = not current (never or previous), NA = prefer not to answer
# Smoking Coding
# -3	Prefer not to answer
# 0	Never
# 1	Previous
# 2	Current
demographics$`Current Smoker` <- ifelse(demographics$`Current Smoker`==-3, NA, demographics$`Current Smoker`)
demographics$`Current Smoker` <- ifelse(demographics$`Current Smoker`==1, 0, demographics$`Current Smoker`)                                 
demographics$`Current Smoker` <- ifelse(demographics$`Current Smoker`==2, 1, demographics$`Current Smoker`)

# Recode health: change to -1 and -3 to NA
# 1 = Excellent
# 2 = Good
# 3 = Fair
# 4 = Poor
# -1 = Do not know
# -3 = Prefer not to answer
demographics$`Health Rating` <- ifelse(demographics$`Health Rating`==-3 | demographics$`Health Rating`==-1, NA, demographics$`Health Rating`)

# MERGE WITH DF OF NON-MHQ PARTICIPANTS

# Read phenotype file with 232k and 58k cases
main_cat_232k <- read.table("/file_path/dep_main_sub_cat", header = T)
main_cat_58k <- read.table("/file_path/dep_main_sub_cat_mhq_controls", header = T)

# get mhq participants
qc_mhq_participants <- mhq_participants %>%
  filter(MHQ_participants==1)

# remove mhq participants
mhq_removed_main_cat_232k <- filter(main_cat_232k, !(ID %in% qc_mhq_participants$ID))
mhq_removed_main_cat_58k <- filter(main_cat_58k, !(ID %in% qc_mhq_participants$ID))

# add mhq cases back to dfs - first make df for MHQ cases only
mhq_cases <- main_cat_58k[,1:2] %>%
  filter(MHQ_depressed == 1)

main_cat_232k <- full_join(mhq_removed_main_cat_232k,mhq_cases)
main_cat_232k <- main_cat_232k[,c(1:3,6,10:12)]

main_cat_58k <- full_join(mhq_removed_main_cat_58k,mhq_cases)
main_cat_58k <- main_cat_58k[,c(1:3,6,10:12)]

# count 'any dep' - five measures (without MHQ) + MHQ cases
main_cat_232k$any_dep <- ifelse(rowSums(main_cat_232k[,2:ncol(main_cat_232k)], na.rm = T) > 0, 1, 0)
main_cat_58k$any_dep <- ifelse(rowSums(main_cat_58k[,2:ncol(main_cat_58k)], na.rm = T) > 0, 1, 0)

# make controls
controls_232k <- read.table("/file_path/dep_main_sub_cat", header = T)
controls_232k$any_dep <- ifelse(rowSums(controls_232k[,2:ncol(controls_232k)], na.rm = T) > 0, 1, 0)
controls_58k <- read.table("/file_path/dep_main_sub_cat_mhq_controls", header = T)
controls_58k$any_dep <- ifelse(rowSums(controls_58k[,2:ncol(controls_58k)], na.rm = T) > 0, 1, 0)

# Make semi-screened and screened controls dfs and merge with demographics
semi_screened <- controls_232k %>%
  dplyr::select(ID = ID, semi_screened = any_dep) %>%
  filter(semi_screened == 0) %>%
  left_join(., demographics, by="ID")

screened <- controls_58k %>%
  dplyr::select(ID = ID, screened = any_dep) %>%
  filter(screened == 0) %>%
  left_join(., demographics, by="ID")

# Make main categories dfs and merge with demographics

icd <- main_cat_232k %>%
  dplyr::select(ID, ICD_any_depression) %>%
  filter(ICD_any_depression == 1) %>%
  left_join(., demographics, by="ID")

sr <- main_cat_232k %>%
  dplyr::select(ID, SR_depression) %>%
  filter(SR_depression == 1) %>%
  left_join(., demographics, by="ID")

smith <- main_cat_232k %>%
  dplyr::select(ID, Smith_Depression) %>%
  filter(Smith_Depression == 1) %>%
  left_join(., demographics, by="ID")

antidep <- main_cat_232k %>%
  dplyr::select(ID, Antidepressant) %>%
  filter(Antidepressant == 1) %>%
  left_join(., demographics, by="ID")

mhq <- main_cat_232k %>%
  dplyr::select(ID, MHQ_depressed) %>%
  filter(MHQ_depressed == 1) %>%
  left_join(., demographics, by="ID")

helpseeking <- main_cat_232k %>%
  dplyr::select(ID, Seen_GP_or_Psychiatrist) %>%
  filter(Seen_GP_or_Psychiatrist == 1) %>%
  left_join(., demographics, by="ID")

any_dep <- main_cat_232k %>%
  dplyr::select(ID, any_dep) %>%
  filter(any_dep == 1) %>%
  left_join(., demographics, by="ID")

# Create summary table
summarise_demo <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","),
  "Prevalence (Semi-screened)" = paste(percent(round(nrow(x)/(nrow(x)+nrow(semi_screened)), digits = 4), digits = 2)),
  "Prevalence (Screened)" = paste(percent(round(nrow(x)/(nrow(x)+nrow(screened)), digits = 2), digits = 0)),
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Health Rating (SD)" = paste(format(mean(x$`Health Rating`, na.rm = T), digits = 2), paste("(", format(sd(x$`Health Rating`, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " ")
)
}

icd_desc <- summarise_demo(icd)
sr_desc <- summarise_demo(sr)
smith_desc <- summarise_demo(smith)
antidep_desc <- summarise_demo(antidep)
mhq_desc <- summarise_demo(mhq)
helpseeking_desc <- summarise_demo(helpseeking)
any_dep_desc <- summarise_demo(any_dep)

# descriptives row for controls
desc_controls <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","),
  "Prevalence (Semi-screened)" = "NA",
  "Prevalence (Screened)" = "NA",
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),
  "Health Rating (SD)" = paste(format(mean(x$`Health Rating`, na.rm = T), digits = 2), paste("(", format(sd(x$`Health Rating`, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " ")
)
}

semi_screened_desc <- desc_controls(semi_screened)
screened_desc <- desc_controls(screened)

# combine data
main_cat_summary <- rbind(semi_screened_desc,
                          screened_desc,
                          icd_desc,
                          sr_desc,
                          smith_desc,
                          antidep_desc,
                          mhq_desc,
                          helpseeking_desc,
                          any_dep_desc)

colnames(main_cat_summary) <- c("Count","Prev (semi-screened)","Prev (screened)","Mean age (SD)","Female (%)","SES* (SD)","Current smoker (%)","Mean BMI (SD)","Mean Health Rating (SD)")
rownames(main_cat_summary) <- c("Semi-screened Controls","Screened Controls","Hospital (ICD-10)","Self-reported Depression","Depression (Smith)","Antidepressant Usage","Lifetime Depression (MHQ)","Help-seeking","Any Depression Measure")

kable(main_cat_summary, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

```{r new_demographics_table_overlapping, echo=FALSE, results = 'asis'}

# Summarise demographic information according to degree of overlap

# Read phenotype file
overlap <- read.table("/file_path/overlapping_measures", header = T)
overlap$any_overlap <- ifelse(rowSums(overlap[,2:ncol(overlap)], na.rm = T) > 0, 1, 0)

# Make overlapping categories dfs and merge with demographics
one <- overlap %>%
  dplyr::select(ID, One_Measure) %>%
  filter(One_Measure == 1) %>%
  left_join(., demographics, by="ID")

two <- overlap %>%
  dplyr::select(ID, Two_Measures) %>%
  filter(Two_Measures == 1) %>%
  left_join(., demographics, by="ID")

three <- overlap %>%
  dplyr::select(ID, Three_Measures) %>%
  filter(Three_Measures == 1) %>%
  left_join(., demographics, by="ID")

four <- overlap %>%
  dplyr::select(ID, Four_Measures) %>%
  filter(Four_Measures == 1) %>%
  left_join(., demographics, by="ID")

five <- overlap %>%
  dplyr::select(ID, Five_Measures) %>%
  filter(Five_Measures == 1) %>%
  left_join(., demographics, by="ID")

any_overlap <- overlap %>%
  dplyr::select(ID, any_overlap) %>%
  filter(any_overlap == 1) %>%
  left_join(., demographics, by="ID")

one_desc <- summarise_demo(one)
two_desc <- summarise_demo(two)
three_desc <- summarise_demo(three)
four_desc <- summarise_demo(four)
five_desc <- summarise_demo(five)
any_overlap_desc <- summarise_demo(any_overlap)

# combine data
overlap_summary <- rbind(semi_screened_desc,
                          screened_desc,
                          one_desc,
                          two_desc,
                          three_desc,
                          four_desc,
                          five_desc,
                          any_overlap_desc)

colnames(overlap_summary) <- c("Count","Prev (semi-screened)","Prev (screened)","Mean age (SD)","Female (%)","SES* (SD)","Current smoker (%)","Mean BMI (SD)","Mean Health Rating (SD)")
rownames(overlap_summary) <- c("Semi-screened Controls","Screened Controls","One Measure","Two Measures","Three Measures","Four Measures","Five Measures","Any Measure")

kable(overlap_summary, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Any Depression (without MHQ) (N = 93,414) vs MHQ cases (N = 28,982)

</br>

```{r new_ttest_demographics_any_dep_vs_mhq, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between non-mhq cases and mhq cases

# Any Depression (without MHQ) (N = 93,414) vs MHQ (N = 28,982)

age_any_vs_mhq <- t.test(any_overlap$Age, mhq$Age) 
table_age_any_vs_mhq <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_any_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_any_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(age_any_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(age_any_vs_mhq$parameter)[1,1],
  "p-value" = formatC(age_any_vs_mhq$p.value, format = "e", digits = 0)
)

sex_any_vs_mhq <- t.test(any_overlap$Sex, mhq$Sex) 
table_sex_any_vs_mhq <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_any_vs_mhq$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_any_vs_mhq$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_any_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(sex_any_vs_mhq$parameter)[1,1],
  "p-value" = formatC(sex_any_vs_mhq$p.value, format = "e", digits = 0)
)

ses_any_vs_mhq <- t.test(any_overlap$SES, mhq$SES) 
table_ses_any_vs_mhq <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_any_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_any_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_any_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(ses_any_vs_mhq$parameter)[1,1],
  "p-value" = formatC(ses_any_vs_mhq$p.value, format = "e", digits = 0)
)

smoke_any_vs_mhq <- t.test(any_overlap$`Current Smoker`, mhq$`Current Smoker`) 
table_smoke_any_vs_mhq <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_any_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_any_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_any_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(smoke_any_vs_mhq$parameter)[1,1],
  "p-value" = formatC(smoke_any_vs_mhq$p.value, format = "e", digits = 0)
)

bmi_any_vs_mhq <- t.test(any_overlap$BMI, mhq$BMI) 
table_bmi_any_vs_mhq <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_any_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_any_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_any_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(bmi_any_vs_mhq$parameter)[1,1],
  "p-value" = formatC(bmi_any_vs_mhq$p.value, format = "e", digits = 0)
)

health_any_vs_mhq <- t.test(any_overlap$`Health Rating`, mhq$`Health Rating`) 
table_health_any_vs_mhq <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_any_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_any_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(health_any_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(health_any_vs_mhq$parameter)[1,1],
  "p-value" = formatC(health_any_vs_mhq$p.value, format = "e", digits = 0)
)

any_vs_mhq <- rbind(table_age_any_vs_mhq,
                  table_sex_any_vs_mhq,
                  table_ses_any_vs_mhq,
                  table_smoke_any_vs_mhq,
                  table_bmi_any_vs_mhq,
                  table_health_any_vs_mhq)

colnames(any_vs_mhq) <- c("Demographic","Non-MHQ","Lifetime Depression MHQ","t-statistic","df","p-value")

any_vs_mhq$`Non-MHQ` <- round(any_vs_mhq$`Non-MHQ`, digits = 2)
any_vs_mhq$`Lifetime Depression MHQ` <- round(any_vs_mhq$`Lifetime Depression MHQ`, digits = 2)
any_vs_mhq$`t-statistic` <- round(any_vs_mhq$`t-statistic`, digits = 0)

kable(any_vs_mhq, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Controls (N = 232,552) vs Any Depression (without MHQ) (N = 93,414)

</br>

```{r new_ttest_demographics_ss_vs_overlap, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between cases and controls

# Semi-screened vs Any Depression Overlap
age_ss_vs_overlap <- t.test(semi_screened$Age, any_overlap$Age) 
table_age_ss_vs_overlap <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(age_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(age_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(age_ss_vs_overlap$p.value, format = "e", digits = 0)
)

sex_ss_vs_overlap <- t.test(semi_screened$Sex, any_overlap$Sex) 
table_sex_ss_vs_overlap <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_ss_vs_overlap$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_ss_vs_overlap$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(sex_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(sex_ss_vs_overlap$p.value, format = "e", digits = 0)
)

ses_ss_vs_overlap <- t.test(semi_screened$SES, any_overlap$SES) 
table_ses_ss_vs_overlap <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(ses_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(ses_ss_vs_overlap$p.value, format = "e", digits = 0)
)

smoke_ss_vs_overlap <- t.test(semi_screened$`Current Smoker`, any_overlap$`Current Smoker`) 
table_smoke_ss_vs_overlap <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(smoke_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(smoke_ss_vs_overlap$p.value, format = "e", digits = 0)
)

bmi_ss_vs_overlap <- t.test(semi_screened$BMI, any_overlap$BMI) 
table_bmi_ss_vs_overlap <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(bmi_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(bmi_ss_vs_overlap$p.value, format = "e", digits = 0)
)

health_ss_vs_overlap <- t.test(semi_screened$`Health Rating`, any_overlap$`Health Rating`) 
table_health_ss_vs_overlap <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_ss_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_ss_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(health_ss_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(health_ss_vs_overlap$parameter)[1,1],
  "p-value" = formatC(health_ss_vs_overlap$p.value, format = "e", digits = 0)
)

ss_vs_overlap <- rbind(table_age_ss_vs_overlap,
                  table_sex_ss_vs_overlap,
                  table_ses_ss_vs_overlap,
                  table_smoke_ss_vs_overlap,
                  table_bmi_ss_vs_overlap,
                  table_health_ss_vs_overlap)

colnames(ss_vs_overlap) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

ss_vs_overlap$`Estimate (controls)` <- round(ss_vs_overlap$`Estimate (controls)`, digits = 2)
ss_vs_overlap$`Estimate (cases)` <- round(ss_vs_overlap$`Estimate (cases)`, digits = 2)
ss_vs_overlap$`t-statistic` <- round(ss_vs_overlap$`t-statistic`, digits = 0)

kable(ss_vs_overlap, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### MHQ controls (N = 57,805) vs Any Depression (without MHQ) (N = 93,414)

</br>

```{r new_ttest_demographics_s_vs_overlap, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between cases and mhq controls

# Screened vs vs Any Depression Overlap
age_s_vs_overlap <- t.test(screened$Age, any_overlap$Age) 
table_age_s_vs_overlap <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(age_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(age_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(age_s_vs_overlap$p.value, format = "e", digits = 0)
)

sex_s_vs_overlap <- t.test(screened$Sex, any_overlap$Sex) 
table_sex_s_vs_overlap <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_s_vs_overlap$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_s_vs_overlap$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(sex_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(sex_s_vs_overlap$p.value, format = "e", digits = 0)
)

ses_s_vs_overlap <- t.test(screened$SES, any_overlap$SES) 
table_ses_s_vs_overlap <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(ses_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(ses_s_vs_overlap$p.value, format = "e", digits = 0)
)

smoke_s_vs_overlap <- t.test(screened$`Current Smoker`, any_overlap$`Current Smoker`) 
table_smoke_s_vs_overlap <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(smoke_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(smoke_s_vs_overlap$p.value, format = "e", digits = 0)
)

bmi_s_vs_overlap <- t.test(screened$BMI, any_overlap$BMI) 
table_bmi_s_vs_overlap <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(bmi_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(bmi_s_vs_overlap$p.value, format = "e", digits = 0)
)

health_s_vs_overlap <- t.test(screened$`Health Rating`, any_overlap$`Health Rating`) 
table_health_s_vs_overlap <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_s_vs_overlap$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_s_vs_overlap$estimate)[2,1],
  "t-statistic" = as.data.frame(health_s_vs_overlap$statistic)[1,1],
  "df" = as.data.frame(health_s_vs_overlap$parameter)[1,1],
  "p-value" = formatC(health_s_vs_overlap$p.value, format = "e", digits = 0)
)

s_vs_overlap <- rbind(table_age_s_vs_overlap,
                  table_sex_s_vs_overlap,
                  table_ses_s_vs_overlap,
                  table_smoke_s_vs_overlap,
                  table_bmi_s_vs_overlap,
                  table_health_s_vs_overlap)

colnames(s_vs_overlap) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

s_vs_overlap$`Estimate (controls)` <- round(s_vs_overlap$`Estimate (controls)`, digits = 2)
s_vs_overlap$`Estimate (cases)` <- round(s_vs_overlap$`Estimate (cases)`, digits = 2)
s_vs_overlap$`t-statistic` <- round(s_vs_overlap$`t-statistic`, digits = 0)

kable(s_vs_overlap, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### Controls (N = 232,552) vs MHQ (N = 28,982)

</br>

```{r new_ttest_demographics_ss_vs_mhq, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between mhq cases and controls

# Semi-screened vs Any Depression Overlap
age_ss_vs_mhq <- t.test(semi_screened$Age, mhq$Age) 
table_age_ss_vs_mhq <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_ss_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_ss_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(age_ss_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(age_ss_vs_mhq$parameter)[1,1],
  "p-value" = formatC(age_ss_vs_mhq$p.value, format = "e", digits = 0)
)

sex_ss_vs_mhq <- t.test(semi_screened$Sex, mhq$Sex) 
table_sex_ss_vs_mhq <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_ss_vs_mhq$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_ss_vs_mhq$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_ss_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(sex_ss_vs_mhq$parameter)[1,1],
  "p-value" = formatC(sex_ss_vs_mhq$p.value, format = "e", digits = 0)
)

ses_ss_vs_mhq <- t.test(semi_screened$SES, mhq$SES) 
table_ses_ss_vs_mhq <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_ss_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_ss_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_ss_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(ses_ss_vs_mhq$parameter)[1,1],
  "p-value" = formatC(ses_ss_vs_mhq$p.value, format = "e", digits = 0)
)

smoke_ss_vs_mhq <- t.test(semi_screened$`Current Smoker`, mhq$`Current Smoker`) 
table_smoke_ss_vs_mhq <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_ss_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_ss_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_ss_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(smoke_ss_vs_mhq$parameter)[1,1],
  "p-value" = formatC(smoke_ss_vs_mhq$p.value, format = "e", digits = 0)
)

bmi_ss_vs_mhq <- t.test(semi_screened$BMI, mhq$BMI) 
table_bmi_ss_vs_mhq <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_ss_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_ss_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_ss_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(bmi_ss_vs_mhq$parameter)[1,1],
  "p-value" = formatC(bmi_ss_vs_mhq$p.value, format = "e", digits = 0)
)

health_ss_vs_mhq <- t.test(semi_screened$`Health Rating`, mhq$`Health Rating`) 
table_health_ss_vs_mhq <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_ss_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_ss_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(health_ss_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(health_ss_vs_mhq$parameter)[1,1],
  "p-value" = formatC(health_ss_vs_mhq$p.value, format = "e", digits = 0)
)

ss_vs_mhq <- rbind(table_age_ss_vs_mhq,
                  table_sex_ss_vs_mhq,
                  table_ses_ss_vs_mhq,
                  table_smoke_ss_vs_mhq,
                  table_bmi_ss_vs_mhq,
                  table_health_ss_vs_mhq)

colnames(ss_vs_mhq) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

ss_vs_mhq$`Estimate (controls)` <- round(ss_vs_mhq$`Estimate (controls)`, digits = 2)
ss_vs_mhq$`Estimate (cases)` <- round(ss_vs_mhq$`Estimate (cases)`, digits = 2)
ss_vs_mhq$`t-statistic` <- round(ss_vs_mhq$`t-statistic`, digits = 0)

kable(ss_vs_mhq, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### MHQ controls (N = 57,805) vs MHQ (N = 28,982)

</br>

```{r new_ttest_demographics_s_vs_mhq, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between mhq cases and mhq controls

# Screened vs vs Any Depression Overlap
age_s_vs_mhq <- t.test(screened$Age, mhq$Age) 
table_age_s_vs_mhq <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_s_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_s_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(age_s_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(age_s_vs_mhq$parameter)[1,1],
  "p-value" = formatC(age_s_vs_mhq$p.value, format = "e", digits = 0)
)

sex_s_vs_mhq <- t.test(screened$Sex, mhq$Sex) 
table_sex_s_vs_mhq <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_s_vs_mhq$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_s_vs_mhq$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_s_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(sex_s_vs_mhq$parameter)[1,1],
  "p-value" = formatC(sex_s_vs_mhq$p.value, format = "e", digits = 0)
)

ses_s_vs_mhq <- t.test(screened$SES, mhq$SES) 
table_ses_s_vs_mhq <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_s_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_s_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_s_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(ses_s_vs_mhq$parameter)[1,1],
  "p-value" = formatC(ses_s_vs_mhq$p.value, format = "e", digits = 0)
)

smoke_s_vs_mhq <- t.test(screened$`Current Smoker`, mhq$`Current Smoker`) 
table_smoke_s_vs_mhq <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_s_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_s_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_s_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(smoke_s_vs_mhq$parameter)[1,1],
  "p-value" = formatC(smoke_s_vs_mhq$p.value, format = "e", digits = 0)
)

bmi_s_vs_mhq <- t.test(screened$BMI, mhq$BMI) 
table_bmi_s_vs_mhq <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_s_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_s_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_s_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(bmi_s_vs_mhq$parameter)[1,1],
  "p-value" = formatC(bmi_s_vs_mhq$p.value, format = "e", digits = 0)
)

health_s_vs_mhq <- t.test(screened$`Health Rating`, mhq$`Health Rating`) 
table_health_s_vs_mhq <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_s_vs_mhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_s_vs_mhq$estimate)[2,1],
  "t-statistic" = as.data.frame(health_s_vs_mhq$statistic)[1,1],
  "df" = as.data.frame(health_s_vs_mhq$parameter)[1,1],
  "p-value" = formatC(health_s_vs_mhq$p.value, format = "e", digits = 0)
)

s_vs_mhq <- rbind(table_age_s_vs_mhq,
                  table_sex_s_vs_mhq,
                  table_ses_s_vs_mhq,
                  table_smoke_s_vs_mhq,
                  table_bmi_s_vs_mhq,
                  table_health_s_vs_mhq)

colnames(s_vs_mhq) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

s_vs_mhq$`Estimate (controls)` <- round(s_vs_mhq$`Estimate (controls)`, digits = 2)
s_vs_mhq$`Estimate (cases)` <- round(s_vs_mhq$`Estimate (cases)`, digits = 2)
s_vs_mhq$`t-statistic` <- round(s_vs_mhq$`t-statistic`, digits = 0)

kable(s_vs_mhq, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

### MHQ participants (N = 126,261) vs Non-MHQ participants (N = 259,443)

</br>

```{r new_ttest_demographics_mhq_participants_vs_non_participants, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between mhq participants and non-participants

# Make df with all participants after genetic QC including MHQ participation and demographics
all_pp <- left_join(qc_ids, mhq_participants)
all_pp <- left_join(all_pp, demographics) 
all_pp <- all_pp[,7:ncol(all_pp)]

mhq_pp <- all_pp[all_pp$MHQ_participants==1,] 
mhq_pp <- mhq_pp[!is.na(mhq_pp$MHQ_participants),]  

non_mhq_pp <- all_pp[all_pp$MHQ_participants==0,]

# MHQ participants vs non-participants

age_mhq_vs_nonmhq <- t.test(mhq_pp$Age, non_mhq_pp$Age) 
table_age_mhq_vs_nonmhq <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_mhq_vs_nonmhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_mhq_vs_nonmhq$estimate)[2,1],
  "t-statistic" = as.data.frame(age_mhq_vs_nonmhq$statistic)[1,1],
  "df" = as.data.frame(age_mhq_vs_nonmhq$parameter)[1,1],
  "p-value" = formatC(age_mhq_vs_nonmhq$p.value, format = "e", digits = 0)
)

sex_mhq_vs_nonmhq <- t.test(mhq_pp$Sex, non_mhq_pp$Sex) 
table_sex_mhq_vs_nonmhq <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_mhq_vs_nonmhq$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_mhq_vs_nonmhq$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_mhq_vs_nonmhq$statistic)[1,1],
  "df" = as.data.frame(sex_mhq_vs_nonmhq$parameter)[1,1],
  "p-value" = formatC(sex_mhq_vs_nonmhq$p.value, format = "e", digits = 0)
)

ses_mhq_vs_nonmhq <- t.test(mhq_pp$SES, non_mhq_pp$SES) 
table_ses_mhq_vs_nonmhq <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_mhq_vs_nonmhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_mhq_vs_nonmhq$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_mhq_vs_nonmhq$statistic)[1,1],
  "df" = as.data.frame(ses_mhq_vs_nonmhq$parameter)[1,1],
  "p-value" = formatC(ses_mhq_vs_nonmhq$p.value, format = "e", digits = 0)
)

smoke_mhq_vs_nonmhq <- t.test(mhq_pp$`Current Smoker`, non_mhq_pp$`Current Smoker`) 
table_smoke_mhq_vs_nonmhq <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_mhq_vs_nonmhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_mhq_vs_nonmhq$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_mhq_vs_nonmhq$statistic)[1,1],
  "df" = as.data.frame(smoke_mhq_vs_nonmhq$parameter)[1,1],
  "p-value" = formatC(smoke_mhq_vs_nonmhq$p.value, format = "e", digits = 0)
)

bmi_mhq_vs_nonmhq <- t.test(mhq_pp$BMI, non_mhq_pp$BMI) 
table_bmi_mhq_vs_nonmhq <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_mhq_vs_nonmhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_mhq_vs_nonmhq$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_mhq_vs_nonmhq$statistic)[1,1],
  "df" = as.data.frame(bmi_mhq_vs_nonmhq$parameter)[1,1],
  "p-value" = formatC(bmi_mhq_vs_nonmhq$p.value, format = "e", digits = 0)
)

health_mhq_vs_nonmhq <- t.test(mhq_pp$`Health Rating`, non_mhq_pp$`Health Rating`) 
table_health_mhq_vs_nonmhq <- data.frame(
  "Demographic" = "Mean Health Rating",
  "Estimate (controls)" = as.data.frame(health_mhq_vs_nonmhq$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(health_mhq_vs_nonmhq$estimate)[2,1],
  "t-statistic" = as.data.frame(health_mhq_vs_nonmhq$statistic)[1,1],
  "df" = as.data.frame(health_mhq_vs_nonmhq$parameter)[1,1],
  "p-value" = formatC(health_mhq_vs_nonmhq$p.value, format = "e", digits = 0)
)

mhq_vs_nonmhq <- rbind(table_age_mhq_vs_nonmhq,
                  table_sex_mhq_vs_nonmhq,
                  table_ses_mhq_vs_nonmhq,
                  table_smoke_mhq_vs_nonmhq,
                  table_bmi_mhq_vs_nonmhq,
                  table_health_mhq_vs_nonmhq)

colnames(mhq_vs_nonmhq) <- c("Demographic","Estimate (MHQ participants)","Estimate (Non-MHQ participants)","t-statistic","df","p-value")

mhq_vs_nonmhq$`Estimate (MHQ participants)` <- round(mhq_vs_nonmhq$`Estimate (MHQ participants)`, digits = 2)
mhq_vs_nonmhq$`Estimate (Non-MHQ participants)` <- round(mhq_vs_nonmhq$`Estimate (Non-MHQ participants)`, digits = 2)
mhq_vs_nonmhq$`t-statistic` <- round(mhq_vs_nonmhq$`t-statistic`, digits = 0)

kable(mhq_vs_nonmhq, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

# One to Five measures in MHQ cases and controls 

</br>

```{r new_make_mhq_case_control_df_with_other_dep_measures, echo=FALSE}

# Calculate the number of non-MHQ depression measures endorsed in mhq participants, stratified by case/control status on the MHQ

# read data (note: final_mhq has psychosis and missing data from mhq removed)
final_mhq <- read.table("/file_path/final_mhq", header = T) 
dep_measures <- read.table("/file_path/depression_phenotypes", header = T) %>%
  dplyr::select(ID, ICD_any_depression, SR_depression, Antidepressant, Smith_Depression, Seen_GP_or_Psychiatrist)

# merge
mhq_one_to_five <- left_join(final_mhq, dep_measures)

# Fix column names
colnames(mhq_one_to_five)[4] <- "Hospital (ICD-10)"
colnames(mhq_one_to_five)[5] <- "Self-reported Depression"
colnames(mhq_one_to_five)[6] <- "Antidepressant Usage"
colnames(mhq_one_to_five)[7] <- "Depression (Smith)"
colnames(mhq_one_to_five)[8] <- "Help-seeking"

# make factors columns for MHQ participant status
mhq_one_to_five$mhq_status <- with(mhq_one_to_five, 
                                   ifelse(mhq_final == 0, "MHQ_Control",
                                     ifelse(mhq_final == 1, "MHQ_Case", 
                                     ifelse(mhq_final == 3, "No_CIDI_PHQ_MHQ_Screen", 
                                     ifelse(mhq_final == 4, "No_CIDI_UKB_Screen", NA)))))

mhq_one_to_five$mhq_cc_status <- with(mhq_one_to_five, 
                                   ifelse(case_control == 0 | is.na(case_control), "No_CIDI",
                                     ifelse(case_control == 1, "MHQ_Case", NA)))

mhq_one_to_five$mhq_three_status <- with(mhq_one_to_five, 
                                   ifelse(mhq_final == 0, "MHQ_Control", 
                                    ifelse(mhq_final == 1, "MHQ_Case", 
                                      ifelse(mhq_final == 3 | mhq_final == 4, "No_CIDI_Screened", NA))))

mhq_one_to_five$count <- rowSums(mhq_one_to_five[,4:8])

```

```{r table_mhq_status, echo=FALSE}

# count number of measures and put in table
breakdown <- expss::cro(mhq_one_to_five$mhq_status, mhq_one_to_five$count)
breakdown

```

**Table:** MHQ participants (after screening for psychosis and removing individuals with missing data on MHQ) divided into those who met the criteria for CIDI lifetime depression (MHQ_Case), those that did not meet CIDI criteria and had no other depression measures (MHQ_Control), those that did not meet CIDI criteria but were screened from MHQ controls because they had other psychopathology on MHQ (No_CIDI_PHQ_MHQ_Screen) and those that did not meet CIDI criteria but were screened from MHQ controls because they had depression measures outside of MHQ (No_CIDI_UKB_Screen).

</br>

```{r new_make_mhq_plot_with_one_to_five_measures_mhq_status, echo=FALSE, results = 'asis', fig.align='center', fig.height=8, fig.width=16}

# Make plot to show number of non-MHQ depression measures in MHQ participants, stratified by MHQ case/control status

# make columns for 0-5 measures to calculate percentages to use in plot
mhq_one_to_five$zero <- ifelse(mhq_one_to_five$count==0, 1, 0)
mhq_one_to_five$one <- ifelse(mhq_one_to_five$count==1, 1, 0)
mhq_one_to_five$two <- ifelse(mhq_one_to_five$count==2, 1, 0)
mhq_one_to_five$three <- ifelse(mhq_one_to_five$count==3, 1, 0)
mhq_one_to_five$four <- ifelse(mhq_one_to_five$count==4, 1, 0)
mhq_one_to_five$five <- ifelse(mhq_one_to_five$count==5, 1, 0)

# percentage of MHQ cases with zero to five measures

zero_case <- data.frame(
  "measure" = "Zero Measures",
  "status" = "MHQ Cases",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$zero==1) / sum(mhq_one_to_five$zero==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$zero==1)
  )

one_case <- data.frame(
  "measure" = "One Measure",
  "status" = "MHQ Cases",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$one==1) / sum(mhq_one_to_five$one==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$one==1)
  )

two_case <- data.frame(
  "measure" = "Two Measures",
  "status" = "MHQ Cases",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$two==1) / sum(mhq_one_to_five$two==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$two==1)
  )

three_case <- data.frame(
  "measure" = "Three Measures",
  "status" = "MHQ Cases",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$three==1) / sum(mhq_one_to_five$three==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$three==1)
  )

four_case <- data.frame(
  "measure" = "Four Measures",
  "status" = "MHQ Cases",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$four==1) / sum(mhq_one_to_five$four==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$four==1)
  )

five_case <- data.frame(
  "measure" = "Five Measures",
  "status" = "MHQ Cases",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$five==1) / sum(mhq_one_to_five$five==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Case" & mhq_one_to_five$five==1)
  )

# percentage of MHQ controls with zero to five measures

zero_control <- data.frame(
  "measure" = "Zero Measures",
  "status" = "MHQ Control",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$zero==1) / sum(mhq_one_to_five$zero==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$zero==1)
  )

one_control <- data.frame(
  "measure" = "One Measure",
  "status" = "MHQ Control",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$one==1) / sum(mhq_one_to_five$one==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$one==1)
  )

two_control <- data.frame(
  "measure" = "Two Measures",
  "status" = "MHQ Control",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$two==1) / sum(mhq_one_to_five$two==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$two==1)
  )

three_control <- data.frame(
  "measure" = "Three Measures",
  "status" = "MHQ Control",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$three==1) / sum(mhq_one_to_five$three==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$three==1)
  )

four_control <- data.frame(
  "measure" = "Four Measures",
  "status" = "MHQ Control",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$four==1) / sum(mhq_one_to_five$four==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$four==1)
  )

five_control <- data.frame(
  "measure" = "Five Measures",
  "status" = "MHQ Control",
  "percent" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$five==1) / sum(mhq_one_to_five$five==1),
  "value" = sum(mhq_one_to_five$mhq_status=="MHQ_Control" & mhq_one_to_five$five==1)
  )

# percentage of No CIDI, screened from controls due to other psychopathology on MHQ

zero_mhq_screen <- data.frame(
  "measure" = "Zero Measures",
  "status" = "Screened with MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$zero==1) / sum(mhq_one_to_five$zero==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$zero==1)
  )

one_mhq_screen <- data.frame(
  "measure" = "One Measure",
  "status" = "Screened with MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$one==1) / sum(mhq_one_to_five$one==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$one==1)
  )

two_mhq_screen <- data.frame(
  "measure" = "Two Measures",
  "status" = "Screened with MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$two==1) / sum(mhq_one_to_five$two==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$two==1)
  )

three_mhq_screen <- data.frame(
  "measure" = "Three Measures",
  "status" = "Screened with MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$three==1) / sum(mhq_one_to_five$three==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$three==1)
  )

four_mhq_screen <- data.frame(
  "measure" = "Four Measures",
  "status" = "Screened with MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$four==1) / sum(mhq_one_to_five$four==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$four==1)
  )

five_mhq_screen <- data.frame(
  "measure" = "Five Measures",
  "status" = "Screened with MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$five==1) / sum(mhq_one_to_five$five==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_PHQ_MHQ_Screen" & mhq_one_to_five$five==1)
  )

# percentage of No CIDI, screened from controls due to other depression from non-MHQ source

zero_non_mhq_screen <- data.frame(
  "measure" = "Zero Measures",
  "status" = "Screened outside MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$zero==1) / sum(mhq_one_to_five$zero==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$zero==1)
  )

one_non_mhq_screen <- data.frame(
  "measure" = "One Measure",
  "status" = "Screened outside MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$one==1) / sum(mhq_one_to_five$one==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$one==1)
  )

two_non_mhq_screen <- data.frame(
  "measure" = "Two Measures",
  "status" = "Screened outside MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$two==1) / sum(mhq_one_to_five$two==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$two==1)
  )

three_non_mhq_screen <- data.frame(
  "measure" = "Three Measures",
  "status" = "Screened outside MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$three==1) / sum(mhq_one_to_five$three==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$three==1)
  )

four_non_mhq_screen <- data.frame(
  "measure" = "Four Measures",
  "status" = "Screened outside MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$four==1) / sum(mhq_one_to_five$four==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$four==1)
  )

five_non_mhq_screen <- data.frame(
  "measure" = "Five Measures",
  "status" = "Screened outside MHQ",
  "percent" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$five==1) / sum(mhq_one_to_five$five==1),
  "value" = sum(mhq_one_to_five$mhq_status=="No_CIDI_UKB_Screen" & mhq_one_to_five$five==1)
  )

mhq_percents <- rbind(zero_case,zero_control,zero_mhq_screen,zero_non_mhq_screen,
                      one_case,one_control,one_mhq_screen,one_non_mhq_screen,
                      two_case,two_control,two_mhq_screen,two_non_mhq_screen,
                      three_case,three_control,three_mhq_screen,three_non_mhq_screen,
                      four_case,four_control,four_mhq_screen,four_non_mhq_screen,
                      five_case,five_control,five_mhq_screen,five_non_mhq_screen)

mhq_percents$value <- format(mhq_percents$value, big.mark = ",")
mhq_percents$value <- gsub(" ", "", mhq_percents$value)
mhq_percents$value1 <- ifelse(mhq_percents$value == "0", "", paste("N = ",mhq_percents$value))
mhq_percents[16,5] <- "N =  91\n\n"
mhq_percents[20,5] <- "N =  18\n"
mhq_percents[24,5] <- "N =  1\n"

# Stacked + percent
cols <- c(`MHQ Control`="pink",`MHQ Cases`="hotpink", `Screened with MHQ`="grey80", `Screened outside MHQ`="grey60") 

ggplot(mhq_percents, aes(fill=status, y=percent, x=measure, label = mhq_percents$value1)) +
     geom_bar(position="fill", stat="identity", width = 0.8) +
     geom_text(size = 4.5, position = position_stack(vjust = 0.5)) +
     ylab(expression(paste("\nProportion"))) +
     xlab("\nNumber of depression measures endorsed by MHQ participants") +
     scale_fill_manual(values = cols) +
     theme(axis.text=element_text(size=14),
       axis.title=element_text(size=16),
       panel.grid.minor.x=element_blank(), 
       panel.grid.major.x=element_blank(), 
       panel.grid.minor.y=element_blank(), 
       panel.grid.major.y=element_blank(),
       panel.background = element_rect(fill = "white"),
       axis.line.y = element_line(color="grey"),
       axis.line.x = element_line(color="grey"),
       legend.title = element_text(size = 14),
       legend.text = element_text(size = 14)) +
  scale_fill_manual(values=c("hotpink", "pink", "grey85", "grey65"), 
                       name="MHQ Participants\n",
                       labels=c("MHQ Case (Lifetime Depression)", "MHQ Control", "Excluded (MHQ responses)","Excluded (non-MHQ measures)")) +
 labs(fill = 'MHQ Participants\n') +
  annotate("text", x = 1:6, y = Inf,  vjust=1, fontface=4, colour = "grey40",
           size=5, label = c(paste("Total = ",format(sum(mhq_one_to_five$zero==1), big.mark = ",")),
                             paste("Total = ",format(sum(mhq_one_to_five$one==1), big.mark = ",")),
                             paste("Total = ",format(sum(mhq_one_to_five$two==1), big.mark = ",")),
                             paste("Total = ",format(sum(mhq_one_to_five$three==1), big.mark = ",")),
                             paste("Total = ",format(sum(mhq_one_to_five$four==1), big.mark = ",")),
                             paste("Total = ",format(sum(mhq_one_to_five$five==1), big.mark = ","))))

```

**Plot:** Number of non-MHQ measures of depression in MHQ participants (after screening for psychosis and removing missing data from MHQ). Bars divided into those who met the criteria for CIDI lifetime depression (MHQ Cases), those that did not meet CIDI criteria and had no other depression measures (MHQ Control), those that did not meet CIDI criteria but were screened from MHQ controls because they had other psychopathology on MHQ (Screened with MHQ), those that did not meet CIDI criteria but were screened from MHQ controls because they had depression measures outside of MHQ. 

</br>

```{r new_upset_mhq_case, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=14}

# upset plot of mhq cases by non-MHQ depression measures

# make column called Lifetime Depression (MHQ) and filter for cases only
mhq_one_to_five$`Lifetime Depression (MHQ)` <- ifelse(mhq_one_to_five$mhq_cc_status=="MHQ_Case", 1, 0)
#table(mhq_one_to_five$`Lifetime Depression (MHQ)`)

mhq_one_to_five_cases <- mhq_one_to_five %>%
  filter(`Lifetime Depression (MHQ)`==1)

upset_mhq_cases <- upset(mhq_one_to_five_cases,
                                sets = c("Hospital (ICD-10)", "Self-reported Depression", "Antidepressant Usage", "Depression (Smith)", "Help-seeking", "Lifetime Depression (MHQ)"),
                                mb.ratio = c(0.6, 0.4),
                                order.by = c("freq","degree"),
                                mainbar.y.label = "N people per intersection",
                                sets.x.label = "\nN people per measure",
                                point.size = 2.5,
                                line.size = 0.5,
                                main.bar.color = c("red",
                                 "darkorange1","darkorange1","darkorange1","darkorange1",
                                 "green4","green4","green4","green4","green4","green4",
                                 "blue","blue","blue","blue","blue","blue","blue",
                                 "deeppink","deeppink","deeppink","deeppink",
                                 "turquoise4"),
                                text.scale = c(1.3, 1.3, 1.3, 1.3, 1.5, 1.5))

upset_mhq_cases

```

**Plot:** Overlap MHQ cases only (after screening for psychosis).  

</br>

```{r new_upset_mhq_non_case, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=14}

# upset plot of mhq controls by non-MHQ depression measures

# filter to controls
mhq_one_to_five_non_cases <- mhq_one_to_five %>%
  filter(`Lifetime Depression (MHQ)`==0)

upset_mhq_non_cases <- upset(mhq_one_to_five_non_cases,
                                sets = c("Hospital (ICD-10)", "Self-reported Depression", "Antidepressant Usage", "Depression (Smith)", "Help-seeking", "Lifetime Depression (MHQ)"),
                                mb.ratio = c(0.6, 0.4),
                                order.by = c("freq","degree"),
                                mainbar.y.label = "N people per intersection",
                                sets.x.label = "\nN people per measure",
                                point.size = 2.5,
                                line.size = 0.5,
                                main.bar.color = c("red","red","red","red",
                                 "darkorange1","darkorange1","darkorange1","darkorange1","darkorange1","darkorange1",
                                 "green4","green4","green4","green4","green4","green4","green4",
                                 "blue","blue","blue","blue",
                                 "deeppink"),
                                text.scale = c(1.3, 1.3, 1.3, 1.3, 1.5, 1.5))

upset_mhq_non_cases

```

**Plot:** Overlap MHQ non-cases only (after screening for psychosis). MHQ non-cases are a mixture of MHQ-controls and MHQ exclusions due to screening for other depression measures. 
